// @kbn/ingest-pipelines-plugin/public/application/constants/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UIM_PIPELINE_UPDATE = exports.UIM_PIPELINE_SIMULATE = exports.UIM_PIPELINE_DELETE_MANY = exports.UIM_PIPELINE_DELETE = exports.UIM_PIPELINE_CREATE = exports.UIM_PIPELINES_LIST_LOAD = exports.UIM_APP_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// UI metric constants
const UIM_APP_NAME = 'ingest_pipelines';
exports.UIM_APP_NAME = UIM_APP_NAME;
const UIM_PIPELINES_LIST_LOAD = 'pipelines_list_load';
exports.UIM_PIPELINES_LIST_LOAD = UIM_PIPELINES_LIST_LOAD;
const UIM_PIPELINE_CREATE = 'pipeline_create';
exports.UIM_PIPELINE_CREATE = UIM_PIPELINE_CREATE;
const UIM_PIPELINE_UPDATE = 'pipeline_update';
exports.UIM_PIPELINE_UPDATE = UIM_PIPELINE_UPDATE;
const UIM_PIPELINE_DELETE = 'pipeline_delete';
exports.UIM_PIPELINE_DELETE = UIM_PIPELINE_DELETE;
const UIM_PIPELINE_DELETE_MANY = 'pipeline_delete_many';
exports.UIM_PIPELINE_DELETE_MANY = UIM_PIPELINE_DELETE_MANY;
const UIM_PIPELINE_SIMULATE = 'pipeline_simulate';
exports.UIM_PIPELINE_SIMULATE = UIM_PIPELINE_SIMULATE;
