// @kbn/security-solution-plugin/public/management/pages/policy/view/artifacts/empty/policy_artifacts_empty_unexisting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PolicyArtifactsEmptyUnexisting = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _use_policy_artifacts_empty_hooks = require("./use_policy_artifacts_empty_hooks");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PolicyArtifactsEmptyUnexisting = /*#__PURE__*/(0, _react.memo)(({
  policyId,
  policyName,
  labels,
  canWriteArtifact = false,
  getPolicyArtifactsPath,
  getArtifactPath
}) => {
  const {
    onClickHandler,
    toRouteUrl
  } = (0, _use_policy_artifacts_empty_hooks.useGetLinkTo)(policyId, policyName, getPolicyArtifactsPath, getArtifactPath, {
    show: 'create'
  });
  return /*#__PURE__*/_react.default.createElement(_eui.EuiPageTemplate.EmptyPrompt, {
    color: "subdued",
    iconType: "plusInCircle",
    "data-test-subj": "policy-artifacts-empty-unexisting",
    title: /*#__PURE__*/_react.default.createElement("h2", null, labels.emptyUnexistingTitle),
    body: labels.emptyUnexistingMessage,
    actions: canWriteArtifact ?
    /*#__PURE__*/
    // eslint-disable-next-line @elastic/eui/href-or-on-click
    _react.default.createElement(_eui.EuiButton, {
      color: "primary",
      fill: true,
      onClick: onClickHandler,
      href: toRouteUrl,
      "data-test-subj": "unexisting-manage-artifacts-button"
    }, labels.emptyUnexistingPrimaryActionButtonTitle) : null
  });
});
exports.PolicyArtifactsEmptyUnexisting = PolicyArtifactsEmptyUnexisting;
PolicyArtifactsEmptyUnexisting.displayName = 'PolicyArtifactsEmptyUnexisting';
