// @kbn/observability-shared-plugin/public/components/profiling/embeddables/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EMBEDDABLE_PROFILING_SEARCH_BAR = exports.EMBEDDABLE_FUNCTIONS = exports.EMBEDDABLE_FLAMEGRAPH = void 0;
Object.defineProperty(exports, "EmbeddableFlamegraph", {
  enumerable: true,
  get: function () {
    return _embeddable_flamegraph.EmbeddableFlamegraph;
  }
});
Object.defineProperty(exports, "EmbeddableFunctions", {
  enumerable: true,
  get: function () {
    return _embeddable_functions.EmbeddableFunctions;
  }
});
Object.defineProperty(exports, "EmbeddableProfilingSearchBar", {
  enumerable: true,
  get: function () {
    return _embeddable_profiling_search_bar.EmbeddableProfilingSearchBar;
  }
});
var _embeddable_flamegraph = require("./embeddable_flamegraph");
var _embeddable_functions = require("./embeddable_functions");
var _embeddable_profiling_search_bar = require("./embeddable_profiling_search_bar");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/** Profiling flamegraph embeddable key */
const EMBEDDABLE_FLAMEGRAPH = 'EMBEDDABLE_FLAMEGRAPH';
/** Profiling flamegraph embeddable */
exports.EMBEDDABLE_FLAMEGRAPH = EMBEDDABLE_FLAMEGRAPH;
/** Profiling functions embeddable key */
const EMBEDDABLE_FUNCTIONS = 'EMBEDDABLE_FUNCTIONS';
/** Profiling functions embeddable */
exports.EMBEDDABLE_FUNCTIONS = EMBEDDABLE_FUNCTIONS;
/** Profiling search bar embeddable key */
const EMBEDDABLE_PROFILING_SEARCH_BAR = 'EMBEDDABLE_PROFILING_SEARCH_BAR';
/** Profiling search bar embeddable */
exports.EMBEDDABLE_PROFILING_SEARCH_BAR = EMBEDDABLE_PROFILING_SEARCH_BAR;
