// @kbn/synthetics-plugin/common/field_names.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.URL_PORT = exports.URL_FULL = exports.STATE_ID = exports.OBSERVER_GEO_NAME = exports.MONITOR_TYPE = exports.MONITOR_NAME = exports.MONITOR_ID = exports.ERROR_MESSAGE = exports.CERT_VALID_NOT_BEFORE = exports.CERT_VALID_NOT_AFTER = exports.CERT_ISSUER_NAME = exports.CERT_HASH_SHA256 = exports.CERT_COMMON_NAME = exports.AGENT_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const AGENT_NAME = 'agent.name';
exports.AGENT_NAME = AGENT_NAME;
const MONITOR_ID = 'monitor.id';
exports.MONITOR_ID = MONITOR_ID;
const MONITOR_NAME = 'monitor.name';
exports.MONITOR_NAME = MONITOR_NAME;
const MONITOR_TYPE = 'monitor.type';
exports.MONITOR_TYPE = MONITOR_TYPE;
const URL_FULL = 'url.full';
exports.URL_FULL = URL_FULL;
const URL_PORT = 'url.port';
exports.URL_PORT = URL_PORT;
const OBSERVER_GEO_NAME = 'observer.geo.name';
exports.OBSERVER_GEO_NAME = OBSERVER_GEO_NAME;
const ERROR_MESSAGE = 'error.message';
exports.ERROR_MESSAGE = ERROR_MESSAGE;
const STATE_ID = 'monitor.state.id';
exports.STATE_ID = STATE_ID;
const CERT_COMMON_NAME = 'tls.server.x509.subject.common_name';
exports.CERT_COMMON_NAME = CERT_COMMON_NAME;
const CERT_ISSUER_NAME = 'tls.server.x509.issuer.common_name';
exports.CERT_ISSUER_NAME = CERT_ISSUER_NAME;
const CERT_VALID_NOT_AFTER = 'tls.server.x509.not_after';
exports.CERT_VALID_NOT_AFTER = CERT_VALID_NOT_AFTER;
const CERT_VALID_NOT_BEFORE = 'tls.server.x509.not_before';
exports.CERT_VALID_NOT_BEFORE = CERT_VALID_NOT_BEFORE;
const CERT_HASH_SHA256 = 'tls.server.hash.sha256';
exports.CERT_HASH_SHA256 = CERT_HASH_SHA256;
