// @kbn/threat-intelligence-plugin/public/modules/indicators/components/table/test_ids.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TIMELINE_BUTTON_TEST_ID = exports.TABLE_UPDATE_PROGRESS_TEST_ID = exports.TABLE_TEST_ID = exports.MORE_ACTIONS_TEST_ID = exports.INVESTIGATE_IN_TIMELINE_TEST_ID = exports.FILTER_OUT_BUTTON_TEST_ID = exports.FILTER_IN_BUTTON_TEST_ID = exports.CELL_POPOVER_TIMELINE_BUTTON_TEST_ID = exports.CELL_POPOVER_FILTER_OUT_BUTTON_TEST_ID = exports.CELL_POPOVER_FILTER_IN_BUTTON_TEST_ID = exports.BUTTON_TEST_ID = exports.ADD_TO_NEW_CASE_TEST_ID = exports.ADD_TO_EXISTING_TEST_ID = exports.ADD_TO_BLOCK_LIST_TEST_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/* Table */
const TABLE_TEST_ID = 'tiIndicatorsTable';
exports.TABLE_TEST_ID = TABLE_TEST_ID;
const TABLE_UPDATE_PROGRESS_TEST_ID = `${TABLE_TEST_ID}-updating`;

/* More Actions */
exports.TABLE_UPDATE_PROGRESS_TEST_ID = TABLE_UPDATE_PROGRESS_TEST_ID;
const MORE_ACTIONS_TEST_ID = 'tiIndicatorTableMoreActionsButton';
exports.MORE_ACTIONS_TEST_ID = MORE_ACTIONS_TEST_ID;
const ADD_TO_EXISTING_TEST_ID = 'tiIndicatorTableAddToExistingCaseContextMenu';
exports.ADD_TO_EXISTING_TEST_ID = ADD_TO_EXISTING_TEST_ID;
const ADD_TO_NEW_CASE_TEST_ID = 'tiIndicatorTableAddToNewCaseContextMenu';
exports.ADD_TO_NEW_CASE_TEST_ID = ADD_TO_NEW_CASE_TEST_ID;
const ADD_TO_BLOCK_LIST_TEST_ID = 'tiIndicatorsTableAddToBlockListContextMenu';

/* Open Flyout Button */
exports.ADD_TO_BLOCK_LIST_TEST_ID = ADD_TO_BLOCK_LIST_TEST_ID;
const BUTTON_TEST_ID = 'tiToggleIndicatorFlyoutButton';

/* Cell Popover Renderer */
exports.BUTTON_TEST_ID = BUTTON_TEST_ID;
const CELL_POPOVER_TIMELINE_BUTTON_TEST_ID = 'tiIndicatorsTableCellPopoverTimelineButton';
exports.CELL_POPOVER_TIMELINE_BUTTON_TEST_ID = CELL_POPOVER_TIMELINE_BUTTON_TEST_ID;
const CELL_POPOVER_FILTER_IN_BUTTON_TEST_ID = 'tiIndicatorsTableCellPopoverFilterInButton';
exports.CELL_POPOVER_FILTER_IN_BUTTON_TEST_ID = CELL_POPOVER_FILTER_IN_BUTTON_TEST_ID;
const CELL_POPOVER_FILTER_OUT_BUTTON_TEST_ID = 'tiIndicatorsTableCellPopoverFilterOutButton';

/* Cell Acctions */
exports.CELL_POPOVER_FILTER_OUT_BUTTON_TEST_ID = CELL_POPOVER_FILTER_OUT_BUTTON_TEST_ID;
const TIMELINE_BUTTON_TEST_ID = 'tiIndicatorsTableCellTimelineButton';
exports.TIMELINE_BUTTON_TEST_ID = TIMELINE_BUTTON_TEST_ID;
const FILTER_IN_BUTTON_TEST_ID = 'tiIndicatorsTableCellFilterInButton';
exports.FILTER_IN_BUTTON_TEST_ID = FILTER_IN_BUTTON_TEST_ID;
const FILTER_OUT_BUTTON_TEST_ID = 'tiIndicatorsTableCellFilterOutButton';

/* Actions Row Cell */
exports.FILTER_OUT_BUTTON_TEST_ID = FILTER_OUT_BUTTON_TEST_ID;
const INVESTIGATE_IN_TIMELINE_TEST_ID = 'tiIndicatorTableInvestigateInTimelineButtonIcon';
exports.INVESTIGATE_IN_TIMELINE_TEST_ID = INVESTIGATE_IN_TIMELINE_TEST_ID;
