// @kbn/embeddable-plugin/public/embeddable_panel/panel_actions/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ACTION_CUSTOMIZE_PANEL", {
  enumerable: true,
  get: function () {
    return _customize_panel_action.ACTION_CUSTOMIZE_PANEL;
  }
});
Object.defineProperty(exports, "ACTION_EDIT_PANEL", {
  enumerable: true,
  get: function () {
    return _edit_panel_action.ACTION_EDIT_PANEL;
  }
});
Object.defineProperty(exports, "ACTION_INSPECT_PANEL", {
  enumerable: true,
  get: function () {
    return _inspect_panel_action.ACTION_INSPECT_PANEL;
  }
});
Object.defineProperty(exports, "CustomTimeRangeBadge", {
  enumerable: true,
  get: function () {
    return _customize_panel_action.CustomTimeRangeBadge;
  }
});
Object.defineProperty(exports, "CustomizePanelAction", {
  enumerable: true,
  get: function () {
    return _customize_panel_action.CustomizePanelAction;
  }
});
Object.defineProperty(exports, "EditPanelAction", {
  enumerable: true,
  get: function () {
    return _edit_panel_action.EditPanelAction;
  }
});
Object.defineProperty(exports, "InspectPanelAction", {
  enumerable: true,
  get: function () {
    return _inspect_panel_action.InspectPanelAction;
  }
});
Object.defineProperty(exports, "REMOVE_PANEL_ACTION", {
  enumerable: true,
  get: function () {
    return _remove_panel_action.REMOVE_PANEL_ACTION;
  }
});
Object.defineProperty(exports, "RemovePanelAction", {
  enumerable: true,
  get: function () {
    return _remove_panel_action.RemovePanelAction;
  }
});
Object.defineProperty(exports, "tracksOverlays", {
  enumerable: true,
  get: function () {
    return _track_overlays.tracksOverlays;
  }
});
var _inspect_panel_action = require("./inspect_panel_action/inspect_panel_action");
var _customize_panel_action = require("./customize_panel_action");
var _edit_panel_action = require("./edit_panel_action/edit_panel_action");
var _remove_panel_action = require("./remove_panel_action/remove_panel_action");
var _track_overlays = require("./track_overlays");
