// @kbn/stack-connectors-plugin/public/connector_types/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.XmattersSeverityOptions = exports.XmattersAuthenticationType = exports.SeverityActionOptions = exports.ServerLogLevelOptions = exports.EventActionOptions = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
let EventActionOptions;
exports.EventActionOptions = EventActionOptions;
(function (EventActionOptions) {
  EventActionOptions["TRIGGER"] = "trigger";
  EventActionOptions["RESOLVE"] = "resolve";
  EventActionOptions["ACKNOWLEDGE"] = "acknowledge";
})(EventActionOptions || (exports.EventActionOptions = EventActionOptions = {}));
let SeverityActionOptions;
exports.SeverityActionOptions = SeverityActionOptions;
(function (SeverityActionOptions) {
  SeverityActionOptions["CRITICAL"] = "critical";
  SeverityActionOptions["ERROR"] = "error";
  SeverityActionOptions["WARNING"] = "warning";
  SeverityActionOptions["INFO"] = "info";
})(SeverityActionOptions || (exports.SeverityActionOptions = SeverityActionOptions = {}));
let ServerLogLevelOptions;
exports.ServerLogLevelOptions = ServerLogLevelOptions;
(function (ServerLogLevelOptions) {
  ServerLogLevelOptions["TRACE"] = "trace";
  ServerLogLevelOptions["DEBUG"] = "debug";
  ServerLogLevelOptions["INFO"] = "info";
  ServerLogLevelOptions["WARN"] = "warn";
  ServerLogLevelOptions["ERROR"] = "error";
  ServerLogLevelOptions["FATAL"] = "fatal";
})(ServerLogLevelOptions || (exports.ServerLogLevelOptions = ServerLogLevelOptions = {}));
let XmattersSeverityOptions;
exports.XmattersSeverityOptions = XmattersSeverityOptions;
(function (XmattersSeverityOptions) {
  XmattersSeverityOptions["CRITICAL"] = "critical";
  XmattersSeverityOptions["HIGH"] = "high";
  XmattersSeverityOptions["MEDIUM"] = "medium";
  XmattersSeverityOptions["LOW"] = "low";
  XmattersSeverityOptions["MINIMAL"] = "minimal";
})(XmattersSeverityOptions || (exports.XmattersSeverityOptions = XmattersSeverityOptions = {}));
let XmattersAuthenticationType;
exports.XmattersAuthenticationType = XmattersAuthenticationType;
(function (XmattersAuthenticationType) {
  XmattersAuthenticationType["Basic"] = "Basic Authentication";
  XmattersAuthenticationType["URL"] = "URL Authentication";
})(XmattersAuthenticationType || (exports.XmattersAuthenticationType = XmattersAuthenticationType = {}));
