// @kbn/observability-plugin/public/components/custom_threshold/i18n_strings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LABEL_LABEL = exports.LABEL_HELP_MESSAGE = exports.EQUATION_HELP_MESSAGE = exports.DELETE_LABEL = exports.CUSTOM_EQUATION = exports.AGGREGATION_LABEL = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const EQUATION_HELP_MESSAGE = _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquationEditor.equationHelpMessage', {
  defaultMessage: 'Supports basic math equations, valid charaters are: A-Z, +, -, /, *, (, ), ?, !, &, :, |, >, <, ='
});
exports.EQUATION_HELP_MESSAGE = EQUATION_HELP_MESSAGE;
const LABEL_LABEL = _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquationEditor.labelLabel', {
  defaultMessage: 'Label (optional)'
});
exports.LABEL_LABEL = LABEL_LABEL;
const LABEL_HELP_MESSAGE = _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquationEditor.labelHelpMessage', {
  defaultMessage: 'Custom label will show on the alert chart and in reason'
});
exports.LABEL_HELP_MESSAGE = LABEL_HELP_MESSAGE;
const CUSTOM_EQUATION = _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquation', {
  defaultMessage: 'Custom equation'
});
exports.CUSTOM_EQUATION = CUSTOM_EQUATION;
const DELETE_LABEL = _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquationEditor.deleteRowButton', {
  defaultMessage: 'Delete'
});
exports.DELETE_LABEL = DELETE_LABEL;
const AGGREGATION_LABEL = name => _i18n.i18n.translate('xpack.observability.customThreshold.rule.alertFlyout.customEquationEditor.aggregationLabel', {
  defaultMessage: 'Aggregation {name}',
  values: {
    name
  }
});
exports.AGGREGATION_LABEL = AGGREGATION_LABEL;
