// @kbn/enterprise-search-plugin/public/applications/enterprise_search_content/components/connectors/connectors_logic.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConnectorsLogic = void 0;
var _kea = require("kea");
var _api = require("../../../../../common/types/api");
var _fetch_connectors = require("../../api/connector/fetch_connectors.api");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const ConnectorsLogic = (0, _kea.kea)({
  actions: {
    fetchConnectors: ({
      fetchCrawlersOnly,
      from,
      size,
      searchQuery
    }) => ({
      fetchCrawlersOnly,
      from,
      searchQuery,
      size
    }),
    onPaginate: newPageIndex => ({
      newPageIndex
    }),
    setIsFirstRequest: true
  },
  connect: {
    actions: [_fetch_connectors.FetchConnectorsApiLogic, ['makeRequest', 'apiSuccess', 'apiError']],
    values: [_fetch_connectors.FetchConnectorsApiLogic, ['data', 'status']]
  },
  listeners: ({
    actions
  }) => ({
    fetchConnectors: async (input, breakpoint) => {
      await breakpoint(150);
      actions.makeRequest(input);
    }
  }),
  path: ['enterprise_search', 'content', 'connectors_logic'],
  reducers: () => ({
    isFirstRequest: [true, {
      apiError: () => false,
      apiSuccess: () => false,
      setIsFirstRequest: () => true
    }],
    searchParams: [{
      fetchCrawlersOnly: false,
      from: 0,
      searchQuery: '',
      size: 10
    }, {
      apiSuccess: ({
        fetchCrawlersOnly,
        searchQuery
      }, {
        meta
      }) => ({
        fetchCrawlersOnly,
        from: meta.page.from,
        searchQuery,
        size: meta.page.size
      }),
      onPaginate: (state, {
        newPageIndex
      }) => ({
        ...state,
        from: (newPageIndex - 1) * state.size
      })
    }]
  }),
  selectors: ({
    selectors
  }) => ({
    connectors: [() => [selectors.data], data => {
      return (data === null || data === void 0 ? void 0 : data.connectors.map(connector => {
        const indexName = connector.index_name;
        if (indexName) {
          return {
            ...connector,
            docsCount: data === null || data === void 0 ? void 0 : data.counts[indexName]
          };
        }
        return connector;
      })) || [];
    }],
    isEmpty: [() => [selectors.data], data => {
      var _ref;
      return (_ref = (data === null || data === void 0 ? void 0 : data.isInitialRequest) && (data === null || data === void 0 ? void 0 : data.connectors) && data.connectors.length === 0) !== null && _ref !== void 0 ? _ref : false;
    }],
    isLoading: [() => [selectors.status, selectors.isFirstRequest], (status, isFirstRequest) => [_api.Status.LOADING, _api.Status.IDLE].includes(status) && isFirstRequest],
    meta: [() => [selectors.data], data => {
      var _data$meta;
      return (_data$meta = data === null || data === void 0 ? void 0 : data.meta) !== null && _data$meta !== void 0 ? _data$meta : {
        page: {
          from: 0,
          size: 20,
          total: 0
        }
      };
    }]
  })
});
exports.ConnectorsLogic = ConnectorsLogic;
