// @kbn/infra-plugin/public/pages/metrics/hosts/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LOCAL_STORAGE_PAGE_SIZE_KEY = exports.LOCAL_STORAGE_HOST_LIMIT_KEY = exports.HOST_METRICS_DOC_HREF = exports.HOST_LIMIT_OPTIONS = exports.DEFAULT_PAGE_SIZE = exports.DEFAULT_HOST_LIMIT = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DEFAULT_HOST_LIMIT = 100;
exports.DEFAULT_HOST_LIMIT = DEFAULT_HOST_LIMIT;
const DEFAULT_PAGE_SIZE = 10;
exports.DEFAULT_PAGE_SIZE = DEFAULT_PAGE_SIZE;
const LOCAL_STORAGE_HOST_LIMIT_KEY = 'hostsView:hostLimitSelection';
exports.LOCAL_STORAGE_HOST_LIMIT_KEY = LOCAL_STORAGE_HOST_LIMIT_KEY;
const LOCAL_STORAGE_PAGE_SIZE_KEY = 'hostsView:pageSizeSelection';
exports.LOCAL_STORAGE_PAGE_SIZE_KEY = LOCAL_STORAGE_PAGE_SIZE_KEY;
const HOST_LIMIT_OPTIONS = [50, 100, 500];
exports.HOST_LIMIT_OPTIONS = HOST_LIMIT_OPTIONS;
const HOST_METRICS_DOC_HREF = 'https://ela.st/docs-infra-host-metrics';
exports.HOST_METRICS_DOC_HREF = HOST_METRICS_DOC_HREF;
