// @kbn/threat-intelligence-plugin/public/modules/indicators/components/flyout/test_ids.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TIMELINE_BUTTON_TEST_ID = exports.TAKE_ACTION_BUTTON_TEST_ID = exports.POPOVER_BUTTON_TEST_ID = exports.INVESTIGATE_IN_TIMELINE_TEST_ID = exports.INDICATORS_FLYOUT_TITLE_TEST_ID = exports.INDICATORS_FLYOUT_TABS_TEST_ID = exports.INDICATORS_FLYOUT_SUBTITLE_TEST_ID = exports.INDICATORS_FLYOUT_OVERVIEW_TITLE = exports.INDICATORS_FLYOUT_OVERVIEW_TABLE = exports.INDICATORS_FLYOUT_OVERVIEW_HIGH_LEVEL_BLOCKS = exports.FLYOUT_TABLE_TEST_ID = exports.FILTER_OUT_BUTTON_TEST_ID = exports.FILTER_IN_BUTTON_TEST_ID = exports.COPY_TO_CLIPBOARD_BUTTON_TEST_ID = exports.CODE_BLOCK_TEST_ID = exports.ADD_TO_NEW_CASE_TEST_ID = exports.ADD_TO_EXISTING_CASE_TEST_ID = exports.ADD_TO_BLOCK_LIST_TEST_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/* Flyout */
const INDICATORS_FLYOUT_TITLE_TEST_ID = 'tiIndicatorFlyoutTitle';
exports.INDICATORS_FLYOUT_TITLE_TEST_ID = INDICATORS_FLYOUT_TITLE_TEST_ID;
const INDICATORS_FLYOUT_SUBTITLE_TEST_ID = 'tiIndicatorFlyoutSubtitle';
exports.INDICATORS_FLYOUT_SUBTITLE_TEST_ID = INDICATORS_FLYOUT_SUBTITLE_TEST_ID;
const INDICATORS_FLYOUT_TABS_TEST_ID = 'tiIndicatorFlyoutTabs';

/* Indicator Value Actions */
exports.INDICATORS_FLYOUT_TABS_TEST_ID = INDICATORS_FLYOUT_TABS_TEST_ID;
const TIMELINE_BUTTON_TEST_ID = 'TimelineButton';
exports.TIMELINE_BUTTON_TEST_ID = TIMELINE_BUTTON_TEST_ID;
const FILTER_IN_BUTTON_TEST_ID = 'FilterInButton';
exports.FILTER_IN_BUTTON_TEST_ID = FILTER_IN_BUTTON_TEST_ID;
const FILTER_OUT_BUTTON_TEST_ID = 'FilterOutButton';
exports.FILTER_OUT_BUTTON_TEST_ID = FILTER_OUT_BUTTON_TEST_ID;
const COPY_TO_CLIPBOARD_BUTTON_TEST_ID = 'CopyToClipboardButton';
exports.COPY_TO_CLIPBOARD_BUTTON_TEST_ID = COPY_TO_CLIPBOARD_BUTTON_TEST_ID;
const POPOVER_BUTTON_TEST_ID = 'PopoverButton';

/* Take Action */
exports.POPOVER_BUTTON_TEST_ID = POPOVER_BUTTON_TEST_ID;
const TAKE_ACTION_BUTTON_TEST_ID = 'tiIndicatorFlyoutTakeActionButton';
exports.TAKE_ACTION_BUTTON_TEST_ID = TAKE_ACTION_BUTTON_TEST_ID;
const INVESTIGATE_IN_TIMELINE_TEST_ID = 'tiIndicatorFlyoutInvestigateInTimelineContextMenu';
exports.INVESTIGATE_IN_TIMELINE_TEST_ID = INVESTIGATE_IN_TIMELINE_TEST_ID;
const ADD_TO_EXISTING_CASE_TEST_ID = 'tiIndicatorFlyoutAddToExistingCaseContextMenu';
exports.ADD_TO_EXISTING_CASE_TEST_ID = ADD_TO_EXISTING_CASE_TEST_ID;
const ADD_TO_NEW_CASE_TEST_ID = 'tiIndicatorFlyoutAddToNewCaseContextMenu';
exports.ADD_TO_NEW_CASE_TEST_ID = ADD_TO_NEW_CASE_TEST_ID;
const ADD_TO_BLOCK_LIST_TEST_ID = 'tiIndicatorFlyoutAddToBlockListContextMenu';

/* Json Tab */
exports.ADD_TO_BLOCK_LIST_TEST_ID = ADD_TO_BLOCK_LIST_TEST_ID;
const CODE_BLOCK_TEST_ID = 'tiFlyoutJsonCodeBlock';

/* Table Tab */
exports.CODE_BLOCK_TEST_ID = CODE_BLOCK_TEST_ID;
const FLYOUT_TABLE_TEST_ID = 'tiFlyoutTable';

/* Overview Tab */
exports.FLYOUT_TABLE_TEST_ID = FLYOUT_TABLE_TEST_ID;
const INDICATORS_FLYOUT_OVERVIEW_TITLE = 'tiFlyoutOverviewTitle';
exports.INDICATORS_FLYOUT_OVERVIEW_TITLE = INDICATORS_FLYOUT_OVERVIEW_TITLE;
const INDICATORS_FLYOUT_OVERVIEW_TABLE = 'tiFlyoutOverviewTableRow';
exports.INDICATORS_FLYOUT_OVERVIEW_TABLE = INDICATORS_FLYOUT_OVERVIEW_TABLE;
const INDICATORS_FLYOUT_OVERVIEW_HIGH_LEVEL_BLOCKS = 'tiFlyoutOverviewHighLevelBlocks';
exports.INDICATORS_FLYOUT_OVERVIEW_HIGH_LEVEL_BLOCKS = INDICATORS_FLYOUT_OVERVIEW_HIGH_LEVEL_BLOCKS;
