// @kbn/remote-clusters-plugin/public/application/constants/ui_metric.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UIM_SHOW_DETAILS_CLICK = exports.UIM_CLUSTER_UPDATE = exports.UIM_CLUSTER_REMOVE_MANY = exports.UIM_CLUSTER_REMOVE = exports.UIM_CLUSTER_ADD = exports.UIM_APP_NAME = exports.UIM_APP_LOAD = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const UIM_APP_NAME = 'remote_clusters';
exports.UIM_APP_NAME = UIM_APP_NAME;
const UIM_APP_LOAD = 'app_load';
exports.UIM_APP_LOAD = UIM_APP_LOAD;
const UIM_CLUSTER_ADD = 'cluster_add';
exports.UIM_CLUSTER_ADD = UIM_CLUSTER_ADD;
const UIM_CLUSTER_UPDATE = 'cluster_update';
exports.UIM_CLUSTER_UPDATE = UIM_CLUSTER_UPDATE;
const UIM_CLUSTER_REMOVE = 'cluster_remove';
exports.UIM_CLUSTER_REMOVE = UIM_CLUSTER_REMOVE;
const UIM_CLUSTER_REMOVE_MANY = 'cluster_remove_many';
exports.UIM_CLUSTER_REMOVE_MANY = UIM_CLUSTER_REMOVE_MANY;
const UIM_SHOW_DETAILS_CLICK = 'show_details_click';
exports.UIM_SHOW_DETAILS_CLICK = UIM_SHOW_DETAILS_CLICK;
