// @kbn/synthetics-plugin/common/constants/ui.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UNNAMED_LOCATION = exports.TEST_RUN_DETAILS_ROUTE = exports.SYNTHETIC_CHECK_STEPS_ROUTE = exports.SYNTHETICS_STEP_DETAIL_ROUTE = exports.SYNTHETICS_SETTINGS_ROUTE = exports.SYNTHETICS_INDEX_PATTERN = exports.STEP_DETAIL_ROUTE = exports.STATUS = exports.SETTINGS_ROUTE = exports.PRIVATE_LOCATIOSN_ROUTE = exports.OVERVIEW_ROUTE = exports.MONITOR_TYPES = exports.MONITOR_ROUTE = exports.MONITOR_NOT_FOUND_ROUTE = exports.MONITOR_MANAGEMENT_ROUTE = exports.MONITOR_HISTORY_ROUTE = exports.MONITOR_ERRORS_ROUTE = exports.MONITOR_EDIT_ROUTE = exports.MONITOR_ADD_ROUTE = exports.MONITORS_ROUTE = exports.LICENSE_NOT_SUPPORTED_ERROR = exports.LICENSE_NOT_ACTIVE_ERROR = exports.LICENSE_MISSING_ERROR = exports.INITIAL_REST_VERSION = exports.GETTING_STARTED_ROUTE = exports.ERROR_DETAILS_ROUTE = exports.CERT_STATUS = exports.CERTIFICATES_ROUTE = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const MONITOR_ROUTE = '/monitor/:monitorId?';
exports.MONITOR_ROUTE = MONITOR_ROUTE;
const MONITOR_NOT_FOUND_ROUTE = '/monitor-not-found/:monitorId';
exports.MONITOR_NOT_FOUND_ROUTE = MONITOR_NOT_FOUND_ROUTE;
const MONITOR_HISTORY_ROUTE = '/monitor/:monitorId/history';
exports.MONITOR_HISTORY_ROUTE = MONITOR_HISTORY_ROUTE;
const MONITOR_ERRORS_ROUTE = '/monitor/:monitorId/errors';
exports.MONITOR_ERRORS_ROUTE = MONITOR_ERRORS_ROUTE;
const MONITOR_ADD_ROUTE = '/add-monitor';
exports.MONITOR_ADD_ROUTE = MONITOR_ADD_ROUTE;
const MONITOR_EDIT_ROUTE = '/edit-monitor/:monitorId';
exports.MONITOR_EDIT_ROUTE = MONITOR_EDIT_ROUTE;
const MONITOR_MANAGEMENT_ROUTE = '/manage-monitors';
exports.MONITOR_MANAGEMENT_ROUTE = MONITOR_MANAGEMENT_ROUTE;
const OVERVIEW_ROUTE = '/';
exports.OVERVIEW_ROUTE = OVERVIEW_ROUTE;
const MONITORS_ROUTE = '/monitors';
exports.MONITORS_ROUTE = MONITORS_ROUTE;
const GETTING_STARTED_ROUTE = '/monitors/getting-started';
exports.GETTING_STARTED_ROUTE = GETTING_STARTED_ROUTE;
const SETTINGS_ROUTE = '/settings';
exports.SETTINGS_ROUTE = SETTINGS_ROUTE;
const PRIVATE_LOCATIOSN_ROUTE = '/settings/private-locations';
exports.PRIVATE_LOCATIOSN_ROUTE = PRIVATE_LOCATIOSN_ROUTE;
const SYNTHETICS_SETTINGS_ROUTE = '/settings/:tabId';
exports.SYNTHETICS_SETTINGS_ROUTE = SYNTHETICS_SETTINGS_ROUTE;
const CERTIFICATES_ROUTE = '/certificates';
exports.CERTIFICATES_ROUTE = CERTIFICATES_ROUTE;
const SYNTHETICS_STEP_DETAIL_ROUTE = '/monitor/:monitorId/test-run/:checkGroupId/step/:stepIndex';
exports.SYNTHETICS_STEP_DETAIL_ROUTE = SYNTHETICS_STEP_DETAIL_ROUTE;
const STEP_DETAIL_ROUTE = '/journey/:checkGroupId/step/:stepIndex';
exports.STEP_DETAIL_ROUTE = STEP_DETAIL_ROUTE;
const SYNTHETIC_CHECK_STEPS_ROUTE = '/journey/:checkGroupId/steps';
exports.SYNTHETIC_CHECK_STEPS_ROUTE = SYNTHETIC_CHECK_STEPS_ROUTE;
const TEST_RUN_DETAILS_ROUTE = '/monitor/:monitorId/test-run/:checkGroupId';
exports.TEST_RUN_DETAILS_ROUTE = TEST_RUN_DETAILS_ROUTE;
const ERROR_DETAILS_ROUTE = '/monitor/:monitorId/errors/:errorStateId';
exports.ERROR_DETAILS_ROUTE = ERROR_DETAILS_ROUTE;
let STATUS;
exports.STATUS = STATUS;
(function (STATUS) {
  STATUS["UP"] = "up";
  STATUS["DOWN"] = "down";
  STATUS["COMPLETE"] = "complete";
  STATUS["FAILED"] = "failed";
  STATUS["SKIPPED"] = "skipped";
})(STATUS || (exports.STATUS = STATUS = {}));
let MONITOR_TYPES;
exports.MONITOR_TYPES = MONITOR_TYPES;
(function (MONITOR_TYPES) {
  MONITOR_TYPES["HTTP"] = "http";
  MONITOR_TYPES["TCP"] = "tcp";
  MONITOR_TYPES["ICMP"] = "icmp";
  MONITOR_TYPES["BROWSER"] = "browser";
})(MONITOR_TYPES || (exports.MONITOR_TYPES = MONITOR_TYPES = {}));
const UNNAMED_LOCATION = 'Unnamed-location';
exports.UNNAMED_LOCATION = UNNAMED_LOCATION;
let CERT_STATUS;
exports.CERT_STATUS = CERT_STATUS;
(function (CERT_STATUS) {
  CERT_STATUS["OK"] = "OK";
  CERT_STATUS["EXPIRING_SOON"] = "EXPIRING_SOON";
  CERT_STATUS["EXPIRED"] = "EXPIRED";
  CERT_STATUS["TOO_OLD"] = "TOO_OLD";
})(CERT_STATUS || (exports.CERT_STATUS = CERT_STATUS = {}));
const SYNTHETICS_INDEX_PATTERN = 'synthetics-*';
exports.SYNTHETICS_INDEX_PATTERN = SYNTHETICS_INDEX_PATTERN;
const LICENSE_NOT_ACTIVE_ERROR = 'License not active';
exports.LICENSE_NOT_ACTIVE_ERROR = LICENSE_NOT_ACTIVE_ERROR;
const LICENSE_MISSING_ERROR = 'Missing license information';
exports.LICENSE_MISSING_ERROR = LICENSE_MISSING_ERROR;
const LICENSE_NOT_SUPPORTED_ERROR = 'License not supported';
exports.LICENSE_NOT_SUPPORTED_ERROR = LICENSE_NOT_SUPPORTED_ERROR;
const INITIAL_REST_VERSION = '2023-10-31';
exports.INITIAL_REST_VERSION = INITIAL_REST_VERSION;
