// @kbn/apm-plugin/common/es_fields/infra_metrics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.KUBERNETES_REPLICASET_NAME = exports.KUBERNETES_REPLICASET = exports.KUBERNETES_NAMESPACE_NAME = exports.KUBERNETES_NAMESPACE = exports.KUBERNETES_DEPLOYMENT_NAME = exports.KUBERNETES_DEPLOYMENT = exports.KUBERNETES_CONTAINER_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// Kubernetes
const KUBERNETES_CONTAINER_NAME = 'kubernetes.container.name';
exports.KUBERNETES_CONTAINER_NAME = KUBERNETES_CONTAINER_NAME;
const KUBERNETES_DEPLOYMENT = 'kubernetes.deployment';
exports.KUBERNETES_DEPLOYMENT = KUBERNETES_DEPLOYMENT;
const KUBERNETES_DEPLOYMENT_NAME = 'kubernetes.deployment.name';
exports.KUBERNETES_DEPLOYMENT_NAME = KUBERNETES_DEPLOYMENT_NAME;
const KUBERNETES_NAMESPACE_NAME = 'kubernetes.namespace.name';
exports.KUBERNETES_NAMESPACE_NAME = KUBERNETES_NAMESPACE_NAME;
const KUBERNETES_NAMESPACE = 'kubernetes.namespace';
exports.KUBERNETES_NAMESPACE = KUBERNETES_NAMESPACE;
const KUBERNETES_REPLICASET = 'kubernetes.replicaset';
exports.KUBERNETES_REPLICASET = KUBERNETES_REPLICASET;
const KUBERNETES_REPLICASET_NAME = 'kubernetes.replicaset.name';
exports.KUBERNETES_REPLICASET_NAME = KUBERNETES_REPLICASET_NAME;
