// @kbn/security-solution-plugin/common/api/endpoint/suggestions/get_suggestions.gen.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GetEndpointSuggestionsResponse = exports.GetEndpointSuggestionsRequestParams = exports.GetEndpointSuggestionsRequestBody = void 0;
var _zod = require("zod");
var _common = require("../model/schema/common.gen");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/*
 * NOTICE: Do not edit this file manually.
 * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
 */

const GetEndpointSuggestionsRequestParams = _zod.z.object({
  query: _zod.z.object({
    suggestion_type: _zod.z.literal('eventFilters').optional()
  })
});
exports.GetEndpointSuggestionsRequestParams = GetEndpointSuggestionsRequestParams;
const GetEndpointSuggestionsRequestBody = _zod.z.object({
  field: _zod.z.string().optional(),
  query: _zod.z.string().optional(),
  filters: _zod.z.unknown(),
  fieldMeta: _zod.z.unknown()
});
exports.GetEndpointSuggestionsRequestBody = GetEndpointSuggestionsRequestBody;
const GetEndpointSuggestionsResponse = _common.SuccessResponse;
exports.GetEndpointSuggestionsResponse = GetEndpointSuggestionsResponse;
