// @kbn/dashboard-plugin/public/dashboard_container/embeddable/api/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addFromLibrary", {
  enumerable: true,
  get: function () {
    return _add_panel_from_library.addFromLibrary;
  }
});
Object.defineProperty(exports, "addOrUpdateEmbeddable", {
  enumerable: true,
  get: function () {
    return _panel_management.addOrUpdateEmbeddable;
  }
});
Object.defineProperty(exports, "replacePanel", {
  enumerable: true,
  get: function () {
    return _panel_management.replacePanel;
  }
});
Object.defineProperty(exports, "runClone", {
  enumerable: true,
  get: function () {
    return _run_save_functions.runClone;
  }
});
Object.defineProperty(exports, "runQuickSave", {
  enumerable: true,
  get: function () {
    return _run_save_functions.runQuickSave;
  }
});
Object.defineProperty(exports, "runSaveAs", {
  enumerable: true,
  get: function () {
    return _run_save_functions.runSaveAs;
  }
});
Object.defineProperty(exports, "showSettings", {
  enumerable: true,
  get: function () {
    return _show_settings.showSettings;
  }
});
var _show_settings = require("./show_settings");
var _add_panel_from_library = require("./add_panel_from_library");
var _run_save_functions = require("./run_save_functions");
var _panel_management = require("./panel_management");
