// @kbn/observability-plugin/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.observabilityRuleCreationValidConsumers = exports.observabilityAlertFeatureIds = exports.SLO_BURN_RATE_RULE_TYPE_ID = exports.MEDIUM_PRIORITY_ACTION_ID = exports.MEDIUM_PRIORITY_ACTION = exports.LOW_PRIORITY_ACTION_ID = exports.LOW_PRIORITY_ACTION = exports.INVALID_EQUATION_REGEX = exports.HIGH_PRIORITY_ACTION_ID = exports.HIGH_PRIORITY_ACTION = exports.ALERT_STATUS_ALL = exports.ALERT_ACTION_ID = exports.ALERT_ACTION = exports.ALERTS_URL_STORAGE_KEY = void 0;
var _i18n = require("@kbn/i18n");
var _ruleDataUtils = require("@kbn/rule-data-utils");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const SLO_BURN_RATE_RULE_TYPE_ID = 'slo.rules.burnRate';
exports.SLO_BURN_RATE_RULE_TYPE_ID = SLO_BURN_RATE_RULE_TYPE_ID;
const INVALID_EQUATION_REGEX = /[^A-Z|+|\-|\s|\d+|\.|\(|\)|\/|\*|>|<|=|\?|\:|&|\!|\|]+/g;
exports.INVALID_EQUATION_REGEX = INVALID_EQUATION_REGEX;
const ALERT_STATUS_ALL = 'all';
exports.ALERT_STATUS_ALL = ALERT_STATUS_ALL;
const ALERTS_URL_STORAGE_KEY = '_a';
exports.ALERTS_URL_STORAGE_KEY = ALERTS_URL_STORAGE_KEY;
const ALERT_ACTION_ID = 'slo.burnRate.alert';
exports.ALERT_ACTION_ID = ALERT_ACTION_ID;
const ALERT_ACTION = {
  id: ALERT_ACTION_ID,
  name: _i18n.i18n.translate('xpack.observability.slo.alerting.burnRate.alertAction', {
    defaultMessage: 'Critical'
  })
};
exports.ALERT_ACTION = ALERT_ACTION;
const HIGH_PRIORITY_ACTION_ID = 'slo.burnRate.high';
exports.HIGH_PRIORITY_ACTION_ID = HIGH_PRIORITY_ACTION_ID;
const HIGH_PRIORITY_ACTION = {
  id: HIGH_PRIORITY_ACTION_ID,
  name: _i18n.i18n.translate('xpack.observability.slo.alerting.burnRate.highPriorityAction', {
    defaultMessage: 'High'
  })
};
exports.HIGH_PRIORITY_ACTION = HIGH_PRIORITY_ACTION;
const MEDIUM_PRIORITY_ACTION_ID = 'slo.burnRate.medium';
exports.MEDIUM_PRIORITY_ACTION_ID = MEDIUM_PRIORITY_ACTION_ID;
const MEDIUM_PRIORITY_ACTION = {
  id: MEDIUM_PRIORITY_ACTION_ID,
  name: _i18n.i18n.translate('xpack.observability.slo.alerting.burnRate.mediumPriorityAction', {
    defaultMessage: 'Medium'
  })
};
exports.MEDIUM_PRIORITY_ACTION = MEDIUM_PRIORITY_ACTION;
const LOW_PRIORITY_ACTION_ID = 'slo.burnRate.low';
exports.LOW_PRIORITY_ACTION_ID = LOW_PRIORITY_ACTION_ID;
const LOW_PRIORITY_ACTION = {
  id: LOW_PRIORITY_ACTION_ID,
  name: _i18n.i18n.translate('xpack.observability.slo.alerting.burnRate.lowPriorityAction', {
    defaultMessage: 'Low'
  })
};
exports.LOW_PRIORITY_ACTION = LOW_PRIORITY_ACTION;
const observabilityAlertFeatureIds = [_ruleDataUtils.AlertConsumers.APM, _ruleDataUtils.AlertConsumers.INFRASTRUCTURE, _ruleDataUtils.AlertConsumers.LOGS, _ruleDataUtils.AlertConsumers.UPTIME, _ruleDataUtils.AlertConsumers.SLO, _ruleDataUtils.AlertConsumers.OBSERVABILITY];
exports.observabilityAlertFeatureIds = observabilityAlertFeatureIds;
const observabilityRuleCreationValidConsumers = [_ruleDataUtils.AlertConsumers.INFRASTRUCTURE, _ruleDataUtils.AlertConsumers.LOGS, _ruleDataUtils.AlertConsumers.OBSERVABILITY];
exports.observabilityRuleCreationValidConsumers = observabilityRuleCreationValidConsumers;
