// @kbn/observability-plugin/common/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ProcessorEvent", {
  enumerable: true,
  get: function () {
    return _processor_event.ProcessorEvent;
  }
});
Object.defineProperty(exports, "ProgressiveLoadingQuality", {
  enumerable: true,
  get: function () {
    return _progressive_loading.ProgressiveLoadingQuality;
  }
});
exports.alertsLocatorID = void 0;
Object.defineProperty(exports, "apmAWSLambdaPriceFactor", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmAWSLambdaPriceFactor;
  }
});
Object.defineProperty(exports, "apmAWSLambdaRequestCostPerMillion", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmAWSLambdaRequestCostPerMillion;
  }
});
Object.defineProperty(exports, "apmEnableContinuousRollups", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmEnableContinuousRollups;
  }
});
Object.defineProperty(exports, "apmEnableProfilingIntegration", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmEnableProfilingIntegration;
  }
});
Object.defineProperty(exports, "apmEnableServiceMetrics", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmEnableServiceMetrics;
  }
});
Object.defineProperty(exports, "apmLabsButton", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmLabsButton;
  }
});
Object.defineProperty(exports, "apmProgressiveLoading", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmProgressiveLoading;
  }
});
Object.defineProperty(exports, "apmServiceGroupMaxNumberOfServices", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmServiceGroupMaxNumberOfServices;
  }
});
Object.defineProperty(exports, "apmServiceInventoryOptimizedSorting", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmServiceInventoryOptimizedSorting;
  }
});
Object.defineProperty(exports, "apmTraceExplorerTab", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.apmTraceExplorerTab;
  }
});
Object.defineProperty(exports, "asAbsoluteDateTime", {
  enumerable: true,
  get: function () {
    return _formatters.asAbsoluteDateTime;
  }
});
Object.defineProperty(exports, "asDuration", {
  enumerable: true,
  get: function () {
    return _formatters.asDuration;
  }
});
Object.defineProperty(exports, "asDynamicBytes", {
  enumerable: true,
  get: function () {
    return _formatters.asDynamicBytes;
  }
});
Object.defineProperty(exports, "asInteger", {
  enumerable: true,
  get: function () {
    return _formatters.asInteger;
  }
});
Object.defineProperty(exports, "asPercent", {
  enumerable: true,
  get: function () {
    return _formatters.asPercent;
  }
});
exports.casesFeatureId = void 0;
Object.defineProperty(exports, "defaultApmServiceEnvironment", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.defaultApmServiceEnvironment;
  }
});
Object.defineProperty(exports, "enableAgentExplorerView", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableAgentExplorerView;
  }
});
Object.defineProperty(exports, "enableAwsLambdaMetrics", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableAwsLambdaMetrics;
  }
});
Object.defineProperty(exports, "enableComparisonByDefault", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableComparisonByDefault;
  }
});
Object.defineProperty(exports, "enableCriticalPath", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableCriticalPath;
  }
});
Object.defineProperty(exports, "enableInfrastructureHostsView", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableInfrastructureHostsView;
  }
});
Object.defineProperty(exports, "enableInfrastructureProfilingIntegration", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableInfrastructureProfilingIntegration;
  }
});
Object.defineProperty(exports, "enableInspectEsQueries", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.enableInspectEsQueries;
  }
});
Object.defineProperty(exports, "formatDurationFromTimeUnitChar", {
  enumerable: true,
  get: function () {
    return _formatters.formatDurationFromTimeUnitChar;
  }
});
Object.defineProperty(exports, "getAlertDetailsUrl", {
  enumerable: true,
  get: function () {
    return _alert_url.getAlertDetailsUrl;
  }
});
Object.defineProperty(exports, "getAlertUrl", {
  enumerable: true,
  get: function () {
    return _alert_url.getAlertUrl;
  }
});
Object.defineProperty(exports, "getDurationFormatter", {
  enumerable: true,
  get: function () {
    return _formatters.getDurationFormatter;
  }
});
Object.defineProperty(exports, "getInspectResponse", {
  enumerable: true,
  get: function () {
    return _get_inspect_response.getInspectResponse;
  }
});
Object.defineProperty(exports, "getProbabilityFromProgressiveLoadingQuality", {
  enumerable: true,
  get: function () {
    return _progressive_loading.getProbabilityFromProgressiveLoadingQuality;
  }
});
Object.defineProperty(exports, "maxSuggestions", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.maxSuggestions;
  }
});
exports.observabilityPaths = exports.observabilityFeatureId = exports.observabilityAppId = void 0;
Object.defineProperty(exports, "profilingAWSCostDiscountRate", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingAWSCostDiscountRate;
  }
});
Object.defineProperty(exports, "profilingCo2PerKWH", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingCo2PerKWH;
  }
});
Object.defineProperty(exports, "profilingCostPervCPUPerHour", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingCostPervCPUPerHour;
  }
});
Object.defineProperty(exports, "profilingDatacenterPUE", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingDatacenterPUE;
  }
});
Object.defineProperty(exports, "profilingPervCPUWattArm64", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingPervCPUWattArm64;
  }
});
Object.defineProperty(exports, "profilingPervCPUWattX86", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingPervCPUWattX86;
  }
});
Object.defineProperty(exports, "profilingUseLegacyCo2Calculation", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.profilingUseLegacyCo2Calculation;
  }
});
exports.syntheticsSettingsLocatorID = exports.syntheticsMonitorDetailLocatorID = exports.syntheticsEditMonitorLocatorID = exports.sloListLocatorID = exports.sloFeatureId = exports.sloEditLocatorID = exports.sloDetailsLocatorID = exports.rulesLocatorID = exports.ruleDetailsLocatorID = void 0;
Object.defineProperty(exports, "syntheticsThrottlingEnabled", {
  enumerable: true,
  get: function () {
    return _ui_settings_keys.syntheticsThrottlingEnabled;
  }
});
exports.uptimeOverviewLocatorID = void 0;
var _formatters = require("./utils/formatters");
var _get_inspect_response = require("./utils/get_inspect_response");
var _alert_url = require("./utils/alerting/alert_url");
var _processor_event = require("./processor_event");
var _ui_settings_keys = require("./ui_settings_keys");
var _progressive_loading = require("./progressive_loading");
var _paths = require("./locators/paths");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const sloFeatureId = 'slo';
exports.sloFeatureId = sloFeatureId;
const casesFeatureId = 'observabilityCases';

// The ID of the observability app. Should more appropriately be called
// 'observability' but it's used in telemetry by applicationUsage so we don't
// want to change it.
exports.casesFeatureId = casesFeatureId;
const observabilityAppId = 'observability-overview';

// Used by feature and "solution" registration
exports.observabilityAppId = observabilityAppId;
const observabilityFeatureId = 'observability';

// Name of a locator created by the uptime plugin. Intended for use
// by other plugins as well, so defined here to prevent cross-references.
exports.observabilityFeatureId = observabilityFeatureId;
const uptimeOverviewLocatorID = 'UPTIME_OVERVIEW_LOCATOR';
exports.uptimeOverviewLocatorID = uptimeOverviewLocatorID;
const syntheticsMonitorDetailLocatorID = 'SYNTHETICS_MONITOR_DETAIL_LOCATOR';
exports.syntheticsMonitorDetailLocatorID = syntheticsMonitorDetailLocatorID;
const syntheticsEditMonitorLocatorID = 'SYNTHETICS_EDIT_MONITOR_LOCATOR';
exports.syntheticsEditMonitorLocatorID = syntheticsEditMonitorLocatorID;
const syntheticsSettingsLocatorID = 'SYNTHETICS_SETTINGS';
exports.syntheticsSettingsLocatorID = syntheticsSettingsLocatorID;
const alertsLocatorID = 'ALERTS_LOCATOR';
exports.alertsLocatorID = alertsLocatorID;
const ruleDetailsLocatorID = 'RULE_DETAILS_LOCATOR';
exports.ruleDetailsLocatorID = ruleDetailsLocatorID;
const rulesLocatorID = 'RULES_LOCATOR';
exports.rulesLocatorID = rulesLocatorID;
const sloDetailsLocatorID = 'SLO_DETAILS_LOCATOR';
exports.sloDetailsLocatorID = sloDetailsLocatorID;
const sloEditLocatorID = 'SLO_EDIT_LOCATOR';
exports.sloEditLocatorID = sloEditLocatorID;
const sloListLocatorID = 'SLO_LIST_LOCATOR';
exports.sloListLocatorID = sloListLocatorID;
const observabilityPaths = _paths.paths.observability;
exports.observabilityPaths = observabilityPaths;
