// @kbn/management-settings-ids/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SECURITY_SOLUTION_DEFAULT_INDEX_ID = exports.SECURITY_SOLUTION_DEFAULT_ANOMALY_SCORE_ID = exports.SECURITY_SOLUTION_DEFAULT_ALERT_TAGS_KEY = exports.SEARCH_TIMEOUT_ID = exports.SEARCH_QUERY_LANGUAGE_ID = exports.SEARCH_INCLUDE_FROZEN_ID = exports.SAVED_OBJECTS_PER_PAGE_ID = exports.SAVED_OBJECTS_LISTING_LIMIT_ID = exports.ROLLUPS_ENABLE_INDEX_PATTERNS_ID = exports.QUERY_STRING_OPTIONS_ID = exports.QUERY_ALLOW_LEADING_WILDCARDS_ID = exports.OBSERVABILITY_PROFILING_ELASTICSEARCH_PLUGIN_ID = exports.OBSERVABILITY_MAX_SUGGESTIONS_ID = exports.OBSERVABILITY_ENABLE_INSPECT_ES_QUERIES_ID = exports.OBSERVABILITY_ENABLE_INFRASTRUCTURE_HOSTS_VIEW_ID = exports.OBSERVABILITY_ENABLE_COMPARISON_BY_DEFAULT_ID = exports.OBSERVABILITY_ENABLE_AWS_LAMBDA_METRICS_ID = exports.OBSERVABILITY_APM_TRACE_EXPLORER_TAB_ID = exports.OBSERVABILITY_APM_SERVICE_INVENTORY_OPTIMIZED_SORTING_ID = exports.OBSERVABILITY_APM_SERVICE_GROUP_MAX_NUMBER_OF_SERVCIE_ID = exports.OBSERVABILITY_APM_PROGRESSIVE_LOADING_ID = exports.OBSERVABILITY_APM_LABS_BUTTON_ID = exports.OBSERVABILITY_APM_ENABLE_SERVICE_METRICS_ID = exports.OBSERVABILITY_APM_ENABLE_PROFILING_INTEGRATION_ID = exports.OBSERVABILITY_APM_ENABLE_CRITICAL_PATH_ID = exports.OBSERVABILITY_APM_ENABLE_CONTINUOUS_ROLLUPS_ID = exports.OBSERVABILITY_APM_DEFAULT_SERVICE_ENVIRONMENT_ID = exports.OBSERVABILITY_APM_AWS_LAMBDA_REQUEST_COST_PER_MILLION_ID = exports.OBSERVABILITY_APM_AWS_LAMBDA_PRICE_FACTOR_ID = exports.OBSERVABILITY_APM_AGENT_EXPLORER_VIEW_ID = exports.NOTIFICATIONS_LIFETIME_WARNING_ID = exports.NOTIFICATIONS_LIFETIME_INFO_ID = exports.NOTIFICATIONS_LIFETIME_ERROR_ID = exports.NOTIFICATIONS_LIFETIME_BANNER_ID = exports.NOTIFICATIONS_BANNER_ID = exports.ML_ANOMALY_DETECTION_RESULTS_TIME_DEFAULTS_ID = exports.ML_ANOMALY_DETECTION_RESULTS_ENABLE_TIME_DEFAULTS_ID = exports.METRICS_MAX_BUCKETS_ID = exports.METRICS_ALLOW_STRING_INDICES_ID = exports.METRICS_ALLOW_CHECKING_FOR_FAILED_SHARDS_ID = exports.META_FIELDS_ID = exports.LABS_DASHBOARD_DEFER_BELOW_FOLD_ID = exports.LABS_DASHBOARDS_ENABLE_UI_ID = exports.LABS_CANVAS_ENABLE_UI_ID = exports.LABS_CANVAS_BY_VALUE_EMBEDDABLE_ID = exports.HISTORY_LIMIT_ID = exports.HISTOGRAM_MAX_BARS_ID = exports.HISTOGRAM_BAR_TARGET_ID = exports.HIDE_ANNOUNCEMENTS_ID = exports.FORMAT_PERCENT_DEFAULT_PATTERN_ID = exports.FORMAT_NUMBER_DEFAULT_PATTERN_ID = exports.FORMAT_NUMBER_DEFAULT_LOCALE_ID = exports.FORMAT_DEFAULT_TYPE_MAP_ID = exports.FORMAT_CURRENCY_DEFAULT_PATTERN_ID = exports.FORMAT_BYTES_DEFAULT_PATTERN_ID = exports.FILTER_EDITOR_SUGGEST_VALUES_ID = exports.FILTERS_PINNED_BY_DEFAULT_ID = exports.FILE_UPLOAD_MAX_SIZE_ID = exports.FIELDS_POPULAR_LIMIT_ID = exports.DOC_TABLE_LEGACY_ID = exports.DOC_TABLE_HIGHLIGHT_ID = exports.DOC_TABLE_HIDE_TIME_COLUMNS_ID = exports.DISCOVER_SORT_DEFAULT_ORDER_ID = exports.DISCOVER_SHOW_MULTI_FIELDS_ID = exports.DISCOVER_SHOW_LEGACY_FIELD_TOP_VALUES_ID = exports.DISCOVER_SHOW_FIELD_STATISTICS_ID = exports.DISCOVER_SEARCH_ON_PAGE_LOAD_ID = exports.DISCOVER_SEARCH_FIELDS_FROM_SOURCE_ID = exports.DISCOVER_SAMPLE_SIZE_ID = exports.DISCOVER_SAMPLE_ROWS_PER_PAGE_ID = exports.DISCOVER_ROW_HEIGHT_OPTION_ID = exports.DISCOVER_MODIFY_COLUMNS_ON_SWITCH_ID = exports.DISCOVER_MAX_DOC_FIELDS_DISPLAYED_ID = exports.DISCOVER_ENABLE_SQL_ID = exports.DISABLE_REQUEST_BATCHING_ID = exports.DISABLE_BATCH_COMPRESSION_ID = exports.DEFAULT_ROUTE_ID = exports.DEFAULT_INDEX_ID = exports.DEFAULT_COLUMNS_ID = exports.DATE_FORMAT_TZ_ID = exports.DATE_FORMAT_SCALED_ID = exports.DATE_FORMAT_NANOS_ID = exports.DATE_FORMAT_ID = exports.DATE_FORMAT_DOW_ID = exports.CSV_SEPARATOR_ID = exports.CSV_QUOTE_VALUES_ID = exports.COURIER_SET_REQUEST_PREFERENCE_ID = exports.COURIER_MAX_CONCURRENT_SHARD_REQUEST_ID = exports.COURIER_IGNORE_FILTER_IF_FIELD_NOT_IN_INDEX_ID = exports.COURIER_CUSTOM_REQUEST_PREFERENCE_ID = exports.CONTEXT_TIE_BREAKER_FIELDS_ID = exports.CONTEXT_STEP_ID = exports.CONTEXT_DEFAULT_SIZE_ID = exports.BANNERS_TEXT_CONTENT_ID = exports.BANNERS_TEXT_COLOR_ID = exports.BANNERS_PLACEMENT_ID = exports.BANNERS_BACKGROUND_COLOR_ID = exports.AUTOCOMPLETE_VALUE_SUGGESTION_METHOD_ID = exports.AUTOCOMPLETE_USE_TIME_RANGE_ID = exports.ACCESSIBILITY_DISABLE_ANIMATIONS_ID = void 0;
exports.XPACK_REPORTING_CUSTOM_PDF_LOGO_ID = exports.VISUALIZATION_USE_LEGACY_TIME_AXIS_ID = exports.VISUALIZATION_LEGACY_HEATMAP_CHARTS_LIBRARY_ID = exports.VISUALIZATION_LEGACY_GAUGE_CHARTS_LIBRARY_ID = exports.VISUALIZATION_HEATMAP_MAX_BUCKETS_ID = exports.VISUALIZATION_ENABLE_LABS_ID = exports.VISUALIZATION_COLOR_MAPPING_ID = exports.TRUNCATE_MAX_HEIGHT_ID = exports.TIMEPICKER_TIME_DEFAULTS_ID = exports.TIMEPICKER_REFRESH_INTERVAL_DEFAULTS_ID = exports.TIMEPICKER_QUICK_RANGES_ID = exports.TIMELION_TARGET_BUCKETS_ID = exports.TIMELION_MIN_INTERVAL_ID = exports.TIMELION_MAX_BUCKETS_ID = exports.TIMELION_ES_TIME_FIELD_ID = exports.TIMELION_ES_DEFAULT_INDEX_ID = exports.THEME_DARK_MODE_ID = exports.STATE_STORE_IN_SESSION_STORAGE_ID = exports.SORT_OPTIONS_ID = exports.SHORT_DOTS_ENABLE_ID = exports.SECURITY_SOLUTION_TIME_DEFAULTS_ID = exports.SECURITY_SOLUTION_SHOW_RELATED_INTEGRATIONS_ID = exports.SECURITY_SOLUTION_RULES_TABLE_REFRESH_ID = exports.SECURITY_SOLUTION_REFRESH_INTERVAL_DEFAULTS_ID = exports.SECURITY_SOLUTION_NEWS_FEED_URL_ID = exports.SECURITY_SOLUTION_IP_REPUTATION_LINKS_ID = exports.SECURITY_SOLUTION_EXCLUDE_COLD_AND_FROZEN_TIERS_IN_ANALYZER = exports.SECURITY_SOLUTION_ENABLE_NEWS_FEED_ID = exports.SECURITY_SOLUTION_ENABLE_GROUPED_NAV_ID = exports.SECURITY_SOLUTION_ENABLE_EXPANDABLE_FLYOUT_SETTING = exports.SECURITY_SOLUTION_ENABLE_CCS_WARNING_ID = exports.SECURITY_SOLUTION_DEFAULT_THREAT_INDEX_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

// General settings
const DISABLE_REQUEST_BATCHING_ID = 'bfetch:disable';
exports.DISABLE_REQUEST_BATCHING_ID = DISABLE_REQUEST_BATCHING_ID;
const DISABLE_BATCH_COMPRESSION_ID = 'bfetch:disableCompression';
exports.DISABLE_BATCH_COMPRESSION_ID = DISABLE_BATCH_COMPRESSION_ID;
const CSV_QUOTE_VALUES_ID = 'csv:quoteValues';
exports.CSV_QUOTE_VALUES_ID = CSV_QUOTE_VALUES_ID;
const CSV_SEPARATOR_ID = 'csv:separator';
exports.CSV_SEPARATOR_ID = CSV_SEPARATOR_ID;
const DATE_FORMAT_ID = 'dateFormat';
exports.DATE_FORMAT_ID = DATE_FORMAT_ID;
const DATE_FORMAT_DOW_ID = 'dateFormat:dow';
exports.DATE_FORMAT_DOW_ID = DATE_FORMAT_DOW_ID;
const DATE_FORMAT_SCALED_ID = 'dateFormat:scaled';
exports.DATE_FORMAT_SCALED_ID = DATE_FORMAT_SCALED_ID;
const DATE_FORMAT_TZ_ID = 'dateFormat:tz';
exports.DATE_FORMAT_TZ_ID = DATE_FORMAT_TZ_ID;
const DATE_FORMAT_NANOS_ID = 'dateNanosFormat';
exports.DATE_FORMAT_NANOS_ID = DATE_FORMAT_NANOS_ID;
const DEFAULT_INDEX_ID = 'defaultIndex';
exports.DEFAULT_INDEX_ID = DEFAULT_INDEX_ID;
const DEFAULT_ROUTE_ID = 'defaultRoute';
exports.DEFAULT_ROUTE_ID = DEFAULT_ROUTE_ID;
const FIELDS_POPULAR_LIMIT_ID = 'fields:popularLimit';
exports.FIELDS_POPULAR_LIMIT_ID = FIELDS_POPULAR_LIMIT_ID;
const FILE_UPLOAD_MAX_SIZE_ID = 'fileUpload:maxFileSize';
exports.FILE_UPLOAD_MAX_SIZE_ID = FILE_UPLOAD_MAX_SIZE_ID;
const FILTER_EDITOR_SUGGEST_VALUES_ID = 'filterEditor:suggestValues';
exports.FILTER_EDITOR_SUGGEST_VALUES_ID = FILTER_EDITOR_SUGGEST_VALUES_ID;
const FILTERS_PINNED_BY_DEFAULT_ID = 'filters:pinnedByDefault';
exports.FILTERS_PINNED_BY_DEFAULT_ID = FILTERS_PINNED_BY_DEFAULT_ID;
const FORMAT_BYTES_DEFAULT_PATTERN_ID = 'format:bytes:defaultPattern';
exports.FORMAT_BYTES_DEFAULT_PATTERN_ID = FORMAT_BYTES_DEFAULT_PATTERN_ID;
const FORMAT_CURRENCY_DEFAULT_PATTERN_ID = 'format:currency:defaultPattern';
exports.FORMAT_CURRENCY_DEFAULT_PATTERN_ID = FORMAT_CURRENCY_DEFAULT_PATTERN_ID;
const FORMAT_DEFAULT_TYPE_MAP_ID = 'format:defaultTypeMap';
exports.FORMAT_DEFAULT_TYPE_MAP_ID = FORMAT_DEFAULT_TYPE_MAP_ID;
const FORMAT_NUMBER_DEFAULT_LOCALE_ID = 'format:number:defaultLocale';
exports.FORMAT_NUMBER_DEFAULT_LOCALE_ID = FORMAT_NUMBER_DEFAULT_LOCALE_ID;
const FORMAT_NUMBER_DEFAULT_PATTERN_ID = 'format:number:defaultPattern';
exports.FORMAT_NUMBER_DEFAULT_PATTERN_ID = FORMAT_NUMBER_DEFAULT_PATTERN_ID;
const FORMAT_PERCENT_DEFAULT_PATTERN_ID = 'format:percent:defaultPattern';
exports.FORMAT_PERCENT_DEFAULT_PATTERN_ID = FORMAT_PERCENT_DEFAULT_PATTERN_ID;
const HIDE_ANNOUNCEMENTS_ID = 'hideAnnouncements';
exports.HIDE_ANNOUNCEMENTS_ID = HIDE_ANNOUNCEMENTS_ID;
const HISTOGRAM_BAR_TARGET_ID = 'histogram:barTarget';
exports.HISTOGRAM_BAR_TARGET_ID = HISTOGRAM_BAR_TARGET_ID;
const HISTOGRAM_MAX_BARS_ID = 'histogram:maxBars';
exports.HISTOGRAM_MAX_BARS_ID = HISTOGRAM_MAX_BARS_ID;
const HISTORY_LIMIT_ID = 'history:limit';
exports.HISTORY_LIMIT_ID = HISTORY_LIMIT_ID;
const META_FIELDS_ID = 'metaFields';
exports.META_FIELDS_ID = META_FIELDS_ID;
const METRICS_ALLOW_CHECKING_FOR_FAILED_SHARDS_ID = 'metrics:allowCheckingForFailedShards';
exports.METRICS_ALLOW_CHECKING_FOR_FAILED_SHARDS_ID = METRICS_ALLOW_CHECKING_FOR_FAILED_SHARDS_ID;
const METRICS_ALLOW_STRING_INDICES_ID = 'metrics:allowStringIndices';
exports.METRICS_ALLOW_STRING_INDICES_ID = METRICS_ALLOW_STRING_INDICES_ID;
const METRICS_MAX_BUCKETS_ID = 'metrics:max_buckets';
exports.METRICS_MAX_BUCKETS_ID = METRICS_MAX_BUCKETS_ID;
const QUERY_ALLOW_LEADING_WILDCARDS_ID = 'query:allowLeadingWildcards';
exports.QUERY_ALLOW_LEADING_WILDCARDS_ID = QUERY_ALLOW_LEADING_WILDCARDS_ID;
const QUERY_STRING_OPTIONS_ID = 'query:queryString:options';
exports.QUERY_STRING_OPTIONS_ID = QUERY_STRING_OPTIONS_ID;
const SAVED_OBJECTS_LISTING_LIMIT_ID = 'savedObjects:listingLimit';
exports.SAVED_OBJECTS_LISTING_LIMIT_ID = SAVED_OBJECTS_LISTING_LIMIT_ID;
const SAVED_OBJECTS_PER_PAGE_ID = 'savedObjects:perPage';
exports.SAVED_OBJECTS_PER_PAGE_ID = SAVED_OBJECTS_PER_PAGE_ID;
const SEARCH_QUERY_LANGUAGE_ID = 'search:queryLanguage';
exports.SEARCH_QUERY_LANGUAGE_ID = SEARCH_QUERY_LANGUAGE_ID;
const SHORT_DOTS_ENABLE_ID = 'shortDots:enable';
exports.SHORT_DOTS_ENABLE_ID = SHORT_DOTS_ENABLE_ID;
const SORT_OPTIONS_ID = 'sort:options';
exports.SORT_OPTIONS_ID = SORT_OPTIONS_ID;
const STATE_STORE_IN_SESSION_STORAGE_ID = 'state:storeInSessionStorage';
exports.STATE_STORE_IN_SESSION_STORAGE_ID = STATE_STORE_IN_SESSION_STORAGE_ID;
const THEME_DARK_MODE_ID = 'theme:darkMode';
exports.THEME_DARK_MODE_ID = THEME_DARK_MODE_ID;
const TIMEPICKER_QUICK_RANGES_ID = 'timepicker:quickRanges';
exports.TIMEPICKER_QUICK_RANGES_ID = TIMEPICKER_QUICK_RANGES_ID;
const TIMEPICKER_REFRESH_INTERVAL_DEFAULTS_ID = 'timepicker:refreshIntervalDefaults';
exports.TIMEPICKER_REFRESH_INTERVAL_DEFAULTS_ID = TIMEPICKER_REFRESH_INTERVAL_DEFAULTS_ID;
const TIMEPICKER_TIME_DEFAULTS_ID = 'timepicker:timeDefaults';

// Presentation labs settings
exports.TIMEPICKER_TIME_DEFAULTS_ID = TIMEPICKER_TIME_DEFAULTS_ID;
const LABS_CANVAS_BY_VALUE_EMBEDDABLE_ID = 'labs:canvas:byValueEmbeddable';
exports.LABS_CANVAS_BY_VALUE_EMBEDDABLE_ID = LABS_CANVAS_BY_VALUE_EMBEDDABLE_ID;
const LABS_CANVAS_ENABLE_UI_ID = 'labs:canvas:enable_ui';
exports.LABS_CANVAS_ENABLE_UI_ID = LABS_CANVAS_ENABLE_UI_ID;
const LABS_DASHBOARD_DEFER_BELOW_FOLD_ID = 'labs:dashboard:deferBelowFold';
exports.LABS_DASHBOARD_DEFER_BELOW_FOLD_ID = LABS_DASHBOARD_DEFER_BELOW_FOLD_ID;
const LABS_DASHBOARDS_ENABLE_UI_ID = 'labs:dashboard:enable_ui';

// Accessibility settings
exports.LABS_DASHBOARDS_ENABLE_UI_ID = LABS_DASHBOARDS_ENABLE_UI_ID;
const ACCESSIBILITY_DISABLE_ANIMATIONS_ID = 'accessibility:disableAnimations';

// Autocomplete settings
exports.ACCESSIBILITY_DISABLE_ANIMATIONS_ID = ACCESSIBILITY_DISABLE_ANIMATIONS_ID;
const AUTOCOMPLETE_USE_TIME_RANGE_ID = 'autocomplete:useTimeRange';
exports.AUTOCOMPLETE_USE_TIME_RANGE_ID = AUTOCOMPLETE_USE_TIME_RANGE_ID;
const AUTOCOMPLETE_VALUE_SUGGESTION_METHOD_ID = 'autocomplete:valueSuggestionMethod';

// Banner settings
exports.AUTOCOMPLETE_VALUE_SUGGESTION_METHOD_ID = AUTOCOMPLETE_VALUE_SUGGESTION_METHOD_ID;
const BANNERS_PLACEMENT_ID = 'banners:placement';
exports.BANNERS_PLACEMENT_ID = BANNERS_PLACEMENT_ID;
const BANNERS_TEXT_CONTENT_ID = 'banners:textContent';
exports.BANNERS_TEXT_CONTENT_ID = BANNERS_TEXT_CONTENT_ID;
const BANNERS_TEXT_COLOR_ID = 'banners:textColor';
exports.BANNERS_TEXT_COLOR_ID = BANNERS_TEXT_COLOR_ID;
const BANNERS_BACKGROUND_COLOR_ID = 'banners:backgroundColor';

// Discover settings
exports.BANNERS_BACKGROUND_COLOR_ID = BANNERS_BACKGROUND_COLOR_ID;
const CONTEXT_DEFAULT_SIZE_ID = 'context:defaultSize';
exports.CONTEXT_DEFAULT_SIZE_ID = CONTEXT_DEFAULT_SIZE_ID;
const CONTEXT_STEP_ID = 'context:step';
exports.CONTEXT_STEP_ID = CONTEXT_STEP_ID;
const CONTEXT_TIE_BREAKER_FIELDS_ID = 'context:tieBreakerFields';
exports.CONTEXT_TIE_BREAKER_FIELDS_ID = CONTEXT_TIE_BREAKER_FIELDS_ID;
const DEFAULT_COLUMNS_ID = 'defaultColumns';
exports.DEFAULT_COLUMNS_ID = DEFAULT_COLUMNS_ID;
const DISCOVER_ENABLE_SQL_ID = 'discover:enableSql';
exports.DISCOVER_ENABLE_SQL_ID = DISCOVER_ENABLE_SQL_ID;
const DISCOVER_MAX_DOC_FIELDS_DISPLAYED_ID = 'discover:maxDocFieldsDisplayed';
exports.DISCOVER_MAX_DOC_FIELDS_DISPLAYED_ID = DISCOVER_MAX_DOC_FIELDS_DISPLAYED_ID;
const DISCOVER_MODIFY_COLUMNS_ON_SWITCH_ID = 'discover:modifyColumnsOnSwitch';
exports.DISCOVER_MODIFY_COLUMNS_ON_SWITCH_ID = DISCOVER_MODIFY_COLUMNS_ON_SWITCH_ID;
const DISCOVER_ROW_HEIGHT_OPTION_ID = 'discover:rowHeightOption';
exports.DISCOVER_ROW_HEIGHT_OPTION_ID = DISCOVER_ROW_HEIGHT_OPTION_ID;
const DISCOVER_SAMPLE_ROWS_PER_PAGE_ID = 'discover:sampleRowsPerPage';
exports.DISCOVER_SAMPLE_ROWS_PER_PAGE_ID = DISCOVER_SAMPLE_ROWS_PER_PAGE_ID;
const DISCOVER_SAMPLE_SIZE_ID = 'discover:sampleSize';
exports.DISCOVER_SAMPLE_SIZE_ID = DISCOVER_SAMPLE_SIZE_ID;
const DISCOVER_SEARCH_FIELDS_FROM_SOURCE_ID = 'discover:searchFieldsFromSource';
exports.DISCOVER_SEARCH_FIELDS_FROM_SOURCE_ID = DISCOVER_SEARCH_FIELDS_FROM_SOURCE_ID;
const DISCOVER_SEARCH_ON_PAGE_LOAD_ID = 'discover:searchOnPageLoad';
exports.DISCOVER_SEARCH_ON_PAGE_LOAD_ID = DISCOVER_SEARCH_ON_PAGE_LOAD_ID;
const DISCOVER_SHOW_FIELD_STATISTICS_ID = 'discover:showFieldStatistics';
exports.DISCOVER_SHOW_FIELD_STATISTICS_ID = DISCOVER_SHOW_FIELD_STATISTICS_ID;
const DISCOVER_SHOW_LEGACY_FIELD_TOP_VALUES_ID = 'discover:showLegacyFieldTopValues';
exports.DISCOVER_SHOW_LEGACY_FIELD_TOP_VALUES_ID = DISCOVER_SHOW_LEGACY_FIELD_TOP_VALUES_ID;
const DISCOVER_SHOW_MULTI_FIELDS_ID = 'discover:showMultiFields';
exports.DISCOVER_SHOW_MULTI_FIELDS_ID = DISCOVER_SHOW_MULTI_FIELDS_ID;
const DISCOVER_SORT_DEFAULT_ORDER_ID = 'discover:sort:defaultOrder';
exports.DISCOVER_SORT_DEFAULT_ORDER_ID = DISCOVER_SORT_DEFAULT_ORDER_ID;
const DOC_TABLE_HIDE_TIME_COLUMNS_ID = 'doc_table:hideTimeColumn';
exports.DOC_TABLE_HIDE_TIME_COLUMNS_ID = DOC_TABLE_HIDE_TIME_COLUMNS_ID;
const DOC_TABLE_HIGHLIGHT_ID = 'doc_table:highlight';
exports.DOC_TABLE_HIGHLIGHT_ID = DOC_TABLE_HIGHLIGHT_ID;
const DOC_TABLE_LEGACY_ID = 'doc_table:legacy';
exports.DOC_TABLE_LEGACY_ID = DOC_TABLE_LEGACY_ID;
const TRUNCATE_MAX_HEIGHT_ID = 'truncate:maxHeight';

// Machine learning settings
exports.TRUNCATE_MAX_HEIGHT_ID = TRUNCATE_MAX_HEIGHT_ID;
const ML_ANOMALY_DETECTION_RESULTS_ENABLE_TIME_DEFAULTS_ID = 'ml:anomalyDetection:results:enableTimeDefaults';
exports.ML_ANOMALY_DETECTION_RESULTS_ENABLE_TIME_DEFAULTS_ID = ML_ANOMALY_DETECTION_RESULTS_ENABLE_TIME_DEFAULTS_ID;
const ML_ANOMALY_DETECTION_RESULTS_TIME_DEFAULTS_ID = 'ml:anomalyDetection:results:timeDefaults';

// Notifications settings
exports.ML_ANOMALY_DETECTION_RESULTS_TIME_DEFAULTS_ID = ML_ANOMALY_DETECTION_RESULTS_TIME_DEFAULTS_ID;
const NOTIFICATIONS_BANNER_ID = 'notifications:banner';
exports.NOTIFICATIONS_BANNER_ID = NOTIFICATIONS_BANNER_ID;
const NOTIFICATIONS_LIFETIME_BANNER_ID = 'notifications:lifetime:banner';
exports.NOTIFICATIONS_LIFETIME_BANNER_ID = NOTIFICATIONS_LIFETIME_BANNER_ID;
const NOTIFICATIONS_LIFETIME_ERROR_ID = 'notifications:lifetime:error';
exports.NOTIFICATIONS_LIFETIME_ERROR_ID = NOTIFICATIONS_LIFETIME_ERROR_ID;
const NOTIFICATIONS_LIFETIME_INFO_ID = 'notifications:lifetime:info';
exports.NOTIFICATIONS_LIFETIME_INFO_ID = NOTIFICATIONS_LIFETIME_INFO_ID;
const NOTIFICATIONS_LIFETIME_WARNING_ID = 'notifications:lifetime:warning';

// Observability settings
exports.NOTIFICATIONS_LIFETIME_WARNING_ID = NOTIFICATIONS_LIFETIME_WARNING_ID;
const OBSERVABILITY_APM_AWS_LAMBDA_PRICE_FACTOR_ID = 'observability:apmAWSLambdaPriceFactor';
exports.OBSERVABILITY_APM_AWS_LAMBDA_PRICE_FACTOR_ID = OBSERVABILITY_APM_AWS_LAMBDA_PRICE_FACTOR_ID;
const OBSERVABILITY_APM_AWS_LAMBDA_REQUEST_COST_PER_MILLION_ID = 'observability:apmAWSLambdaRequestCostPerMillion';
exports.OBSERVABILITY_APM_AWS_LAMBDA_REQUEST_COST_PER_MILLION_ID = OBSERVABILITY_APM_AWS_LAMBDA_REQUEST_COST_PER_MILLION_ID;
const OBSERVABILITY_APM_AGENT_EXPLORER_VIEW_ID = 'observability:apmAgentExplorerView';
exports.OBSERVABILITY_APM_AGENT_EXPLORER_VIEW_ID = OBSERVABILITY_APM_AGENT_EXPLORER_VIEW_ID;
const OBSERVABILITY_APM_DEFAULT_SERVICE_ENVIRONMENT_ID = 'observability:apmDefaultServiceEnvironment';
exports.OBSERVABILITY_APM_DEFAULT_SERVICE_ENVIRONMENT_ID = OBSERVABILITY_APM_DEFAULT_SERVICE_ENVIRONMENT_ID;
const OBSERVABILITY_APM_ENABLE_CRITICAL_PATH_ID = 'observability:apmEnableCriticalPath';
exports.OBSERVABILITY_APM_ENABLE_CRITICAL_PATH_ID = OBSERVABILITY_APM_ENABLE_CRITICAL_PATH_ID;
const OBSERVABILITY_APM_LABS_BUTTON_ID = 'observability:apmLabsButton';
exports.OBSERVABILITY_APM_LABS_BUTTON_ID = OBSERVABILITY_APM_LABS_BUTTON_ID;
const OBSERVABILITY_APM_PROGRESSIVE_LOADING_ID = 'observability:apmProgressiveLoading';
exports.OBSERVABILITY_APM_PROGRESSIVE_LOADING_ID = OBSERVABILITY_APM_PROGRESSIVE_LOADING_ID;
const OBSERVABILITY_APM_SERVICE_GROUP_MAX_NUMBER_OF_SERVCIE_ID = 'observability:apmServiceGroupMaxNumberOfServices';
exports.OBSERVABILITY_APM_SERVICE_GROUP_MAX_NUMBER_OF_SERVCIE_ID = OBSERVABILITY_APM_SERVICE_GROUP_MAX_NUMBER_OF_SERVCIE_ID;
const OBSERVABILITY_APM_SERVICE_INVENTORY_OPTIMIZED_SORTING_ID = 'observability:apmServiceInventoryOptimizedSorting';
exports.OBSERVABILITY_APM_SERVICE_INVENTORY_OPTIMIZED_SORTING_ID = OBSERVABILITY_APM_SERVICE_INVENTORY_OPTIMIZED_SORTING_ID;
const OBSERVABILITY_APM_TRACE_EXPLORER_TAB_ID = 'observability:apmTraceExplorerTab';
exports.OBSERVABILITY_APM_TRACE_EXPLORER_TAB_ID = OBSERVABILITY_APM_TRACE_EXPLORER_TAB_ID;
const OBSERVABILITY_ENABLE_AWS_LAMBDA_METRICS_ID = 'observability:enableAwsLambdaMetrics';
exports.OBSERVABILITY_ENABLE_AWS_LAMBDA_METRICS_ID = OBSERVABILITY_ENABLE_AWS_LAMBDA_METRICS_ID;
const OBSERVABILITY_ENABLE_COMPARISON_BY_DEFAULT_ID = 'observability:enableComparisonByDefault';
exports.OBSERVABILITY_ENABLE_COMPARISON_BY_DEFAULT_ID = OBSERVABILITY_ENABLE_COMPARISON_BY_DEFAULT_ID;
const OBSERVABILITY_ENABLE_INFRASTRUCTURE_HOSTS_VIEW_ID = 'observability:enableInfrastructureHostsView';
exports.OBSERVABILITY_ENABLE_INFRASTRUCTURE_HOSTS_VIEW_ID = OBSERVABILITY_ENABLE_INFRASTRUCTURE_HOSTS_VIEW_ID;
const OBSERVABILITY_ENABLE_INSPECT_ES_QUERIES_ID = 'observability:enableInspectEsQueries';
exports.OBSERVABILITY_ENABLE_INSPECT_ES_QUERIES_ID = OBSERVABILITY_ENABLE_INSPECT_ES_QUERIES_ID;
const OBSERVABILITY_MAX_SUGGESTIONS_ID = 'observability:maxSuggestions';
exports.OBSERVABILITY_MAX_SUGGESTIONS_ID = OBSERVABILITY_MAX_SUGGESTIONS_ID;
const OBSERVABILITY_PROFILING_ELASTICSEARCH_PLUGIN_ID = 'observability:profilingElasticsearchPlugin';
exports.OBSERVABILITY_PROFILING_ELASTICSEARCH_PLUGIN_ID = OBSERVABILITY_PROFILING_ELASTICSEARCH_PLUGIN_ID;
const OBSERVABILITY_APM_ENABLE_SERVICE_METRICS_ID = 'observability:apmEnableServiceMetrics';
exports.OBSERVABILITY_APM_ENABLE_SERVICE_METRICS_ID = OBSERVABILITY_APM_ENABLE_SERVICE_METRICS_ID;
const OBSERVABILITY_APM_ENABLE_CONTINUOUS_ROLLUPS_ID = 'observability:apmEnableContinuousRollups';
exports.OBSERVABILITY_APM_ENABLE_CONTINUOUS_ROLLUPS_ID = OBSERVABILITY_APM_ENABLE_CONTINUOUS_ROLLUPS_ID;
const OBSERVABILITY_APM_ENABLE_PROFILING_INTEGRATION_ID = 'observability:apmEnableProfilingIntegration';

// Reporting settings
exports.OBSERVABILITY_APM_ENABLE_PROFILING_INTEGRATION_ID = OBSERVABILITY_APM_ENABLE_PROFILING_INTEGRATION_ID;
const XPACK_REPORTING_CUSTOM_PDF_LOGO_ID = 'xpackReporting:customPdfLogo';

// Rollups settings
exports.XPACK_REPORTING_CUSTOM_PDF_LOGO_ID = XPACK_REPORTING_CUSTOM_PDF_LOGO_ID;
const ROLLUPS_ENABLE_INDEX_PATTERNS_ID = 'rollups.enableIndexPatterns';

// Search settings
exports.ROLLUPS_ENABLE_INDEX_PATTERNS_ID = ROLLUPS_ENABLE_INDEX_PATTERNS_ID;
const COURIER_CUSTOM_REQUEST_PREFERENCE_ID = 'courier:customRequestPreference';
exports.COURIER_CUSTOM_REQUEST_PREFERENCE_ID = COURIER_CUSTOM_REQUEST_PREFERENCE_ID;
const COURIER_IGNORE_FILTER_IF_FIELD_NOT_IN_INDEX_ID = 'courier:ignoreFilterIfFieldNotInIndex';
exports.COURIER_IGNORE_FILTER_IF_FIELD_NOT_IN_INDEX_ID = COURIER_IGNORE_FILTER_IF_FIELD_NOT_IN_INDEX_ID;
const COURIER_MAX_CONCURRENT_SHARD_REQUEST_ID = 'courier:maxConcurrentShardRequests';
exports.COURIER_MAX_CONCURRENT_SHARD_REQUEST_ID = COURIER_MAX_CONCURRENT_SHARD_REQUEST_ID;
const COURIER_SET_REQUEST_PREFERENCE_ID = 'courier:setRequestPreference';
exports.COURIER_SET_REQUEST_PREFERENCE_ID = COURIER_SET_REQUEST_PREFERENCE_ID;
const SEARCH_INCLUDE_FROZEN_ID = 'search:includeFrozen';
exports.SEARCH_INCLUDE_FROZEN_ID = SEARCH_INCLUDE_FROZEN_ID;
const SEARCH_TIMEOUT_ID = 'search:timeout';

// Security solution settings
exports.SEARCH_TIMEOUT_ID = SEARCH_TIMEOUT_ID;
const SECURITY_SOLUTION_REFRESH_INTERVAL_DEFAULTS_ID = 'securitySolution:refreshIntervalDefaults';
exports.SECURITY_SOLUTION_REFRESH_INTERVAL_DEFAULTS_ID = SECURITY_SOLUTION_REFRESH_INTERVAL_DEFAULTS_ID;
const SECURITY_SOLUTION_TIME_DEFAULTS_ID = 'securitySolution:timeDefaults';
exports.SECURITY_SOLUTION_TIME_DEFAULTS_ID = SECURITY_SOLUTION_TIME_DEFAULTS_ID;
const SECURITY_SOLUTION_DEFAULT_INDEX_ID = 'securitySolution:defaultIndex';
exports.SECURITY_SOLUTION_DEFAULT_INDEX_ID = SECURITY_SOLUTION_DEFAULT_INDEX_ID;
const SECURITY_SOLUTION_DEFAULT_THREAT_INDEX_ID = 'securitySolution:defaultThreatIndex';
exports.SECURITY_SOLUTION_DEFAULT_THREAT_INDEX_ID = SECURITY_SOLUTION_DEFAULT_THREAT_INDEX_ID;
const SECURITY_SOLUTION_DEFAULT_ANOMALY_SCORE_ID = 'securitySolution:defaultAnomalyScore';
exports.SECURITY_SOLUTION_DEFAULT_ANOMALY_SCORE_ID = SECURITY_SOLUTION_DEFAULT_ANOMALY_SCORE_ID;
const SECURITY_SOLUTION_ENABLE_GROUPED_NAV_ID = 'securitySolution:enableGroupedNav';
exports.SECURITY_SOLUTION_ENABLE_GROUPED_NAV_ID = SECURITY_SOLUTION_ENABLE_GROUPED_NAV_ID;
const SECURITY_SOLUTION_RULES_TABLE_REFRESH_ID = 'securitySolution:rulesTableRefresh';
exports.SECURITY_SOLUTION_RULES_TABLE_REFRESH_ID = SECURITY_SOLUTION_RULES_TABLE_REFRESH_ID;
const SECURITY_SOLUTION_ENABLE_NEWS_FEED_ID = 'securitySolution:enableNewsFeed';
exports.SECURITY_SOLUTION_ENABLE_NEWS_FEED_ID = SECURITY_SOLUTION_ENABLE_NEWS_FEED_ID;
const SECURITY_SOLUTION_NEWS_FEED_URL_ID = 'securitySolution:newsFeedUrl';
exports.SECURITY_SOLUTION_NEWS_FEED_URL_ID = SECURITY_SOLUTION_NEWS_FEED_URL_ID;
const SECURITY_SOLUTION_IP_REPUTATION_LINKS_ID = 'securitySolution:ipReputationLinks';
exports.SECURITY_SOLUTION_IP_REPUTATION_LINKS_ID = SECURITY_SOLUTION_IP_REPUTATION_LINKS_ID;
const SECURITY_SOLUTION_ENABLE_CCS_WARNING_ID = 'securitySolution:enableCcsWarning';
exports.SECURITY_SOLUTION_ENABLE_CCS_WARNING_ID = SECURITY_SOLUTION_ENABLE_CCS_WARNING_ID;
const SECURITY_SOLUTION_SHOW_RELATED_INTEGRATIONS_ID = 'securitySolution:showRelatedIntegrations';
exports.SECURITY_SOLUTION_SHOW_RELATED_INTEGRATIONS_ID = SECURITY_SOLUTION_SHOW_RELATED_INTEGRATIONS_ID;
const SECURITY_SOLUTION_DEFAULT_ALERT_TAGS_KEY = 'securitySolution:alertTags';
/** This Kibana Advanced Setting allows users to enable/disable the Expandable Flyout */
exports.SECURITY_SOLUTION_DEFAULT_ALERT_TAGS_KEY = SECURITY_SOLUTION_DEFAULT_ALERT_TAGS_KEY;
const SECURITY_SOLUTION_ENABLE_EXPANDABLE_FLYOUT_SETTING = 'securitySolution:enableExpandableFlyout';
/** This Kibana Advanced Setting allows users to enable/disable querying cold and frozen data tiers in analyzer */
exports.SECURITY_SOLUTION_ENABLE_EXPANDABLE_FLYOUT_SETTING = SECURITY_SOLUTION_ENABLE_EXPANDABLE_FLYOUT_SETTING;
const SECURITY_SOLUTION_EXCLUDE_COLD_AND_FROZEN_TIERS_IN_ANALYZER = 'securitySolution:excludeColdAndFrozenTiersInAnalyzer';

// Timelion settings
exports.SECURITY_SOLUTION_EXCLUDE_COLD_AND_FROZEN_TIERS_IN_ANALYZER = SECURITY_SOLUTION_EXCLUDE_COLD_AND_FROZEN_TIERS_IN_ANALYZER;
const TIMELION_ES_DEFAULT_INDEX_ID = 'timelion:es.default_index';
exports.TIMELION_ES_DEFAULT_INDEX_ID = TIMELION_ES_DEFAULT_INDEX_ID;
const TIMELION_ES_TIME_FIELD_ID = 'timelion:es.timefield';
exports.TIMELION_ES_TIME_FIELD_ID = TIMELION_ES_TIME_FIELD_ID;
const TIMELION_MAX_BUCKETS_ID = 'timelion:max_buckets';
exports.TIMELION_MAX_BUCKETS_ID = TIMELION_MAX_BUCKETS_ID;
const TIMELION_MIN_INTERVAL_ID = 'timelion:min_interval';
exports.TIMELION_MIN_INTERVAL_ID = TIMELION_MIN_INTERVAL_ID;
const TIMELION_TARGET_BUCKETS_ID = 'timelion:target_buckets';

// Visualization settings
exports.TIMELION_TARGET_BUCKETS_ID = TIMELION_TARGET_BUCKETS_ID;
const VISUALIZATION_COLOR_MAPPING_ID = 'visualization:colorMapping';
exports.VISUALIZATION_COLOR_MAPPING_ID = VISUALIZATION_COLOR_MAPPING_ID;
const VISUALIZATION_HEATMAP_MAX_BUCKETS_ID = 'visualization:heatmap:maxBuckets';
exports.VISUALIZATION_HEATMAP_MAX_BUCKETS_ID = VISUALIZATION_HEATMAP_MAX_BUCKETS_ID;
const VISUALIZATION_USE_LEGACY_TIME_AXIS_ID = 'visualization:useLegacyTimeAxis';
exports.VISUALIZATION_USE_LEGACY_TIME_AXIS_ID = VISUALIZATION_USE_LEGACY_TIME_AXIS_ID;
const VISUALIZATION_LEGACY_GAUGE_CHARTS_LIBRARY_ID = 'visualization:visualize:legacyGaugeChartsLibrary';
exports.VISUALIZATION_LEGACY_GAUGE_CHARTS_LIBRARY_ID = VISUALIZATION_LEGACY_GAUGE_CHARTS_LIBRARY_ID;
const VISUALIZATION_LEGACY_HEATMAP_CHARTS_LIBRARY_ID = 'visualization:visualize:legacyHeatmapChartsLibrary';
exports.VISUALIZATION_LEGACY_HEATMAP_CHARTS_LIBRARY_ID = VISUALIZATION_LEGACY_HEATMAP_CHARTS_LIBRARY_ID;
const VISUALIZATION_ENABLE_LABS_ID = 'visualize:enableLabs';
exports.VISUALIZATION_ENABLE_LABS_ID = VISUALIZATION_ENABLE_LABS_ID;
