// @kbn/security-solution-plugin/public/assistant/header_link.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AssistantHeaderLink = void 0;
var _eui = require("@elastic/eui");
var _react = _interopRequireWildcard(require("react"));
var _i18n = require("@kbn/i18n");
var _assistant_context = require("@kbn/elastic-assistant/impl/assistant_context");
var _elasticAssistant = require("@kbn/elastic-assistant");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const isMac = navigator.platform.toLowerCase().indexOf('mac') >= 0;
const TOOLTIP_CONTENT = _i18n.i18n.translate('xpack.securitySolution.globalHeader.assistantHeaderLinkShortcutTooltip', {
  values: {
    keyboardShortcut: isMac ? '⌘ ;' : 'Ctrl ;'
  },
  defaultMessage: 'Keyboard shortcut {keyboardShortcut}'
});
const LINK_LABEL = _i18n.i18n.translate('xpack.securitySolution.globalHeader.assistantHeaderLink', {
  defaultMessage: 'AI Assistant'
});

/**
 * Elastic AI Assistant header link
 */
const AssistantHeaderLink = () => {
  const {
    showAssistantOverlay,
    assistantAvailability
  } = (0, _assistant_context.useAssistantContext)();
  const showOverlay = (0, _react.useCallback)(() => showAssistantOverlay({
    showOverlay: true
  }), [showAssistantOverlay]);
  if (!assistantAvailability.hasAssistantPrivilege) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_eui.EuiToolTip, {
    content: TOOLTIP_CONTENT
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiHeaderLink, {
    "data-test-subj": "assistantHeaderLink",
    color: "primary",
    onClick: showOverlay
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
    gutterSize: "s",
    alignItems: "center"
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
    grow: false
  }, /*#__PURE__*/_react.default.createElement(_elasticAssistant.AssistantAvatar, {
    size: "xs"
  })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
    grow: false
  }, LINK_LABEL))));
};
exports.AssistantHeaderLink = AssistantHeaderLink;
