// @kbn/security-solution-plugin/public/explore/network/components/port/helpers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.URL_PORT_FIELD_NAME = exports.SOURCE_PORT_FIELD_NAME = exports.SERVER_PORT_FIELD_NAME = exports.PORT_NAMES = exports.DESTINATION_PORT_FIELD_NAME = exports.CLIENT_PORT_FIELD_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
const CLIENT_PORT_FIELD_NAME = 'client.port';
exports.CLIENT_PORT_FIELD_NAME = CLIENT_PORT_FIELD_NAME;
const DESTINATION_PORT_FIELD_NAME = 'destination.port';
exports.DESTINATION_PORT_FIELD_NAME = DESTINATION_PORT_FIELD_NAME;
const SERVER_PORT_FIELD_NAME = 'server.port';
exports.SERVER_PORT_FIELD_NAME = SERVER_PORT_FIELD_NAME;
const SOURCE_PORT_FIELD_NAME = 'source.port';
exports.SOURCE_PORT_FIELD_NAME = SOURCE_PORT_FIELD_NAME;
const URL_PORT_FIELD_NAME = 'url.port';
exports.URL_PORT_FIELD_NAME = URL_PORT_FIELD_NAME;
const PORT_NAMES = [CLIENT_PORT_FIELD_NAME, DESTINATION_PORT_FIELD_NAME, SERVER_PORT_FIELD_NAME, SOURCE_PORT_FIELD_NAME, URL_PORT_FIELD_NAME];
exports.PORT_NAMES = PORT_NAMES;
