// @kbn/lists-plugin/server/routes/list_index/read_list_index_route.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.readListIndexRoute = void 0;
var _securitysolutionIoTsUtils = require("@kbn/securitysolution-io-ts-utils");
var _securitysolutionEsUtils = require("@kbn/securitysolution-es-utils");
var _securitysolutionListConstants = require("@kbn/securitysolution-list-constants");
var _api = require("../../../common/api");
var _utils = require("../utils");
var _2 = require("..");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const readListIndexRoute = router => {
  router.versioned.get({
    access: 'public',
    options: {
      tags: ['access:lists-read']
    },
    path: _securitysolutionListConstants.LIST_INDEX
  }).addVersion({
    validate: false,
    version: '2023-10-31'
  }, async (context, _, response) => {
    const siemResponse = (0, _utils.buildSiemResponse)(response);
    try {
      const lists = await (0, _2.getListClient)(context);
      const listDataStreamExists = await lists.getListDataStreamExists();
      const listItemDataStreamExists = await lists.getListItemDataStreamExists();
      if (listDataStreamExists && listItemDataStreamExists) {
        const [validated, errors] = (0, _securitysolutionIoTsUtils.validate)({
          list_index: listDataStreamExists,
          list_item_index: listItemDataStreamExists
        }, _api.readListIndexResponse);
        if (errors != null) {
          return siemResponse.error({
            body: errors,
            statusCode: 500
          });
        } else {
          return response.ok({
            body: validated !== null && validated !== void 0 ? validated : {}
          });
        }
      } else if (!listDataStreamExists && listItemDataStreamExists) {
        return siemResponse.error({
          body: `data stream ${lists.getListName()} does not exist`,
          statusCode: 404
        });
      } else if (!listItemDataStreamExists && listDataStreamExists) {
        return siemResponse.error({
          body: `data stream ${lists.getListItemName()} does not exist`,
          statusCode: 404
        });
      } else {
        return siemResponse.error({
          body: `data stream ${lists.getListName()} and data stream ${lists.getListItemName()} does not exist`,
          statusCode: 404
        });
      }
    } catch (err) {
      const error = (0, _securitysolutionEsUtils.transformError)(err);
      return siemResponse.error({
        body: error.message,
        statusCode: error.statusCode
      });
    }
  });
};
exports.readListIndexRoute = readListIndexRoute;
