// @kbn/cloud-security-posture-plugin/public/pages/benchmarks/test_subjects.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BENCHMARKS_TABLE_DATA_TEST_SUBJ = exports.BENCHMARKS_TABLE_COLUMNS = exports.BENCHMARKS_PAGE_HEADER = exports.ADD_INTEGRATION_TEST_SUBJ = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const BENCHMARKS_PAGE_HEADER = 'benchmarks-page-header';
exports.BENCHMARKS_PAGE_HEADER = BENCHMARKS_PAGE_HEADER;
const BENCHMARKS_TABLE_DATA_TEST_SUBJ = 'csp_benchmarks_table';
exports.BENCHMARKS_TABLE_DATA_TEST_SUBJ = BENCHMARKS_TABLE_DATA_TEST_SUBJ;
const ADD_INTEGRATION_TEST_SUBJ = 'csp_add_integration';
exports.ADD_INTEGRATION_TEST_SUBJ = ADD_INTEGRATION_TEST_SUBJ;
const BENCHMARKS_TABLE_COLUMNS = {
  INTEGRATION_NAME: 'benchmarks-table-column-integration-name',
  MONITORING: 'benchmarks-table-column-monitoring',
  RULES: 'benchmarks-table-column-rules',
  INTEGRATION: 'benchmarks-table-column-integration',
  AGENT_POLICY: 'benchmarks-table-column-agent-policy',
  NUMBER_OF_AGENTS: 'benchmarks-table-column-number-of-agents',
  CREATED_BY: 'benchmarks-table-column-created-by',
  CREATED_AT: 'benchmarks-table-column-created-at'
};
exports.BENCHMARKS_TABLE_COLUMNS = BENCHMARKS_TABLE_COLUMNS;
