// @kbn/ux-plugin/public/components/app/rum_dashboard/utils/test_helper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.render = exports.core = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@testing-library/react");
var _rxjs = require("rxjs");
var _history = require("history");
var _sharedUxRouter = require("@kbn/shared-ux-router");
var _common = require("@kbn/kibana-react-plugin/common");
var _public = require("@kbn/kibana-react-plugin/public");
var _url_params_context = require("../../../../context/url_params_context/url_params_context");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const core = {
  http: {
    basePath: {
      prepend: jest.fn()
    }
  },
  uiSettings: {
    get: key => true,
    get$: key => (0, _rxjs.of)(true)
  }
};
exports.core = core;
const render = (component, options) => {
  var _options$customHistor;
  const history = (_options$customHistor = options === null || options === void 0 ? void 0 : options.customHistory) !== null && _options$customHistor !== void 0 ? _options$customHistor : (0, _history.createMemoryHistory)();
  history.location.key = 'TestKeyForTesting';
  return (0, _react2.render)( /*#__PURE__*/_react.default.createElement(_sharedUxRouter.Router, {
    history: history
  }, /*#__PURE__*/_react.default.createElement(_public.KibanaContextProvider, {
    services: {
      ...core
    }
  }, /*#__PURE__*/_react.default.createElement(_url_params_context.UrlParamsProvider, null, /*#__PURE__*/_react.default.createElement(_common.EuiThemeProvider, null, component)))));
};
exports.render = render;
