// @kbn/stack-connectors-plugin/common/sentinelone/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SUB_ACTION = exports.SENTINELONE_TITLE = exports.SENTINELONE_CONNECTOR_ID = exports.API_MAX_RESULTS = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const SENTINELONE_TITLE = 'Sentinel One';
exports.SENTINELONE_TITLE = SENTINELONE_TITLE;
const SENTINELONE_CONNECTOR_ID = '.sentinelone';
exports.SENTINELONE_CONNECTOR_ID = SENTINELONE_CONNECTOR_ID;
const API_MAX_RESULTS = 1000;
exports.API_MAX_RESULTS = API_MAX_RESULTS;
let SUB_ACTION;
exports.SUB_ACTION = SUB_ACTION;
(function (SUB_ACTION) {
  SUB_ACTION["KILL_PROCESS"] = "killProcess";
  SUB_ACTION["EXECUTE_SCRIPT"] = "executeScript";
  SUB_ACTION["GET_AGENTS"] = "getAgents";
  SUB_ACTION["ISOLATE_HOST"] = "isolateHost";
  SUB_ACTION["RELEASE_HOST"] = "releaseHost";
  SUB_ACTION["GET_REMOTE_SCRIPTS"] = "getRemoteScripts";
  SUB_ACTION["GET_REMOTE_SCRIPT_STATUS"] = "getRemoteScriptStatus";
  SUB_ACTION["GET_REMOTE_SCRIPT_RESULTS"] = "getRemoteScriptResults";
})(SUB_ACTION || (exports.SUB_ACTION = SUB_ACTION = {}));
