// @kbn/security-solution-plugin/public/explore/network/components/source_destination/network.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Network = void 0;
var _eui = require("@elastic/eui");
var _fp = require("lodash/fp");
var _react = _interopRequireDefault(require("react"));
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _direction = require("../direction");
var _draggables = require("../../../../common/components/draggables");
var i18n = _interopRequireWildcard(require("./translations"));
var _field_names = require("./field_names");
var _formatted_bytes = require("../../../../common/components/formatted_bytes");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const EuiFlexItemMarginRight = (0, _styledComponents.default)(_eui.EuiFlexItem)`
  margin-right: 3px;
`;
EuiFlexItemMarginRight.displayName = 'EuiFlexItemMarginRight';
const Stats = (0, _styledComponents.default)(_eui.EuiText)`
  margin: 0 5px;
`;
Stats.displayName = 'Stats';

/**
 * Renders a row of draggable badges containing fields from the
 * `Network` category of fields
 */
const Network = /*#__PURE__*/_react.default.memo(({
  bytes,
  communityId,
  contextId,
  direction,
  eventId,
  isDraggable,
  packets,
  protocol,
  transport
}) => /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
  alignItems: "center",
  justifyContent: "center",
  gutterSize: "none"
}, direction != null ? (0, _fp.uniq)(direction).map(dir => /*#__PURE__*/_react.default.createElement(EuiFlexItemMarginRight, {
  grow: false,
  key: dir
}, /*#__PURE__*/_react.default.createElement(_direction.DirectionBadge, {
  contextId: contextId,
  direction: dir,
  eventId: eventId,
  isDraggable: isDraggable
}))) : null, protocol != null ? (0, _fp.uniq)(protocol).map(proto => /*#__PURE__*/_react.default.createElement(EuiFlexItemMarginRight, {
  grow: false,
  key: proto
}, /*#__PURE__*/_react.default.createElement(_draggables.DraggableBadge, {
  contextId: contextId,
  eventId: eventId,
  field: _field_names.NETWORK_PROTOCOL_FIELD_NAME,
  isDraggable: isDraggable,
  value: proto,
  isAggregatable: true,
  fieldType: "keyword"
}))) : null, bytes != null ? (0, _fp.uniq)(bytes).map(b => !isNaN(Number(b)) ? /*#__PURE__*/_react.default.createElement(EuiFlexItemMarginRight, {
  grow: false,
  key: b
}, /*#__PURE__*/_react.default.createElement(_draggables.DefaultDraggable, {
  field: _field_names.NETWORK_BYTES_FIELD_NAME,
  id: `network-default-draggable-${contextId}-${eventId}-${_field_names.NETWORK_BYTES_FIELD_NAME}-${b}`,
  isDraggable: isDraggable,
  value: b
}, /*#__PURE__*/_react.default.createElement(Stats, {
  size: "xs"
}, /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement(_formatted_bytes.PreferenceFormattedBytes, {
  value: b
}))))) : null) : null, packets != null ? (0, _fp.uniq)(packets).map(p => /*#__PURE__*/_react.default.createElement(EuiFlexItemMarginRight, {
  grow: false,
  key: p
}, /*#__PURE__*/_react.default.createElement(_draggables.DefaultDraggable, {
  field: _field_names.NETWORK_PACKETS_FIELD_NAME,
  id: `network-default-draggable-${contextId}-${eventId}-${_field_names.NETWORK_PACKETS_FIELD_NAME}-${p}`,
  isDraggable: isDraggable,
  value: p
}, /*#__PURE__*/_react.default.createElement(Stats, {
  size: "xs"
}, /*#__PURE__*/_react.default.createElement("span", null, `${p} ${i18n.PACKETS}`))))) : null, transport != null ? (0, _fp.uniq)(transport).map(trans => /*#__PURE__*/_react.default.createElement(EuiFlexItemMarginRight, {
  grow: false,
  key: trans
}, /*#__PURE__*/_react.default.createElement(_draggables.DraggableBadge, {
  contextId: contextId,
  "data-test-subj": "network-transport",
  eventId: eventId,
  field: _field_names.NETWORK_TRANSPORT_FIELD_NAME,
  isDraggable: isDraggable,
  value: trans,
  isAggregatable: true,
  fieldType: "keyword"
}))) : null, communityId != null ? (0, _fp.uniq)(communityId).map(trans => /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
  grow: false,
  key: trans
}, /*#__PURE__*/_react.default.createElement(_draggables.DraggableBadge, {
  contextId: contextId,
  eventId: eventId,
  field: _field_names.NETWORK_COMMUNITY_ID_FIELD_NAME,
  isDraggable: isDraggable,
  value: trans,
  isAggregatable: true,
  fieldType: "keyword"
}))) : null));
exports.Network = Network;
Network.displayName = 'Network';
