// @kbn/maps-plugin/public/inspector/map_adapter/map_view_component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _map_details = require("./map_details");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

class MapViewComponent extends _react.Component {
  constructor(...args) {
    super(...args);
    (0, _defineProperty2.default)(this, "state", this.props.adapters.map.getMapState());
    (0, _defineProperty2.default)(this, "_onMapChange", () => {
      this.setState(this.props.adapters.map.getMapState());
    });
  }
  componentDidMount() {
    this.props.adapters.map.on('change', this._onMapChange);
  }
  componentWillUnmount() {
    this.props.adapters.map.removeListener('change', this._onMapChange);
  }
  render() {
    return /*#__PURE__*/_react.default.createElement(_map_details.MapDetails, {
      centerLon: this.state.stats.center[0],
      centerLat: this.state.stats.center[1],
      zoom: this.state.stats.zoom,
      style: this.state.style
    });
  }
}

// default export required for React.Lazy
// eslint-disable-next-line import/no-default-export
var _default = MapViewComponent;
exports.default = _default;
module.exports = exports.default;
