// @kbn/watcher-plugin/public/application/lib/breadcrumbs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.statusBreadcrumb = exports.listBreadcrumb = exports.editBreadcrumb = exports.createBreadcrumb = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const listBreadcrumb = {
  text: _i18n.i18n.translate('xpack.watcher.breadcrumb.listLabel', {
    defaultMessage: 'Watcher'
  }),
  href: '/watches'
};
exports.listBreadcrumb = listBreadcrumb;
const createBreadcrumb = {
  text: _i18n.i18n.translate('xpack.watcher.breadcrumb.createLabel', {
    defaultMessage: 'Create'
  })
};
exports.createBreadcrumb = createBreadcrumb;
const editBreadcrumb = {
  text: _i18n.i18n.translate('xpack.watcher.breadcrumb.editLabel', {
    defaultMessage: 'Edit'
  })
};
exports.editBreadcrumb = editBreadcrumb;
const statusBreadcrumb = {
  text: _i18n.i18n.translate('xpack.watcher.breadcrumb.statusLabel', {
    defaultMessage: 'Status'
  })
};
exports.statusBreadcrumb = statusBreadcrumb;
