// @kbn/security-solution-plugin/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension/components/endpoint_event_collection_preset.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EndpointEventCollectionPreset = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _constants = require("../constants");
var _help_text_with_padding = require("./help_text_with_padding");
var _translations = require("../translations");
var _use_get_protections_unavailable_component = require("../../../policy_settings_form/hooks/use_get_protections_unavailable_component");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const NOOP = () => {};
/**
 * Display ONLY the event collection option on the screen along with the upselling message
 */
const EndpointEventCollectionPreset = /*#__PURE__*/(0, _react.memo)(({
  onChange,
  newPolicy
}) => {
  const UpsellToIncludePolicyProtections = (0, _use_get_protections_unavailable_component.useGetProtectionsUnavailableComponent)();
  const preset = 'DataCollection';
  const policyInputs = (0, _react.useMemo)(() => {
    return [{
      enabled: true,
      streams: [],
      type: _constants.ENDPOINT_INTEGRATION_CONFIG_KEY,
      config: {
        _config: {
          value: {
            type: 'endpoint',
            endpointConfig: {
              preset
            }
          }
        }
      }
    }];
  }, []);
  (0, _react.useEffect)(() => {
    var _inputs$, _inputs$$config;
    const inputs = newPolicy.inputs;
    if (inputs.length === 0) {
      onChange({
        isValid: false,
        updatedPolicy: {
          ...newPolicy,
          name: '',
          inputs: policyInputs
        }
      });
      return;
    }
    if (((_inputs$ = inputs[0]) === null || _inputs$ === void 0 ? void 0 : (_inputs$$config = _inputs$.config) === null || _inputs$$config === void 0 ? void 0 : _inputs$$config._config.value.endpointConfig.preset) !== preset) {
      onChange({
        isValid: true,
        updatedPolicy: {
          ...newPolicy,
          inputs: policyInputs
        }
      });
    }
  }, [newPolicy, onChange, policyInputs]);
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test-subj": "endpointDataCollectionOnlyPreset"
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
    size: "m"
  }), /*#__PURE__*/_react.default.createElement(_eui.EuiFormRow, {
    fullWidth: true,
    helpText: /*#__PURE__*/_react.default.createElement(_help_text_with_padding.HelpTextWithPadding, null, _translations.DATA_COLLECTION_HELP_TEXT)
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
    gutterSize: "s"
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
    grow: false
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiRadio, {
    id: "endpoint_data_collection_only_preset",
    onChange: NOOP,
    disabled: true,
    checked: true
  })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, null, _translations.DATA_COLLECTION))), UpsellToIncludePolicyProtections && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
    size: "xl"
  }), /*#__PURE__*/_react.default.createElement(UpsellToIncludePolicyProtections, null)));
});
exports.EndpointEventCollectionPreset = EndpointEventCollectionPreset;
EndpointEventCollectionPreset.displayName = 'EndpointEventCollectionPreset';
