// @kbn/monaco/src/common/error_listener.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ANTLREErrorListener = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

class ANTLREErrorListener {
  constructor() {
    (0, _defineProperty2.default)(this, "errors", []);
  }
  syntaxError(recognizer, offendingSymbol, line, column, message) {
    let endColumn = column + 1;
    if (offendingSymbol !== null && offendingSymbol !== void 0 && offendingSymbol._text) {
      endColumn = column + offendingSymbol._text.length;
    }
    this.errors.push({
      startLineNumber: line,
      endLineNumber: line,
      startColumn: column,
      endColumn,
      message
    });
  }
  getErrors() {
    return this.errors;
  }
}
exports.ANTLREErrorListener = ANTLREErrorListener;
