// @kbn/infra-plugin/public/pages/logs/stream/components/stream_page_template.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogStreamPageTemplate = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _public = require("@kbn/core/public");
var _i18n = require("@kbn/i18n");
var _react = _interopRequireDefault(require("react"));
var _page_template = require("../../../../page_template.styles");
var _page_template2 = require("../../shared/page_template");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const LogStreamPageTemplate = /*#__PURE__*/_react.default.memo(props => /*#__PURE__*/_react.default.createElement("div", {
  className: _public.APP_WRAPPER_CLASS
}, /*#__PURE__*/_react.default.createElement(_page_template2.LogsPageTemplate, (0, _extends2.default)({
  pageHeader: {
    pageTitle: streamTitle
  },
  pageSectionProps: {
    contentProps: {
      css: _page_template.fullHeightContentStyles
    }
  }
}, props))));
exports.LogStreamPageTemplate = LogStreamPageTemplate;
const streamTitle = _i18n.i18n.translate('xpack.infra.logs.streamPageTitle', {
  defaultMessage: 'Stream'
});
