// @kbn/ml-plugin/public/application/timeseriesexplorer/timeseriesexplorer_constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TIME_FIELD_NAME = exports.MAX_SCHEDULED_EVENTS = exports.CHARTS_POINT_TARGET = exports.APP_STATE_ACTION = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/*
 * Contains values for ML time series explorer.
 */

const APP_STATE_ACTION = {
  CLEAR: 'CLEAR',
  SET_DETECTOR_INDEX: 'SET_DETECTOR_INDEX',
  SET_ENTITIES: 'SET_ENTITIES',
  SET_FORECAST_ID: 'SET_FORECAST_ID',
  SET_ZOOM: 'SET_ZOOM',
  UNSET_ZOOM: 'UNSET_ZOOM',
  SET_FUNCTION_DESCRIPTION: 'SET_FUNCTION_DESCRIPTION'
};
exports.APP_STATE_ACTION = APP_STATE_ACTION;
const CHARTS_POINT_TARGET = 500;

// Max number of scheduled events displayed per bucket.
exports.CHARTS_POINT_TARGET = CHARTS_POINT_TARGET;
const MAX_SCHEDULED_EVENTS = 10;
exports.MAX_SCHEDULED_EVENTS = MAX_SCHEDULED_EVENTS;
const TIME_FIELD_NAME = 'timestamp';
exports.TIME_FIELD_NAME = TIME_FIELD_NAME;
