// @kbn/data-view-management-plugin/public/components/edit_index_pattern/relationships_table/i18n.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typeFieldName = exports.typeFieldDescription = exports.titleFieldName = exports.titleFieldDescription = exports.filterTitle = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const typeFieldName = _i18n.i18n.translate('indexPatternManagement.objectsTable.relationships.columnTypeName', {
  defaultMessage: 'Type'
});
exports.typeFieldName = typeFieldName;
const typeFieldDescription = _i18n.i18n.translate('indexPatternManagement.objectsTable.relationships.columnTypeDescription', {
  defaultMessage: 'Type of the saved object'
});
exports.typeFieldDescription = typeFieldDescription;
const titleFieldName = _i18n.i18n.translate('indexPatternManagement.objectsTable.relationships.columnTitleName', {
  defaultMessage: 'Title'
});
exports.titleFieldName = titleFieldName;
const titleFieldDescription = _i18n.i18n.translate('indexPatternManagement.objectsTable.relationships.columnTitleDescription', {
  defaultMessage: 'Title of the saved object'
});
exports.titleFieldDescription = titleFieldDescription;
const filterTitle = _i18n.i18n.translate('indexPatternManagement.objectsTable.relationships.search.filters.type.name', {
  defaultMessage: 'Type'
});
exports.filterTitle = filterTitle;
