// @kbn/enterprise-search-plugin/public/applications/workplace_search/views/content_sources/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SYNC_MANAGEMENT_THUMBNAILS_LABEL = exports.SYNC_MANAGEMENT_THUMBNAILS_GLOBAL_CONFIG_LABEL = exports.SYNC_MANAGEMENT_SYNCHRONIZE_LABEL = exports.SYNC_MANAGEMENT_CONTENT_EXTRACTION_LABEL = exports.SYNC_DIAGNOSTICS_TITLE = exports.SYNC_DIAGNOSTICS_DESCRIPTION = exports.SYNC_DIAGNOSTICS_BUTTON = exports.SYNCHRONIZATION_DISABLED_TITLE = exports.SYNCHRONIZATION_DISABLED_DESCRIPTION = exports.STATUS_TEXT = exports.STATUS_LABEL = exports.STATUS_HEADING = exports.STATUS_HEADER = exports.SOURCE_SYNC_FREQUENCY_TITLE = exports.SOURCE_SYNC_FREQUENCY_LINK_LABEL = exports.SOURCE_SYNC_CONFIRM_TITLE = exports.SOURCE_SYNC_CONFIRM_MESSAGE = exports.SOURCE_SYNCHRONIZATION_TOGGLE_LABEL = exports.SOURCE_SYNCHRONIZATION_TOGGLE_DESCRIPTION = exports.SOURCE_SYNCHRONIZATION_TITLE = exports.SOURCE_SYNCHRONIZATION_FREQUENCY_TITLE = exports.SOURCE_SYNCHRONIZATION_DESCRIPTION = exports.SOURCE_SYNCHRONIZATION_BUTTON_LABEL = exports.SOURCE_SETTINGS_TITLE = exports.SOURCE_SETTINGS_HEADING = exports.SOURCE_SETTINGS_DESCRIPTION = exports.SOURCE_REMOVE_TITLE = exports.SOURCE_REMOVE_DESCRIPTION = exports.SOURCE_OVERVIEW_TITLE = exports.SOURCE_NAME_LABEL = exports.SOURCE_FREQUENCY_DESCRIPTION = exports.SOURCE_DISABLED_CALLOUT_TITLE = exports.SOURCE_DISABLED_CALLOUT_DESCRIPTION = exports.SOURCE_DISABLED_CALLOUT_BUTTON = exports.SOURCE_CONTENT_TITLE = exports.SOURCE_CONFIG_TITLE = exports.SOURCE_CONFIG_LINK = exports.SOURCE_BLOCKED_TIME_WINDOWS_LINK_LABEL = exports.SOURCE_ASSETS_AND_OBJECTS_OBJECTS_LABEL = exports.SOURCE_ASSETS_AND_OBJECTS_LEARN_MORE_LINK = exports.SOURCE_ASSETS_AND_OBJECTS_DESCRIPTION = exports.SOURCE_ASSETS_AND_OBJECTS_ASSETS_LABEL = exports.SOURCES_NO_CONTENT_TITLE = exports.SEARCH_CONTENT_PLACEHOLDER = exports.RESET_BUTTON = exports.REMOTE_SOURCE_LABEL = exports.READY_TEXT = exports.PRIVATE_ORG_SOURCES_TITLE = exports.PRIVATE_LINK_TITLE = exports.PRIVATE_HEADER_TITLE = exports.PRIVATE_HEADER_DESCRIPTION = exports.PRIVATE_EMPTY_TITLE = exports.PERSONAL_DASHBOARD_SOURCE_ERROR = exports.PERSONAL_DASHBOARD_READ_ONLY_MODE_WARNING = exports.PERMISSIONS_SYNC_LABEL = exports.PERMISSIONS_SYNC_DESCRIPTION = exports.ORG_SOURCES_LINK = exports.ORG_SOURCES_HEADER_TITLE = exports.ORG_SOURCES_HEADER_DESCRIPTION = exports.ORG_SOURCES_EMPTY_TITLE = exports.ORG_SOURCES_EMPTY_DESCRIPTION = exports.NO_CONTENT_MESSAGE = exports.NEXT_SYNC_RUNNING_MESSAGE = exports.LICENSE_CALLOUT_TITLE = exports.LICENSE_CALLOUT_DESCRIPTION = exports.LEARN_MORE_LINK = exports.LEARN_CUSTOM_FEATURES_BUTTON = exports.LAST_UPDATED_HEADING = exports.ITEMS_HEADER = exports.INCREMENTAL_SYNC_LABEL = exports.INCREMENTAL_SYNC_DESCRIPTION = exports.ID_LABEL = exports.GROUP_ACCESS_TITLE = exports.GO_BUTTON = exports.FULL_SYNC_LABEL = exports.FULL_SYNC_DESCRIPTION = exports.FILTER_CONTENT_PLACEHOLDER = exports.EXTERNAL_IDENTITIES_LINK = exports.EVENT_HEADER = exports.EMPTY_ACTIVITY_TITLE = exports.DOWNLOAD_DIAGNOSTIC_BUTTON = exports.DOC_PERMISSIONS_DESCRIPTION = exports.DOCUMENT_PERMISSIONS_TITLE = exports.DOCUMENT_PERMISSIONS_TEXT = exports.DOCUMENT_PERMISSIONS_LINK = exports.DOCUMENT_PERMISSIONS_DISABLED_TEXT = exports.DELETION_SYNC_LABEL = exports.DELETION_SYNC_DESCRIPTION = exports.CUSTOM_DOCUMENTATION_LINK = exports.CUSTOM_CALLOUT_TITLE = exports.CREATED_LABEL = exports.CONTENT_TYPE_HEADER = exports.CONTENT_SUMMARY_TITLE = exports.CONTENT_SUMMARY_LOADING_TEXT = exports.CONTENT_LOADING_TEXT = exports.CONFIGURATION_TITLE = exports.BLOCKED_TIME_WINDOWS_TITLE = exports.BLOCKED_EMPTY_STATE_TITLE = exports.BLOCKED_EMPTY_STATE_DESCRIPTION = exports.ADDITIONAL_CONFIG_HEADING = void 0;
exports.UTC_TITLE = exports.UNDERSTAND_BUTTON = exports.TOTAL_DOCUMENTS_LABEL = exports.TITLE_HEADING = exports.TIME_HEADER = exports.SYNC_UNSAVED_CHANGES_MESSAGE = exports.SYNC_SETTINGS_UPDATED_MESSAGE = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const SOURCES_NO_CONTENT_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.noContent.title', {
  defaultMessage: 'No content yet'
});
exports.SOURCES_NO_CONTENT_TITLE = SOURCES_NO_CONTENT_TITLE;
const CONTENT_SUMMARY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.contentSummary.title', {
  defaultMessage: 'Content summary'
});
exports.CONTENT_SUMMARY_TITLE = CONTENT_SUMMARY_TITLE;
const CONTENT_SUMMARY_LOADING_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.contentSummaryLoading.text', {
  defaultMessage: 'Loading summary details...'
});
exports.CONTENT_SUMMARY_LOADING_TEXT = CONTENT_SUMMARY_LOADING_TEXT;
const CONTENT_TYPE_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.contentType.header', {
  defaultMessage: 'Content type'
});
exports.CONTENT_TYPE_HEADER = CONTENT_TYPE_HEADER;
const ITEMS_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.items.header', {
  defaultMessage: 'Items'
});
exports.ITEMS_HEADER = ITEMS_HEADER;
const EVENT_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.event.header', {
  defaultMessage: 'Event'
});
exports.EVENT_HEADER = EVENT_HEADER;
const STATUS_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.status.header', {
  defaultMessage: 'Status'
});
exports.STATUS_HEADER = STATUS_HEADER;
const TIME_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.time.header', {
  defaultMessage: 'Time'
});
exports.TIME_HEADER = TIME_HEADER;
const TOTAL_DOCUMENTS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.totalDocuments.label', {
  defaultMessage: 'Total documents'
});
exports.TOTAL_DOCUMENTS_LABEL = TOTAL_DOCUMENTS_LABEL;
const EMPTY_ACTIVITY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.emptyActivity.title', {
  defaultMessage: 'There is no recent activity'
});
exports.EMPTY_ACTIVITY_TITLE = EMPTY_ACTIVITY_TITLE;
const GROUP_ACCESS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.groupAccess.title', {
  defaultMessage: 'Group access'
});
exports.GROUP_ACCESS_TITLE = GROUP_ACCESS_TITLE;
const CONFIGURATION_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.configuration.title', {
  defaultMessage: 'Configuration'
});
exports.CONFIGURATION_TITLE = CONFIGURATION_TITLE;
const DOCUMENT_PERMISSIONS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.documentPermissions.title', {
  defaultMessage: 'Document-level permissions'
});
exports.DOCUMENT_PERMISSIONS_TITLE = DOCUMENT_PERMISSIONS_TITLE;
const DOCUMENT_PERMISSIONS_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.documentPermissions.text', {
  defaultMessage: 'Using document-level permissions'
});
exports.DOCUMENT_PERMISSIONS_TEXT = DOCUMENT_PERMISSIONS_TEXT;
const DOCUMENT_PERMISSIONS_DISABLED_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.documentPermissionsDisabled.text', {
  defaultMessage: 'Disabled for this source'
});
exports.DOCUMENT_PERMISSIONS_DISABLED_TEXT = DOCUMENT_PERMISSIONS_DISABLED_TEXT;
const LEARN_MORE_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.learnMore.link', {
  defaultMessage: 'Learn more'
});
exports.LEARN_MORE_LINK = LEARN_MORE_LINK;
const STATUS_HEADING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.status.heading', {
  defaultMessage: 'Everything looks good'
});
exports.STATUS_HEADING = STATUS_HEADING;
const STATUS_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.status.text', {
  defaultMessage: 'Your endpoints are ready to accept requests.'
});
exports.STATUS_TEXT = STATUS_TEXT;
const ADDITIONAL_CONFIG_HEADING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.additionalConfig.heading', {
  defaultMessage: 'Requires additional configuration'
});
exports.ADDITIONAL_CONFIG_HEADING = ADDITIONAL_CONFIG_HEADING;
const EXTERNAL_IDENTITIES_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.externalIdentities.link', {
  defaultMessage: 'External Identities API'
});
exports.EXTERNAL_IDENTITIES_LINK = EXTERNAL_IDENTITIES_LINK;
const ID_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.id.label', {
  defaultMessage: 'Source Identifier'
});
exports.ID_LABEL = ID_LABEL;
const LEARN_CUSTOM_FEATURES_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.learnCustom.features.button', {
  defaultMessage: 'Learn about Platinum features'
});
exports.LEARN_CUSTOM_FEATURES_BUTTON = LEARN_CUSTOM_FEATURES_BUTTON;
const DOC_PERMISSIONS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.docPermissions.description', {
  defaultMessage: 'Document-level permissions manage content access on individual or group attributes. Allow or deny access to specific documents.'
});
exports.DOC_PERMISSIONS_DESCRIPTION = DOC_PERMISSIONS_DESCRIPTION;
const CUSTOM_CALLOUT_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.customCallout.title', {
  defaultMessage: 'Getting started with custom sources?'
});
exports.CUSTOM_CALLOUT_TITLE = CUSTOM_CALLOUT_TITLE;
const NO_CONTENT_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.noContentEmpty.message', {
  defaultMessage: "This source doesn't have any content yet"
});
exports.NO_CONTENT_MESSAGE = NO_CONTENT_MESSAGE;
const CUSTOM_DOCUMENTATION_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.customSourceDocs.link', {
  defaultMessage: 'documentation'
});
exports.CUSTOM_DOCUMENTATION_LINK = CUSTOM_DOCUMENTATION_LINK;
const TITLE_HEADING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.displaySettings.title.heading', {
  defaultMessage: 'Title'
});
exports.TITLE_HEADING = TITLE_HEADING;
const LAST_UPDATED_HEADING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.displaySettings.lastUpdated.heading', {
  defaultMessage: 'Last updated'
});
exports.LAST_UPDATED_HEADING = LAST_UPDATED_HEADING;
const GO_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.displaySettings.go.button', {
  defaultMessage: 'Go'
});
exports.GO_BUTTON = GO_BUTTON;
const RESET_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.displaySettings.reset.button', {
  defaultMessage: 'Reset'
});
exports.RESET_BUTTON = RESET_BUTTON;
const SOURCE_CONTENT_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceContent.title', {
  defaultMessage: 'Source content'
});
exports.SOURCE_CONTENT_TITLE = SOURCE_CONTENT_TITLE;
const SEARCH_CONTENT_PLACEHOLDER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceContent.searchBar.search.placeholder', {
  defaultMessage: 'Search content...'
});
exports.SEARCH_CONTENT_PLACEHOLDER = SEARCH_CONTENT_PLACEHOLDER;
const FILTER_CONTENT_PLACEHOLDER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceContent.searchBar.filter.placeholder', {
  defaultMessage: 'Filter content...'
});
exports.FILTER_CONTENT_PLACEHOLDER = FILTER_CONTENT_PLACEHOLDER;
const CONTENT_LOADING_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.contentLoading.text', {
  defaultMessage: 'Loading content...'
});
exports.CONTENT_LOADING_TEXT = CONTENT_LOADING_TEXT;
const REMOTE_SOURCE_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.remoteSource.label', {
  defaultMessage: 'Remote source'
});
exports.REMOTE_SOURCE_LABEL = REMOTE_SOURCE_LABEL;
const CREATED_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.created.label', {
  defaultMessage: 'Created: '
});
exports.CREATED_LABEL = CREATED_LABEL;
const STATUS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.status.label', {
  defaultMessage: 'Status: '
});
exports.STATUS_LABEL = STATUS_LABEL;
const READY_TEXT = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.ready.text', {
  defaultMessage: 'Ready to search'
});
exports.READY_TEXT = READY_TEXT;
const SOURCE_SETTINGS_HEADING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.settings.heading', {
  defaultMessage: 'Settings'
});
exports.SOURCE_SETTINGS_HEADING = SOURCE_SETTINGS_HEADING;
const SOURCE_SETTINGS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.settings.title', {
  defaultMessage: 'Content source name'
});
exports.SOURCE_SETTINGS_TITLE = SOURCE_SETTINGS_TITLE;
const SOURCE_SETTINGS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.settings.description', {
  defaultMessage: 'Customize the name of this content source.'
});
exports.SOURCE_SETTINGS_DESCRIPTION = SOURCE_SETTINGS_DESCRIPTION;
const SOURCE_CONFIG_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.config.title', {
  defaultMessage: 'Content source configuration'
});
exports.SOURCE_CONFIG_TITLE = SOURCE_CONFIG_TITLE;
const SYNC_MANAGEMENT_SYNCHRONIZE_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.syncManagementSynchronizeLabel', {
  defaultMessage: 'Synchronize this source'
});
exports.SYNC_MANAGEMENT_SYNCHRONIZE_LABEL = SYNC_MANAGEMENT_SYNCHRONIZE_LABEL;
const SYNC_MANAGEMENT_THUMBNAILS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.syncManagementThumbnailsLabel', {
  defaultMessage: 'Sync thumbnails'
});
exports.SYNC_MANAGEMENT_THUMBNAILS_LABEL = SYNC_MANAGEMENT_THUMBNAILS_LABEL;
const SYNC_MANAGEMENT_THUMBNAILS_GLOBAL_CONFIG_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.syncManagementGlobalConfigLabel', {
  defaultMessage: 'Sync thumbnails - disabled at global configuration level'
});
exports.SYNC_MANAGEMENT_THUMBNAILS_GLOBAL_CONFIG_LABEL = SYNC_MANAGEMENT_THUMBNAILS_GLOBAL_CONFIG_LABEL;
const SYNC_MANAGEMENT_CONTENT_EXTRACTION_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.contentSources.syncManagementContentExtractionLabel', {
  defaultMessage: 'Sync full-text from files'
});
exports.SYNC_MANAGEMENT_CONTENT_EXTRACTION_LABEL = SYNC_MANAGEMENT_CONTENT_EXTRACTION_LABEL;
const SOURCE_CONFIG_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.config.link', {
  defaultMessage: 'Edit connector settings'
});
exports.SOURCE_CONFIG_LINK = SOURCE_CONFIG_LINK;
const SOURCE_REMOVE_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.remove.title', {
  defaultMessage: 'Remove this content source'
});
exports.SOURCE_REMOVE_TITLE = SOURCE_REMOVE_TITLE;
const SOURCE_REMOVE_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.remove.description', {
  defaultMessage: 'This action cannot be undone.'
});
exports.SOURCE_REMOVE_DESCRIPTION = SOURCE_REMOVE_DESCRIPTION;
const SYNC_DIAGNOSTICS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.syncDiagnosticsTitle', {
  defaultMessage: 'Sync Diagnostics'
});
exports.SYNC_DIAGNOSTICS_TITLE = SYNC_DIAGNOSTICS_TITLE;
const SYNC_DIAGNOSTICS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.syncDiagnosticsDescription', {
  defaultMessage: 'Retrieve relevant diagnostics data for troubleshooting active synchronization processes.'
});
exports.SYNC_DIAGNOSTICS_DESCRIPTION = SYNC_DIAGNOSTICS_DESCRIPTION;
const SYNC_DIAGNOSTICS_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.syncDiagnosticsButton', {
  defaultMessage: 'Download diagnostics data'
});
exports.SYNC_DIAGNOSTICS_BUTTON = SYNC_DIAGNOSTICS_BUTTON;
const DOWNLOAD_DIAGNOSTIC_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.downloadDiagnosticButton', {
  defaultMessage: 'Download diagnostic bundle'
});
exports.DOWNLOAD_DIAGNOSTIC_BUTTON = DOWNLOAD_DIAGNOSTIC_BUTTON;
const SOURCE_NAME_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceName.label', {
  defaultMessage: 'Source name'
});
exports.SOURCE_NAME_LABEL = SOURCE_NAME_LABEL;
const ORG_SOURCES_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.org.link', {
  defaultMessage: 'Add an organization content source'
});
exports.ORG_SOURCES_LINK = ORG_SOURCES_LINK;
const ORG_SOURCES_HEADER_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.org.title', {
  defaultMessage: 'Organization sources'
});
exports.ORG_SOURCES_HEADER_TITLE = ORG_SOURCES_HEADER_TITLE;
const ORG_SOURCES_HEADER_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.org.description', {
  defaultMessage: 'Organization sources are available to the entire organization and can be assigned to specific user groups.'
});
exports.ORG_SOURCES_HEADER_DESCRIPTION = ORG_SOURCES_HEADER_DESCRIPTION;
const PRIVATE_LINK_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.private.link', {
  defaultMessage: 'Add a private content source'
});
exports.PRIVATE_LINK_TITLE = PRIVATE_LINK_TITLE;
const PRIVATE_HEADER_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.private.header.title', {
  defaultMessage: 'My private content sources'
});
exports.PRIVATE_HEADER_TITLE = PRIVATE_HEADER_TITLE;
const PRIVATE_HEADER_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.private.header.description', {
  defaultMessage: 'Private content sources are available only to you.'
});
exports.PRIVATE_HEADER_DESCRIPTION = PRIVATE_HEADER_DESCRIPTION;
const PRIVATE_ORG_SOURCES_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.private.privateOrg.header.title', {
  defaultMessage: 'Organizational content sources'
});
exports.PRIVATE_ORG_SOURCES_TITLE = PRIVATE_ORG_SOURCES_TITLE;
const PRIVATE_EMPTY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.private.empty.title', {
  defaultMessage: 'You have no private sources'
});
exports.PRIVATE_EMPTY_TITLE = PRIVATE_EMPTY_TITLE;
const ORG_SOURCES_EMPTY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.org.empty.title', {
  defaultMessage: 'No content source available'
});
exports.ORG_SOURCES_EMPTY_TITLE = ORG_SOURCES_EMPTY_TITLE;
const ORG_SOURCES_EMPTY_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.org.empty.description', {
  defaultMessage: 'Once content sources are shared with you, they will be displayed here, and available via the search experience.'
});
exports.ORG_SOURCES_EMPTY_DESCRIPTION = ORG_SOURCES_EMPTY_DESCRIPTION;
const LICENSE_CALLOUT_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.licenseCallout.title', {
  defaultMessage: 'Private Sources are no longer available'
});
exports.LICENSE_CALLOUT_TITLE = LICENSE_CALLOUT_TITLE;
const LICENSE_CALLOUT_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.licenseCallout.description', {
  defaultMessage: 'Contact your search experience administrator for more information.'
});
exports.LICENSE_CALLOUT_DESCRIPTION = LICENSE_CALLOUT_DESCRIPTION;
const SOURCE_DISABLED_CALLOUT_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceDisabled.title', {
  defaultMessage: 'Content source is disabled'
});
exports.SOURCE_DISABLED_CALLOUT_TITLE = SOURCE_DISABLED_CALLOUT_TITLE;
const SOURCE_DISABLED_CALLOUT_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceDisabled.description', {
  defaultMessage: 'Your organization’s license level has changed. Your data is safe, but document-level permissions are no longer supported and searching of this source has been disabled. Upgrade to a Platinum license to re-enable this source.'
});
exports.SOURCE_DISABLED_CALLOUT_DESCRIPTION = SOURCE_DISABLED_CALLOUT_DESCRIPTION;
const SOURCE_DISABLED_CALLOUT_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceDisabled.button', {
  defaultMessage: 'Explore Platinum license'
});
exports.SOURCE_DISABLED_CALLOUT_BUTTON = SOURCE_DISABLED_CALLOUT_BUTTON;
const DOCUMENT_PERMISSIONS_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.documentPermissionsLink', {
  defaultMessage: 'Learn more about document-level permission configuration'
});
exports.DOCUMENT_PERMISSIONS_LINK = DOCUMENT_PERMISSIONS_LINK;
const UNDERSTAND_BUTTON = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.understandButton', {
  defaultMessage: 'I understand'
});
exports.UNDERSTAND_BUTTON = UNDERSTAND_BUTTON;
const PERSONAL_DASHBOARD_READ_ONLY_MODE_WARNING = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.personalDashboard.readOnlyMode.warning', {
  defaultMessage: 'Workplace Search is currently available for search only, due to regular maintenance. Contact your system administrator for more information.'
});
exports.PERSONAL_DASHBOARD_READ_ONLY_MODE_WARNING = PERSONAL_DASHBOARD_READ_ONLY_MODE_WARNING;
const PERSONAL_DASHBOARD_SOURCE_ERROR = error => _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.personalDashboardSourceError', {
  defaultMessage: 'Could not connect the source, reach out to your admin for help. Error message: {error}',
  values: {
    error
  }
});
exports.PERSONAL_DASHBOARD_SOURCE_ERROR = PERSONAL_DASHBOARD_SOURCE_ERROR;
const SOURCE_OVERVIEW_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceOverviewTitle', {
  defaultMessage: 'Source overview'
});
exports.SOURCE_OVERVIEW_TITLE = SOURCE_OVERVIEW_TITLE;
const SOURCE_SYNCHRONIZATION_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationDescription', {
  defaultMessage: 'Enable or disable synchronization of data from this content source to Workplace Search.'
});
exports.SOURCE_SYNCHRONIZATION_DESCRIPTION = SOURCE_SYNCHRONIZATION_DESCRIPTION;
const SOURCE_FREQUENCY_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceFrequencyDescription', {
  defaultMessage: 'Manage the frequency of data synchronization from Workplace search to this content source. Sync more frequently to ensure your data is up to date. Sync less frequently to reduce the burden on third party servers.'
});
exports.SOURCE_FREQUENCY_DESCRIPTION = SOURCE_FREQUENCY_DESCRIPTION;
const SOURCE_ASSETS_AND_OBJECTS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceAssetsAndObjectsDescription', {
  defaultMessage: 'Flexibly manage the documents to be synchronized and made available for search using granular controls below.'
});
exports.SOURCE_ASSETS_AND_OBJECTS_DESCRIPTION = SOURCE_ASSETS_AND_OBJECTS_DESCRIPTION;
const SOURCE_ASSETS_AND_OBJECTS_LEARN_MORE_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceAssetsAndObjectsLearnMoreLink', {
  defaultMessage: 'Learn more about sync objects types.'
});
exports.SOURCE_ASSETS_AND_OBJECTS_LEARN_MORE_LINK = SOURCE_ASSETS_AND_OBJECTS_LEARN_MORE_LINK;
const SOURCE_ASSETS_AND_OBJECTS_ASSETS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceAssetsAndObjectsAssetsLabel', {
  defaultMessage: 'Assets'
});
exports.SOURCE_ASSETS_AND_OBJECTS_ASSETS_LABEL = SOURCE_ASSETS_AND_OBJECTS_ASSETS_LABEL;
const SOURCE_ASSETS_AND_OBJECTS_OBJECTS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceAssetsAndObjectsObjectsLabel', {
  defaultMessage: 'Objects'
});
exports.SOURCE_ASSETS_AND_OBJECTS_OBJECTS_LABEL = SOURCE_ASSETS_AND_OBJECTS_OBJECTS_LABEL;
const SOURCE_SYNCHRONIZATION_TOGGLE_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationToggleLabel', {
  defaultMessage: 'Synchronize this source'
});
exports.SOURCE_SYNCHRONIZATION_TOGGLE_LABEL = SOURCE_SYNCHRONIZATION_TOGGLE_LABEL;
const SOURCE_SYNCHRONIZATION_TOGGLE_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationToggleDescription', {
  defaultMessage: 'Source content will automatically be kept in sync.'
});
exports.SOURCE_SYNCHRONIZATION_TOGGLE_DESCRIPTION = SOURCE_SYNCHRONIZATION_TOGGLE_DESCRIPTION;
const SOURCE_SYNCHRONIZATION_FREQUENCY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationFrequencyTitle', {
  defaultMessage: 'Synchronization frequency'
});
exports.SOURCE_SYNCHRONIZATION_FREQUENCY_TITLE = SOURCE_SYNCHRONIZATION_FREQUENCY_TITLE;
const SOURCE_SYNCHRONIZATION_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationTitle', {
  defaultMessage: 'Synchronization'
});
exports.SOURCE_SYNCHRONIZATION_TITLE = SOURCE_SYNCHRONIZATION_TITLE;
const SOURCE_SYNCHRONIZATION_BUTTON_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSynchronizationButtonLabel', {
  defaultMessage: 'Synchronize content'
});
exports.SOURCE_SYNCHRONIZATION_BUTTON_LABEL = SOURCE_SYNCHRONIZATION_BUTTON_LABEL;
const SOURCE_SYNC_FREQUENCY_LINK_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSyncFrequencyLinkLabel', {
  defaultMessage: 'sync frequency'
});
exports.SOURCE_SYNC_FREQUENCY_LINK_LABEL = SOURCE_SYNC_FREQUENCY_LINK_LABEL;
const SOURCE_BLOCKED_TIME_WINDOWS_LINK_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceBlockedTimeWindowsLinkLabel', {
  defaultMessage: 'blocked time windows'
});
exports.SOURCE_BLOCKED_TIME_WINDOWS_LINK_LABEL = SOURCE_BLOCKED_TIME_WINDOWS_LINK_LABEL;
const SOURCE_SYNC_CONFIRM_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSyncConfirmTitle', {
  defaultMessage: 'Start new content sync?'
});
exports.SOURCE_SYNC_CONFIRM_TITLE = SOURCE_SYNC_CONFIRM_TITLE;
const SOURCE_SYNC_CONFIRM_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSyncConfirmMessage', {
  defaultMessage: 'Are you sure you would like to continue with this request and stop all other syncs?'
});
exports.SOURCE_SYNC_CONFIRM_MESSAGE = SOURCE_SYNC_CONFIRM_MESSAGE;
const SOURCE_SYNC_FREQUENCY_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.sourceSyncFrequencyTitle', {
  defaultMessage: 'Sync frequency'
});
exports.SOURCE_SYNC_FREQUENCY_TITLE = SOURCE_SYNC_FREQUENCY_TITLE;
const BLOCKED_TIME_WINDOWS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.blockedWindowsTitle', {
  defaultMessage: 'Blocked time windows'
});
exports.BLOCKED_TIME_WINDOWS_TITLE = BLOCKED_TIME_WINDOWS_TITLE;
const SYNCHRONIZATION_DISABLED_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.synchronizationDisabledTitle', {
  defaultMessage: 'Source synchronization is disabled.'
});
exports.SYNCHRONIZATION_DISABLED_TITLE = SYNCHRONIZATION_DISABLED_TITLE;
const SYNCHRONIZATION_DISABLED_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.synchronizationDisabledDescription', {
  defaultMessage: 'Contact your administrator to enable synchronization controls.'
});
exports.SYNCHRONIZATION_DISABLED_DESCRIPTION = SYNCHRONIZATION_DISABLED_DESCRIPTION;
const FULL_SYNC_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.fullSyncLabel', {
  defaultMessage: 'Full sync'
});
exports.FULL_SYNC_LABEL = FULL_SYNC_LABEL;
const FULL_SYNC_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.fullSyncDescription', {
  defaultMessage: 'Retrieves all the documents possible from the content source'
});
exports.FULL_SYNC_DESCRIPTION = FULL_SYNC_DESCRIPTION;
const INCREMENTAL_SYNC_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.incrementalSyncLabel', {
  defaultMessage: 'Incremental sync'
});
exports.INCREMENTAL_SYNC_LABEL = INCREMENTAL_SYNC_LABEL;
const INCREMENTAL_SYNC_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.incrementalSyncDescription', {
  defaultMessage: 'Retrieves documents/updates since the last sync job'
});
exports.INCREMENTAL_SYNC_DESCRIPTION = INCREMENTAL_SYNC_DESCRIPTION;
const DELETION_SYNC_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.deletionSyncLabel', {
  defaultMessage: 'Deletion sync'
});
exports.DELETION_SYNC_LABEL = DELETION_SYNC_LABEL;
const DELETION_SYNC_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.deletionSyncDescription', {
  defaultMessage: 'Removes documents that no longer exist in the content source'
});
exports.DELETION_SYNC_DESCRIPTION = DELETION_SYNC_DESCRIPTION;
const PERMISSIONS_SYNC_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.permissionsSyncLabel', {
  defaultMessage: 'Permissions sync'
});
exports.PERMISSIONS_SYNC_LABEL = PERMISSIONS_SYNC_LABEL;
const PERMISSIONS_SYNC_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.permissionsSyncDescription', {
  defaultMessage: 'Retrieves all permission changes since the last sync job'
});
exports.PERMISSIONS_SYNC_DESCRIPTION = PERMISSIONS_SYNC_DESCRIPTION;
const BLOCKED_EMPTY_STATE_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.blockedEmptyStateTitle', {
  defaultMessage: 'You have no blocked time windows'
});
exports.BLOCKED_EMPTY_STATE_TITLE = BLOCKED_EMPTY_STATE_TITLE;
const BLOCKED_EMPTY_STATE_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.blockedEmptyStateDescription', {
  defaultMessage: 'Add a blocked time window to only perform syncs at the right time.'
});
exports.BLOCKED_EMPTY_STATE_DESCRIPTION = BLOCKED_EMPTY_STATE_DESCRIPTION;
const SYNC_SETTINGS_UPDATED_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.syncSettingsUpdatedMessage', {
  defaultMessage: 'Source synchronization settings updated.'
});
exports.SYNC_SETTINGS_UPDATED_MESSAGE = SYNC_SETTINGS_UPDATED_MESSAGE;
const SYNC_UNSAVED_CHANGES_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.syncUnsavedChangesMessage', {
  defaultMessage: 'Your changes have not been saved. Are you sure you want to leave?'
});
exports.SYNC_UNSAVED_CHANGES_MESSAGE = SYNC_UNSAVED_CHANGES_MESSAGE;
const NEXT_SYNC_RUNNING_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.nextSyncRunningMessage', {
  defaultMessage: 'as soon as the currently running job finishes'
});
exports.NEXT_SYNC_RUNNING_MESSAGE = NEXT_SYNC_RUNNING_MESSAGE;
const UTC_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.utcTitle', {
  defaultMessage: 'All times are in UTC'
});
exports.UTC_TITLE = UTC_TITLE;
