// @kbn/ecs-data-quality-dashboard-plugin/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PLUGIN_NAME = exports.PLUGIN_ID = exports.INTERNAL_API_VERSION = exports.GET_UNALLOWED_FIELD_VALUES = exports.GET_INDEX_STATS = exports.GET_INDEX_MAPPINGS = exports.GET_ILM_EXPLAIN = exports.BASE_PATH = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PLUGIN_ID = 'ecsDataQualityDashboard';
exports.PLUGIN_ID = PLUGIN_ID;
const PLUGIN_NAME = 'ecsDataQualityDashboard';
exports.PLUGIN_NAME = PLUGIN_NAME;
const BASE_PATH = '/internal/ecs_data_quality_dashboard';
exports.BASE_PATH = BASE_PATH;
const GET_INDEX_STATS = `${BASE_PATH}/stats/{pattern}`;
exports.GET_INDEX_STATS = GET_INDEX_STATS;
const GET_INDEX_MAPPINGS = `${BASE_PATH}/mappings/{pattern}`;
exports.GET_INDEX_MAPPINGS = GET_INDEX_MAPPINGS;
const GET_UNALLOWED_FIELD_VALUES = `${BASE_PATH}/unallowed_field_values`;
exports.GET_UNALLOWED_FIELD_VALUES = GET_UNALLOWED_FIELD_VALUES;
const GET_ILM_EXPLAIN = `${BASE_PATH}/ilm_explain/{pattern}`;
exports.GET_ILM_EXPLAIN = GET_ILM_EXPLAIN;
const INTERNAL_API_VERSION = '1';
exports.INTERNAL_API_VERSION = INTERNAL_API_VERSION;
