// @kbn/vis-type-xy-plugin/public/types/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThresholdLineStyle = exports.ScaleType = exports.InterpolationMode = exports.ChartMode = exports.AxisType = exports.AxisMode = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
let ChartMode;
exports.ChartMode = ChartMode;
(function (ChartMode) {
  ChartMode["Normal"] = "normal";
  ChartMode["Stacked"] = "stacked";
})(ChartMode || (exports.ChartMode = ChartMode = {}));
let InterpolationMode;
exports.InterpolationMode = InterpolationMode;
(function (InterpolationMode) {
  InterpolationMode["Linear"] = "linear";
  InterpolationMode["Cardinal"] = "cardinal";
  InterpolationMode["StepAfter"] = "step-after";
})(InterpolationMode || (exports.InterpolationMode = InterpolationMode = {}));
let AxisType;
exports.AxisType = AxisType;
(function (AxisType) {
  AxisType["Category"] = "category";
  AxisType["Value"] = "value";
})(AxisType || (exports.AxisType = AxisType = {}));
let ScaleType;
exports.ScaleType = ScaleType;
(function (ScaleType) {
  ScaleType["Linear"] = "linear";
  ScaleType["Log"] = "log";
  ScaleType["SquareRoot"] = "square root";
})(ScaleType || (exports.ScaleType = ScaleType = {}));
let AxisMode;
exports.AxisMode = AxisMode;
(function (AxisMode) {
  AxisMode["Normal"] = "normal";
  AxisMode["Percentage"] = "percentage";
  AxisMode["Wiggle"] = "wiggle";
  AxisMode["Silhouette"] = "silhouette";
})(AxisMode || (exports.AxisMode = AxisMode = {}));
let ThresholdLineStyle;
exports.ThresholdLineStyle = ThresholdLineStyle;
(function (ThresholdLineStyle) {
  ThresholdLineStyle["Full"] = "full";
  ThresholdLineStyle["Dashed"] = "dashed";
  ThresholdLineStyle["DotDashed"] = "dot-dashed";
})(ThresholdLineStyle || (exports.ThresholdLineStyle = ThresholdLineStyle = {}));
