// @kbn/infra-plugin/public/components/asset_details/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.INTEGRATION_NAME = exports.ContentTabIds = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
let ContentTabIds;
exports.ContentTabIds = ContentTabIds;
(function (ContentTabIds) {
  ContentTabIds["OVERVIEW"] = "overview";
  ContentTabIds["METADATA"] = "metadata";
  ContentTabIds["PROCESSES"] = "processes";
  ContentTabIds["PROFILING"] = "profiling";
  ContentTabIds["ANOMALIES"] = "anomalies";
  ContentTabIds["OSQUERY"] = "osquery";
  ContentTabIds["LOGS"] = "logs";
  ContentTabIds["LINK_TO_APM"] = "linkToApm";
})(ContentTabIds || (exports.ContentTabIds = ContentTabIds = {}));
let INTEGRATION_NAME;
exports.INTEGRATION_NAME = INTEGRATION_NAME;
(function (INTEGRATION_NAME) {
  INTEGRATION_NAME["kubernetes"] = "kubernetes";
})(INTEGRATION_NAME || (exports.INTEGRATION_NAME = INTEGRATION_NAME = {}));
