// @kbn/enterprise-search-plugin/public/applications/enterprise_search_content/routes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SETUP_GUIDE_PATH = exports.SETTINGS_PATH = exports.SEARCH_INDICES_PATH = exports.SEARCH_INDEX_TAB_PATH = exports.SEARCH_INDEX_TAB_DETAIL_PATH = exports.SEARCH_INDEX_PATH = exports.SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = exports.ROOT_PATH = exports.OLD_SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = exports.NEW_INDEX_SELECT_CONNECTOR_PATH = exports.NEW_INDEX_SELECT_CONNECTOR_NATIVE_PATH = exports.NEW_INDEX_SELECT_CONNECTOR_CLIENTS_PATH = exports.NEW_INDEX_PATH = exports.NEW_INDEX_METHOD_PATH = exports.NEW_ES_INDEX_PATH = exports.NEW_DIRECT_UPLOAD_PATH = exports.NEW_API_PATH = exports.ML_NOTIFICATIONS_PATH = exports.ML_MANAGE_TRAINED_MODELS_PATH = exports.ERROR_STATE_PATH = exports.DEV_TOOLS_CONSOLE_PATH = exports.CRAWLERS_PATH = exports.CONNECTORS_PATH = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const ROOT_PATH = '/';
exports.ROOT_PATH = ROOT_PATH;
const SETUP_GUIDE_PATH = '/setup_guide';
exports.SETUP_GUIDE_PATH = SETUP_GUIDE_PATH;
const ERROR_STATE_PATH = '/error_state';
exports.ERROR_STATE_PATH = ERROR_STATE_PATH;
const SEARCH_INDICES_PATH = `${ROOT_PATH}search_indices`;
exports.SEARCH_INDICES_PATH = SEARCH_INDICES_PATH;
const CONNECTORS_PATH = `${ROOT_PATH}connectors`;
exports.CONNECTORS_PATH = CONNECTORS_PATH;
const CRAWLERS_PATH = `${ROOT_PATH}crawlers`;
exports.CRAWLERS_PATH = CRAWLERS_PATH;
const SETTINGS_PATH = `${ROOT_PATH}settings`;
exports.SETTINGS_PATH = SETTINGS_PATH;
const NEW_INDEX_PATH = `${SEARCH_INDICES_PATH}/new_index`;
exports.NEW_INDEX_PATH = NEW_INDEX_PATH;
const NEW_INDEX_METHOD_PATH = `${NEW_INDEX_PATH}/:type`;
exports.NEW_INDEX_METHOD_PATH = NEW_INDEX_METHOD_PATH;
const NEW_API_PATH = `${NEW_INDEX_PATH}/api`;
exports.NEW_API_PATH = NEW_API_PATH;
const NEW_ES_INDEX_PATH = `${NEW_INDEX_PATH}/elasticsearch`;
exports.NEW_ES_INDEX_PATH = NEW_ES_INDEX_PATH;
const NEW_DIRECT_UPLOAD_PATH = `${NEW_INDEX_PATH}/upload`;
exports.NEW_DIRECT_UPLOAD_PATH = NEW_DIRECT_UPLOAD_PATH;
const NEW_INDEX_SELECT_CONNECTOR_PATH = `${NEW_INDEX_PATH}/select_connector`;
exports.NEW_INDEX_SELECT_CONNECTOR_PATH = NEW_INDEX_SELECT_CONNECTOR_PATH;
const NEW_INDEX_SELECT_CONNECTOR_NATIVE_PATH = `${NEW_INDEX_PATH}/select_connector?filter=native`;
exports.NEW_INDEX_SELECT_CONNECTOR_NATIVE_PATH = NEW_INDEX_SELECT_CONNECTOR_NATIVE_PATH;
const NEW_INDEX_SELECT_CONNECTOR_CLIENTS_PATH = `${NEW_INDEX_PATH}/select_connector?filter=connector_clients`;
exports.NEW_INDEX_SELECT_CONNECTOR_CLIENTS_PATH = NEW_INDEX_SELECT_CONNECTOR_CLIENTS_PATH;
const SEARCH_INDEX_PATH = `${SEARCH_INDICES_PATH}/:indexName`;
exports.SEARCH_INDEX_PATH = SEARCH_INDEX_PATH;
const SEARCH_INDEX_TAB_PATH = `${SEARCH_INDEX_PATH}/:tabId`;
exports.SEARCH_INDEX_TAB_PATH = SEARCH_INDEX_TAB_PATH;
const SEARCH_INDEX_TAB_DETAIL_PATH = `${SEARCH_INDEX_TAB_PATH}/:detailId`;
exports.SEARCH_INDEX_TAB_DETAIL_PATH = SEARCH_INDEX_TAB_DETAIL_PATH;
const SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = `${SEARCH_INDEX_PATH}/domain_management/:domainId`;
exports.SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH;
const OLD_SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = `${SEARCH_INDEX_PATH}/crawler/domains/:domainId`;
exports.OLD_SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH = OLD_SEARCH_INDEX_CRAWLER_DOMAIN_DETAIL_PATH;
const ML_MANAGE_TRAINED_MODELS_PATH = '/app/ml/trained_models';
exports.ML_MANAGE_TRAINED_MODELS_PATH = ML_MANAGE_TRAINED_MODELS_PATH;
const ML_NOTIFICATIONS_PATH = '/app/ml/notifications';
exports.ML_NOTIFICATIONS_PATH = ML_NOTIFICATIONS_PATH;
const DEV_TOOLS_CONSOLE_PATH = '/app/dev_tools#/console';
exports.DEV_TOOLS_CONSOLE_PATH = DEV_TOOLS_CONSOLE_PATH;
