// @kbn/deeplinks-observability/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OBSERVABILITY_OVERVIEW_APP_ID = exports.OBSERVABILITY_ONBOARDING_APP_ID = exports.OBSERVABILITY_LOG_EXPLORER = exports.METRICS_APP_ID = exports.LOGS_APP_ID = exports.APM_APP_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const LOGS_APP_ID = 'logs';
exports.LOGS_APP_ID = LOGS_APP_ID;
const OBSERVABILITY_LOG_EXPLORER = 'observability-log-explorer';
exports.OBSERVABILITY_LOG_EXPLORER = OBSERVABILITY_LOG_EXPLORER;
const OBSERVABILITY_OVERVIEW_APP_ID = 'observability-overview';
exports.OBSERVABILITY_OVERVIEW_APP_ID = OBSERVABILITY_OVERVIEW_APP_ID;
const METRICS_APP_ID = 'metrics';
exports.METRICS_APP_ID = METRICS_APP_ID;
const APM_APP_ID = 'apm';
exports.APM_APP_ID = APM_APP_ID;
const OBSERVABILITY_ONBOARDING_APP_ID = 'observabilityOnboarding';
exports.OBSERVABILITY_ONBOARDING_APP_ID = OBSERVABILITY_ONBOARDING_APP_ID;
