// @kbn/security-solution-plugin/public/common/containers/events/last_event_time/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTimelineLastEventTime = void 0;
var _fastDeepEqual = _interopRequireDefault(require("fast-deep-equal"));
var _fp = require("lodash/fp");
var _react = require("react");
var _rxjs = require("rxjs");
var _common = require("@kbn/data-plugin/common");
var _kibana = require("../../../lib/kibana");
var _timeline = require("../../../../../common/search_strategy/timeline");
var i18n = _interopRequireWildcard(require("./translations"));
var _use_app_toasts = require("../../../hooks/use_app_toasts");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const useTimelineLastEventTime = ({
  indexKey,
  indexNames,
  details
}) => {
  const {
    data
  } = (0, _kibana.useKibana)().services;
  const refetch = (0, _react.useRef)(_fp.noop);
  const abortCtrl = (0, _react.useRef)(new AbortController());
  const searchSubscription$ = (0, _react.useRef)(new _rxjs.Subscription());
  const [loading, setLoading] = (0, _react.useState)(false);
  const [TimelineLastEventTimeRequest, setTimelineLastEventTimeRequest] = (0, _react.useState)({
    defaultIndex: indexNames,
    factoryQueryType: _timeline.TimelineEventsQueries.lastEventTime,
    indexKey,
    details
  });
  const [timelineLastEventTimeResponse, setTimelineLastEventTimeResponse] = (0, _react.useState)({
    lastSeen: null,
    refetch: refetch.current,
    errorMessage: undefined
  });
  const {
    addError
  } = (0, _use_app_toasts.useAppToasts)();
  const timelineLastEventTimeSearch = (0, _react.useCallback)(request => {
    const asyncSearch = async () => {
      abortCtrl.current = new AbortController();
      setLoading(true);
      searchSubscription$.current = data.search.search(request, {
        strategy: 'timelineSearchStrategy',
        abortSignal: abortCtrl.current.signal
      }).subscribe({
        next: response => {
          if (!(0, _common.isRunningResponse)(response)) {
            setLoading(false);
            setTimelineLastEventTimeResponse(prevResponse => ({
              ...prevResponse,
              errorMessage: undefined,
              lastSeen: response.lastSeen,
              refetch: refetch.current
            }));
          }
        },
        error: msg => {
          setLoading(false);
          addError(msg, {
            title: i18n.FAIL_LAST_EVENT_TIME
          });
          setTimelineLastEventTimeResponse(prevResponse => ({
            ...prevResponse,
            errorMessage: msg.message
          }));
        }
      });
    };
    searchSubscription$.current.unsubscribe();
    abortCtrl.current.abort();
    asyncSearch();
    refetch.current = asyncSearch;
  }, [data.search, addError]);
  (0, _react.useEffect)(() => {
    setTimelineLastEventTimeRequest(prevRequest => {
      const myRequest = {
        ...prevRequest,
        defaultIndex: indexNames,
        indexKey,
        details
      };
      if (!(0, _fastDeepEqual.default)(prevRequest, myRequest)) {
        return myRequest;
      }
      return prevRequest;
    });
  }, [indexNames, details, indexKey]);
  (0, _react.useEffect)(() => {
    timelineLastEventTimeSearch(TimelineLastEventTimeRequest);
    return () => {
      searchSubscription$.current.unsubscribe();
      abortCtrl.current.abort();
    };
  }, [TimelineLastEventTimeRequest, timelineLastEventTimeSearch]);
  return [loading, timelineLastEventTimeResponse];
};
exports.useTimelineLastEventTime = useTimelineLastEventTime;
