// @kbn/rollup-plugin/public/crud_app/sections/job_list/detail_panel/detail_panel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.JOB_DETAILS_TABS = exports.DetailPanel = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _i18n = require("@kbn/i18n");
var _i18nReact = require("@kbn/i18n-react");
var _eui = require("@elastic/eui");
var _common = require("../../../../../common");
var _services = require("../../../services");
var _kibana_services = require("../../../../kibana_services");
var _components = require("../../components");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const JOB_DETAILS_TABS = [_components.JOB_DETAILS_TAB_SUMMARY, _components.JOB_DETAILS_TAB_TERMS, _components.JOB_DETAILS_TAB_HISTOGRAM, _components.JOB_DETAILS_TAB_METRICS, _components.JOB_DETAILS_TAB_JSON];
exports.JOB_DETAILS_TABS = JOB_DETAILS_TABS;
const tabToUiMetricMap = {
  [_components.JOB_DETAILS_TAB_SUMMARY]: _common.UIM_DETAIL_PANEL_SUMMARY_TAB_CLICK,
  [_components.JOB_DETAILS_TAB_TERMS]: _common.UIM_DETAIL_PANEL_TERMS_TAB_CLICK,
  [_components.JOB_DETAILS_TAB_HISTOGRAM]: _common.UIM_DETAIL_PANEL_HISTOGRAM_TAB_CLICK,
  [_components.JOB_DETAILS_TAB_METRICS]: _common.UIM_DETAIL_PANEL_METRICS_TAB_CLICK,
  [_components.JOB_DETAILS_TAB_JSON]: _common.UIM_DETAIL_PANEL_JSON_TAB_CLICK
};
class DetailPanel extends _react.Component {
  constructor(props) {
    super(props);
  }
  renderTabs() {
    const {
      panelType,
      job,
      openDetailPanel
    } = this.props;
    if (!job) {
      return;
    }
    const {
      id,
      terms,
      histogram,
      metrics
    } = job;
    const renderedTabs = [];
    JOB_DETAILS_TABS.map((tab, index) => {
      if (tab === _components.JOB_DETAILS_TAB_TERMS && !terms.length) {
        return;
      }
      if (tab === _components.JOB_DETAILS_TAB_HISTOGRAM && !histogram.length) {
        return;
      }
      if (tab === _components.JOB_DETAILS_TAB_METRICS && !metrics.length) {
        return;
      }
      const isSelected = tab === panelType;
      renderedTabs.push( /*#__PURE__*/_react.default.createElement(_eui.EuiTab, {
        onClick: () => {
          (0, _kibana_services.trackUiMetric)(_services.METRIC_TYPE.CLICK, tabToUiMetricMap[tab]);
          openDetailPanel({
            panelType: tab,
            jobId: id
          });
        },
        isSelected: isSelected,
        "data-test-subj": `detailPanelTab${isSelected ? 'Selected' : ''}`,
        key: index
      }, _components.tabToHumanizedMap[tab]));
    });
    return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
      size: "s"
    }), /*#__PURE__*/_react.default.createElement(_eui.EuiTabs, null, renderedTabs));
  }
  renderJob() {
    const {
      panelType,
      job
    } = this.props;
    const {
      status,
      documentsProcessed,
      pagesProcessed,
      rollupsIndexed,
      triggerCount,
      json
    } = job;
    const stats = {
      status,
      documentsProcessed,
      pagesProcessed,
      rollupsIndexed,
      triggerCount
    };
    return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutBody, {
      "data-test-subj": "rollupJobDetailTabContent"
    }, /*#__PURE__*/_react.default.createElement(_eui.EuiErrorBoundary, null, /*#__PURE__*/_react.default.createElement(_components.JobDetails, {
      tab: panelType,
      job: job,
      stats: stats,
      json: json
    }))), /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutFooter, null, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
      justifyContent: "flexEnd"
    }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
      grow: false
    }, /*#__PURE__*/_react.default.createElement(_components.JobActionMenu, {
      iconSide: "left",
      jobs: [job],
      anchorPosition: "upRight",
      detailPanel: true,
      iconType: "arrowUp",
      label: _i18n.i18n.translate('xpack.rollupJobs.detailPanel.jobActionMenu.buttonLabel', {
        defaultMessage: 'Manage'
      })
    })))));
  }
  render() {
    const {
      isOpen,
      isLoading,
      closeDetailPanel,
      job,
      jobId
    } = this.props;
    if (!isOpen) {
      return null;
    }
    let content;
    if (isLoading) {
      content = /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutBody, {
        "data-test-subj": "rollupJobDetailLoading"
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
        justifyContent: "flexStart",
        alignItems: "center",
        gutterSize: "s"
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
        grow: false
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiLoadingSpinner, {
        size: "m"
      })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
        grow: false
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiText, null, /*#__PURE__*/_react.default.createElement(_eui.EuiTextColor, {
        color: "subdued"
      }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
        id: "xpack.rollupJobs.detailPanel.loadingLabel",
        defaultMessage: "Loading rollup job\u2026"
      }))))));
    } else if (job) {
      content = this.renderJob();
    } else {
      content = /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutBody, {
        "data-test-subj": "rollupJobDetailJobNotFound"
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, {
        justifyContent: "flexStart",
        alignItems: "center",
        gutterSize: "s"
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
        grow: false
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiIcon, {
        size: "m",
        type: "warning",
        color: "danger"
      })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, {
        grow: false
      }, /*#__PURE__*/_react.default.createElement(_eui.EuiText, null, /*#__PURE__*/_react.default.createElement(_eui.EuiTextColor, {
        color: "subdued"
      }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
        id: "xpack.rollupJobs.detailPanel.notFoundLabel",
        defaultMessage: "Rollup job not found"
      }))))));
    }
    return /*#__PURE__*/_react.default.createElement(_eui.EuiFlyout, {
      "data-test-subj": "rollupJobDetailFlyout",
      onClose: closeDetailPanel,
      "aria-labelledby": "rollupJobDetailsFlyoutTitle",
      size: "m",
      maxWidth: 550
    }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutHeader, null, /*#__PURE__*/_react.default.createElement(_eui.EuiTitle, {
      size: "m",
      id: "rollupJobDetailsFlyoutTitle",
      "data-test-subj": "rollupJobDetailsFlyoutTitle"
    }, /*#__PURE__*/_react.default.createElement("h2", null, jobId)), this.renderTabs()), content);
  }
}
exports.DetailPanel = DetailPanel;
(0, _defineProperty2.default)(DetailPanel, "propTypes", {
  isOpen: _propTypes.default.bool.isRequired,
  isLoading: _propTypes.default.bool,
  job: _propTypes.default.object,
  jobId: _propTypes.default.string,
  panelType: _propTypes.default.oneOf(JOB_DETAILS_TABS),
  closeDetailPanel: _propTypes.default.func.isRequired,
  openDetailPanel: _propTypes.default.func.isRequired
});
(0, _defineProperty2.default)(DetailPanel, "defaultProps", {
  panelType: JOB_DETAILS_TABS[0]
});
