// @kbn/spaces-plugin/public/share_saved_objects_to_space/components/relatives_footer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RelativesFooter = void 0;
var _eui = require("@elastic/eui");
var _react = _interopRequireWildcard(require("react"));
var _i18nReact = require("@kbn/i18n-react");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const RelativesFooter = props => {
  const {
    savedObjectTarget,
    referenceGraph,
    isDisabled
  } = props;
  const relativesCount = (0, _react.useMemo)(() => {
    const {
      type,
      id
    } = savedObjectTarget;
    return referenceGraph.filter(x => (x.type !== type || x.id !== id) && x.spaces.length > 0 && !x.isMissing).length;
  }, [savedObjectTarget, referenceGraph]);
  if (relativesCount > 0) {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiText, {
      size: "s",
      color: isDisabled ? 'subdued' : undefined
    }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
      id: "xpack.spaces.shareToSpace.relativesControl.description",
      defaultMessage: "{relativesCount} related {relativesCount, plural, one {object} other {objects}} will also change.",
      values: {
        relativesCount
      }
    })), /*#__PURE__*/_react.default.createElement(_eui.EuiHorizontalRule, {
      margin: "s"
    }));
  }
  return null;
};
exports.RelativesFooter = RelativesFooter;
