// @kbn/ui-actions-enhanced-plugin/public/drilldowns/drilldown_manager/containers/create_drilldown_form/create_drilldown_form.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreateDrilldownForm = void 0;
var React = _interopRequireWildcard(require("react"));
var _i18n = require("@kbn/i18n");
var _useMountedState = _interopRequireDefault(require("react-use/lib/useMountedState"));
var _drilldown_manager_title = require("../drilldown_manager_title");
var _context = require("../context");
var _action_factory_picker = require("../action_factory_picker");
var _drilldown_manager_footer = require("../drilldown_manager_footer");
var _drilldown_state_form = require("../drilldown_state_form");
var _button_submit = require("../../components/button_submit");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const txtCreateDrilldown = _i18n.i18n.translate('uiActionsEnhanced.drilldowns.containers.createDrilldownForm.title', {
  defaultMessage: 'Create Drilldown',
  description: 'Drilldowns flyout title for new drilldown form.'
});
const txtCreateDrilldownButton = _i18n.i18n.translate('uiActionsEnhanced.drilldowns.containers.createDrilldownForm.primaryButton', {
  defaultMessage: 'Create drilldown',
  description: 'Primary button on new drilldown creation form.'
});
const CreateDrilldownForm = () => {
  const isMounted = (0, _useMountedState.default)();
  const drilldowns = (0, _context.useDrilldownManager)();
  const drilldownState = drilldowns.getDrilldownState();
  const error = drilldownState.useError();
  const [disabled, setDisabled] = React.useState(false);
  const handleCreate = () => {
    setDisabled(true);
    drilldowns.createDrilldown().finally(() => {
      if (!isMounted()) return;
      setDisabled(false);
    });
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_drilldown_manager_title.DrilldownManagerTitle, null, txtCreateDrilldown), /*#__PURE__*/React.createElement(_action_factory_picker.ActionFactoryPicker, null), !!drilldownState && /*#__PURE__*/React.createElement(_drilldown_state_form.DrilldownStateForm, {
    state: drilldownState,
    disabled: disabled
  }), !!drilldownState && /*#__PURE__*/React.createElement(_drilldown_manager_footer.DrilldownManagerFooter, null, /*#__PURE__*/React.createElement(_button_submit.ButtonSubmit, {
    disabled: disabled || !!error,
    onClick: handleCreate
  }, txtCreateDrilldownButton)));
};
exports.CreateDrilldownForm = CreateDrilldownForm;
