// @kbn/security-solution-plugin/public/timelines/components/side_panel/new_user_detail/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getOktaUserIndex = exports.getEntraUserIndex = exports.ONE_WEEK_IN_HOURS = exports.OKTA_PACKAGE_NAME = exports.MANAGED_USER_QUERY_ID = exports.INSTALL_EA_INTEGRATIONS_HREF = exports.ENTRA_ID_PACKAGE_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
const getEntraUserIndex = (spaceId = 'default') => `logs-entityanalytics_entra_id.user-${spaceId}`;
exports.getEntraUserIndex = getEntraUserIndex;
const ENTRA_ID_PACKAGE_NAME = 'entityanalytics_entra_id';
exports.ENTRA_ID_PACKAGE_NAME = ENTRA_ID_PACKAGE_NAME;
const INSTALL_EA_INTEGRATIONS_HREF = `browse/security?q=entityanalytics`;
exports.INSTALL_EA_INTEGRATIONS_HREF = INSTALL_EA_INTEGRATIONS_HREF;
const ONE_WEEK_IN_HOURS = 24 * 7;
exports.ONE_WEEK_IN_HOURS = ONE_WEEK_IN_HOURS;
const MANAGED_USER_QUERY_ID = 'managedUserDetailsQuery';
exports.MANAGED_USER_QUERY_ID = MANAGED_USER_QUERY_ID;
const getOktaUserIndex = (spaceId = 'default') => `logs-entityanalytics_okta.user-${spaceId}`;
exports.getOktaUserIndex = getOktaUserIndex;
const OKTA_PACKAGE_NAME = 'entityanalytics_okta';
exports.OKTA_PACKAGE_NAME = OKTA_PACKAGE_NAME;
