// @kbn/security-solution-plugin/public/timelines/store/timeline/manage_timeline_id.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ManageEpicTimelineId = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

class ManageEpicTimelineId {
  constructor() {
    (0, _defineProperty2.default)(this, "timelineId", null);
    (0, _defineProperty2.default)(this, "version", null);
    (0, _defineProperty2.default)(this, "templateTimelineId", null);
    (0, _defineProperty2.default)(this, "templateVersion", null);
  }
  getTimelineId() {
    return this.timelineId;
  }
  getTimelineVersion() {
    return this.version;
  }
  getTemplateTimelineId() {
    return this.templateTimelineId;
  }
  getTemplateTimelineVersion() {
    return this.templateVersion;
  }
  setTimelineId(timelineId) {
    this.timelineId = timelineId;
  }
  setTimelineVersion(version) {
    this.version = version;
  }
  setTemplateTimelineId(templateTimelineId) {
    this.templateTimelineId = templateTimelineId;
  }
  setTemplateTimelineVersion(templateVersion) {
    this.templateVersion = templateVersion;
  }
}
exports.ManageEpicTimelineId = ManageEpicTimelineId;
