// @kbn/cloud-security-posture-plugin/public/pages/rules/test_subjects.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getCspRuleTemplatesTableRowItemTestId = exports.CSP_RULES_TABLE_ROW_ITEM_NAME = exports.CSP_RULES_TABLE_ITEM_SWITCH = exports.CSP_RULES_TABLE = exports.CSP_RULES_SHARED_VALUES = exports.CSP_RULES_SAVE_BUTTON = exports.CSP_RULES_FLYOUT_CONTAINER = exports.CSP_RULES_CONTAINER = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const CSP_RULES_CONTAINER = 'csp_rules_container';
exports.CSP_RULES_CONTAINER = CSP_RULES_CONTAINER;
const CSP_RULES_SHARED_VALUES = 'csp_rules_shared_values';
exports.CSP_RULES_SHARED_VALUES = CSP_RULES_SHARED_VALUES;
const CSP_RULES_TABLE_ITEM_SWITCH = 'csp_rules_table_item_switch';
exports.CSP_RULES_TABLE_ITEM_SWITCH = CSP_RULES_TABLE_ITEM_SWITCH;
const CSP_RULES_SAVE_BUTTON = 'csp_rules_table_save_button';
exports.CSP_RULES_SAVE_BUTTON = CSP_RULES_SAVE_BUTTON;
const CSP_RULES_TABLE = 'csp_rules_table';
exports.CSP_RULES_TABLE = CSP_RULES_TABLE;
const CSP_RULES_TABLE_ROW_ITEM_NAME = 'csp_rules_table_row_item_name';
exports.CSP_RULES_TABLE_ROW_ITEM_NAME = CSP_RULES_TABLE_ROW_ITEM_NAME;
const CSP_RULES_FLYOUT_CONTAINER = 'csp_rules_flyout_container';
exports.CSP_RULES_FLYOUT_CONTAINER = CSP_RULES_FLYOUT_CONTAINER;
const getCspRuleTemplatesTableRowItemTestId = id => `${CSP_RULES_TABLE_ROW_ITEM_NAME}_${id}`;
exports.getCspRuleTemplatesTableRowItemTestId = getCspRuleTemplatesTableRowItemTestId;
