// @kbn/ml-plugin/public/application/model_management/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  ModelsTableToConfigMapping: true,
  TrainedModelLink: true
};
exports.ModelsTableToConfigMapping = void 0;
Object.defineProperty(exports, "TrainedModelLink", {
  enumerable: true,
  get: function () {
    return _model_link.TrainedModelLink;
  }
});
var _models_list = require("./models_list");
Object.keys(_models_list).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _models_list[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _models_list[key];
    }
  });
});
var _model_link = require("./model_link");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const ModelsTableToConfigMapping = {
  id: 'model_id',
  description: 'description',
  createdAt: 'create_time',
  type: 'type',
  modelType: 'model_type'
};
exports.ModelsTableToConfigMapping = ModelsTableToConfigMapping;
