// @kbn/synthetics-plugin/public/apps/synthetics/components/monitors_page/management/disabled_callout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DisabledCallout = void 0;
var _react = _interopRequireDefault(require("react"));
var _eui = require("@elastic/eui");
var labels = _interopRequireWildcard(require("./labels"));
var _hooks = require("../../../hooks");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DisabledCallout = ({
  total
}) => {
  const {
    enablement,
    invalidApiKeyError,
    loading
  } = (0, _hooks.useEnablement)();
  const showDisableCallout = !enablement.isEnabled && total > 0;
  const showInvalidApiKeyCallout = invalidApiKeyError && total > 0;
  if (!showDisableCallout && !showInvalidApiKeyCallout) {
    return null;
  }
  return !enablement.canEnable && !loading ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiCallOut, {
    title: labels.CALLOUT_MANAGEMENT_DISABLED,
    color: "warning"
  }, /*#__PURE__*/_react.default.createElement("p", null, labels.CALLOUT_MANAGEMENT_DESCRIPTION), /*#__PURE__*/_react.default.createElement("p", null, labels.CALLOUT_MANAGEMENT_CONTACT_ADMIN, ' ', /*#__PURE__*/_react.default.createElement(_eui.EuiLink, {
    "data-test-subj": "syntheticsMonitorManagementPageLink",
    href: "https://www.elastic.co/guide/en/observability/current/synthetics-get-started-ui.html#uptime-set-up-prereq",
    target: "_blank"
  }, labels.LEARN_MORE_LABEL))), /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
    size: "m"
  })) : null;
};
exports.DisabledCallout = DisabledCallout;
