// @kbn/canvas-plugin/public/routes/workpad/workpad_routing_context.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WorkpadRoutingContextComponent = exports.WorkpadRoutingContext = void 0;
var _react = _interopRequireWildcard(require("react"));
var _use_routing_context = require("./hooks/use_routing_context");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const basicWorkpadRoutingContext = {
  gotoPage: page => undefined,
  getUrl: page => '',
  isFullscreen: false,
  setFullscreen: fullscreen => undefined,
  autoplayInterval: 0,
  setAutoplayInterval: interval => undefined,
  isAutoplayPaused: true,
  setIsAutoplayPaused: isPaused => undefined,
  nextPage: () => undefined,
  previousPage: () => undefined,
  refreshInterval: 0,
  setRefreshInterval: interval => undefined,
  undo: () => undefined,
  redo: () => undefined
};
const WorkpadRoutingContext = /*#__PURE__*/(0, _react.createContext)(basicWorkpadRoutingContext);
exports.WorkpadRoutingContext = WorkpadRoutingContext;
const WorkpadRoutingContextComponent = ({
  children
}) => {
  const routingContext = (0, _use_routing_context.useRoutingContext)();
  return /*#__PURE__*/_react.default.createElement(WorkpadRoutingContext.Provider, {
    value: routingContext
  }, children);
};
exports.WorkpadRoutingContextComponent = WorkpadRoutingContextComponent;
