// @kbn/security-solution-plugin/server/endpoint/lib/protection_updates_note/saved_object_mappings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.protectionUpdatesNoteType = exports.protectionUpdatesNoteSavedObjectType = exports.protectionUpdatesNoteSavedObjectMappings = void 0;
var _coreSavedObjectsServer = require("@kbn/core-saved-objects-server");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const protectionUpdatesNoteSavedObjectType = 'policy-settings-protection-updates-note';
exports.protectionUpdatesNoteSavedObjectType = protectionUpdatesNoteSavedObjectType;
const protectionUpdatesNoteSavedObjectMappings = {
  properties: {
    note: {
      type: 'text',
      index: false
    }
  }
};
exports.protectionUpdatesNoteSavedObjectMappings = protectionUpdatesNoteSavedObjectMappings;
const protectionUpdatesNoteType = {
  name: protectionUpdatesNoteSavedObjectType,
  indexPattern: _coreSavedObjectsServer.SECURITY_SOLUTION_SAVED_OBJECT_INDEX,
  hidden: false,
  namespaceType: 'single',
  mappings: protectionUpdatesNoteSavedObjectMappings
};
exports.protectionUpdatesNoteType = protectionUpdatesNoteType;
