// @kbn/data-plugin/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SCRIPT_LANGUAGES_ROUTE_LATEST_VERSION = exports.SAVED_QUERY_BASE_URL = exports.KQL_TELEMETRY_ROUTE_LATEST_VERSION = exports.KIBANA_USER_QUERY_LANGUAGE_KEY = exports.DEFAULT_QUERY_LANGUAGE = void 0;
Object.defineProperty(exports, "UI_SETTINGS", {
  enumerable: true,
  get: function () {
    return _constants.UI_SETTINGS;
  }
});
var _constants = require("@kbn/data-service/src/constants");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const DEFAULT_QUERY_LANGUAGE = 'kuery';
exports.DEFAULT_QUERY_LANGUAGE = DEFAULT_QUERY_LANGUAGE;
const KIBANA_USER_QUERY_LANGUAGE_KEY = 'kibana.userQueryLanguage';
exports.KIBANA_USER_QUERY_LANGUAGE_KEY = KIBANA_USER_QUERY_LANGUAGE_KEY;
const SAVED_QUERY_BASE_URL = '/internal/saved_query';
exports.SAVED_QUERY_BASE_URL = SAVED_QUERY_BASE_URL;
const KQL_TELEMETRY_ROUTE_LATEST_VERSION = '1';
exports.KQL_TELEMETRY_ROUTE_LATEST_VERSION = KQL_TELEMETRY_ROUTE_LATEST_VERSION;
const SCRIPT_LANGUAGES_ROUTE_LATEST_VERSION = '1';
exports.SCRIPT_LANGUAGES_ROUTE_LATEST_VERSION = SCRIPT_LANGUAGES_ROUTE_LATEST_VERSION;
