// @kbn/cases-plugin/public/containers/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.casesQueriesKeys = exports.casesMutationsKeys = exports.DEFAULT_TABLE_LIMIT = exports.DEFAULT_TABLE_ACTIVE_PAGE = exports.DEFAULT_QUERY_PARAMS = exports.DEFAULT_FILTER_OPTIONS = void 0;
var _types = require("./types");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DEFAULT_TABLE_ACTIVE_PAGE = 1;
exports.DEFAULT_TABLE_ACTIVE_PAGE = DEFAULT_TABLE_ACTIVE_PAGE;
const DEFAULT_TABLE_LIMIT = 10;
exports.DEFAULT_TABLE_LIMIT = DEFAULT_TABLE_LIMIT;
const casesQueriesKeys = {
  all: ['cases'],
  users: ['users'],
  connectors: ['connectors'],
  alerts: ['alerts'],
  userActions: ['user-actions'],
  connectorsList: () => [...casesQueriesKeys.connectors, 'list'],
  casesList: () => [...casesQueriesKeys.all, 'list'],
  casesMetrics: () => [...casesQueriesKeys.casesList(), 'metrics'],
  casesStatuses: () => [...casesQueriesKeys.casesList(), 'statuses'],
  cases: params => [...casesQueriesKeys.casesList(), 'all-cases', params],
  caseView: () => [...casesQueriesKeys.all, 'case'],
  case: id => [...casesQueriesKeys.caseView(), id],
  caseFiles: (id, params) => [...casesQueriesKeys.case(id), 'files', params],
  caseFileStats: id => [...casesQueriesKeys.case(id), 'files', 'stats'],
  caseMetrics: (id, features) => [...casesQueriesKeys.case(id), 'metrics', features],
  caseConnectors: id => [...casesQueriesKeys.case(id), 'connectors'],
  caseUsers: id => [...casesQueriesKeys.case(id), 'users'],
  caseUserActions: (id, params) => [...casesQueriesKeys.case(id), ...casesQueriesKeys.userActions, params],
  caseUserActionsStats: id => [...casesQueriesKeys.case(id), ...casesQueriesKeys.userActions, 'stats'],
  userProfiles: () => [...casesQueriesKeys.users, 'user-profiles'],
  userProfilesList: ids => [...casesQueriesKeys.userProfiles(), ids],
  currentUser: () => [...casesQueriesKeys.users, 'current-user'],
  suggestUsers: params => [...casesQueriesKeys.users, 'suggest', params],
  connectorTypes: () => [...casesQueriesKeys.connectors, 'types'],
  license: () => [...casesQueriesKeys.connectors, 'license'],
  tags: () => [...casesQueriesKeys.all, 'tags'],
  categories: () => [...casesQueriesKeys.all, 'categories'],
  alertFeatureIds: alertRegistrationContexts => [...casesQueriesKeys.alerts, 'features', alertRegistrationContexts],
  configuration: params => [...casesQueriesKeys.all, 'configuration', params]
};
exports.casesQueriesKeys = casesQueriesKeys;
const casesMutationsKeys = {
  createCase: ['create-case'],
  deleteCases: ['delete-cases'],
  updateCase: ['update-case'],
  updateCases: ['update-cases'],
  pushCase: ['push-case'],
  updateComment: ['update-comment'],
  deleteComment: ['delete-comment'],
  deleteFileAttachment: ['delete-file-attachment'],
  bulkCreateAttachments: ['bulk-create-attachments'],
  persistCaseConfiguration: ['persist-case-configuration']
};
exports.casesMutationsKeys = casesMutationsKeys;
const DEFAULT_SEARCH_FIELDS = ['title', 'description'];
const DEFAULT_FILTER_OPTIONS = {
  search: '',
  searchFields: DEFAULT_SEARCH_FIELDS,
  severity: [],
  assignees: [],
  reporters: [],
  status: [],
  tags: [],
  owner: [],
  category: [],
  customFields: {}
};
exports.DEFAULT_FILTER_OPTIONS = DEFAULT_FILTER_OPTIONS;
const DEFAULT_QUERY_PARAMS = {
  page: DEFAULT_TABLE_ACTIVE_PAGE,
  perPage: DEFAULT_TABLE_LIMIT,
  sortField: _types.SortFieldCase.createdAt,
  sortOrder: 'desc'
};
exports.DEFAULT_QUERY_PARAMS = DEFAULT_QUERY_PARAMS;
