// @kbn/monaco/src/esql/lib/autocomplete/dymanic_item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DynamicAutocompleteItem = void 0;
exports.isDynamicAutocompleteItem = isDynamicAutocompleteItem;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
let DynamicAutocompleteItem;
exports.DynamicAutocompleteItem = DynamicAutocompleteItem;
(function (DynamicAutocompleteItem) {
  DynamicAutocompleteItem["SourceIdentifier"] = "SourceIdentifier";
  DynamicAutocompleteItem["FieldIdentifier"] = "FieldIdentifier";
  DynamicAutocompleteItem["PolicyIdentifier"] = "PolicyIdentifier";
  DynamicAutocompleteItem["PolicyFieldIdentifier"] = "PolicyFieldIdentifier";
  DynamicAutocompleteItem["PolicyMatchingFieldIdentifier"] = "PolicyMatchingFieldIdentifier";
})(DynamicAutocompleteItem || (exports.DynamicAutocompleteItem = DynamicAutocompleteItem = {}));
const DynamicAutocompleteItems = Object.values(DynamicAutocompleteItem);
function isDynamicAutocompleteItem(v) {
  return DynamicAutocompleteItems.some(dai => dai === v);
}
