// @kbn/management-plugin/public/components/management_sections.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.managementSections = exports.StackSection = exports.SecuritySection = exports.KibanaSection = exports.InsightsAndAlertingSection = exports.IngestSection = exports.DataSection = void 0;
var _i18n = require("@kbn/i18n");
var _types = require("../types");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const ingestTitle = _i18n.i18n.translate('management.sections.ingestTitle', {
  defaultMessage: 'Ingest'
});
const ingestTip = _i18n.i18n.translate('management.sections.ingestTip', {
  defaultMessage: 'Manage how to transform data and load it into the cluster'
});
const dataTitle = _i18n.i18n.translate('management.sections.dataTitle', {
  defaultMessage: 'Data'
});
const dataTip = _i18n.i18n.translate('management.sections.dataTip', {
  defaultMessage: 'Manage your cluster data and backups'
});
const insightsAndAlertingTitle = _i18n.i18n.translate('management.sections.insightsAndAlertingTitle', {
  defaultMessage: 'Alerts and Insights'
});
const insightsAndAlertingTip = _i18n.i18n.translate('management.sections.insightsAndAlertingTip', {
  defaultMessage: 'Manage how to detect changes in your data'
});
const sectionTitle = _i18n.i18n.translate('management.sections.section.title', {
  defaultMessage: 'Security'
});
const sectionTip = _i18n.i18n.translate('management.sections.section.tip', {
  defaultMessage: 'Control access to features and data'
});
const kibanaTitle = _i18n.i18n.translate('management.sections.kibanaTitle', {
  defaultMessage: 'Kibana'
});
const kibanaTip = _i18n.i18n.translate('management.sections.kibanaTip', {
  defaultMessage: 'Customize Kibana and manage saved objects'
});
const stackTitle = _i18n.i18n.translate('management.sections.stackTitle', {
  defaultMessage: 'Stack'
});
const stackTip = _i18n.i18n.translate('management.sections.stackTip', {
  defaultMessage: 'Manage your license and upgrade the Stack'
});
const IngestSection = {
  id: _types.ManagementSectionId.Ingest,
  title: ingestTitle,
  tip: ingestTip,
  order: 0
};
exports.IngestSection = IngestSection;
const DataSection = {
  id: _types.ManagementSectionId.Data,
  title: dataTitle,
  tip: dataTip,
  order: 1
};
exports.DataSection = DataSection;
const InsightsAndAlertingSection = {
  id: _types.ManagementSectionId.InsightsAndAlerting,
  title: insightsAndAlertingTitle,
  tip: insightsAndAlertingTip,
  order: 2
};
exports.InsightsAndAlertingSection = InsightsAndAlertingSection;
const SecuritySection = {
  id: 'security',
  title: sectionTitle,
  tip: sectionTip,
  order: 3
};
exports.SecuritySection = SecuritySection;
const KibanaSection = {
  id: _types.ManagementSectionId.Kibana,
  title: kibanaTitle,
  tip: kibanaTip,
  order: 4
};
exports.KibanaSection = KibanaSection;
const StackSection = {
  id: _types.ManagementSectionId.Stack,
  title: stackTitle,
  tip: stackTip,
  order: 4
};
exports.StackSection = StackSection;
const managementSections = [IngestSection, DataSection, InsightsAndAlertingSection, SecuritySection, KibanaSection, StackSection];
exports.managementSections = managementSections;
