// @kbn/synthetics-plugin/public/apps/synthetics/components/step_details_page/step_waterfall_chart/waterfall/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SIDEBAR_GROW_SIZE = exports.RESOURCE_TITLE_FONT_SIZE = exports.MAIN_GROW_SIZE = exports.FIXED_AXIS_HEIGHT = exports.CHART_LEGEND_PADDING = exports.CHART_HEADER_HEIGHT = exports.CANVAS_MAX_ITEMS = exports.BAR_HEIGHT = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// Pixel value
const BAR_HEIGHT = 24;
// Flex grow value
exports.BAR_HEIGHT = BAR_HEIGHT;
const MAIN_GROW_SIZE = 8;
// Flex grow value
exports.MAIN_GROW_SIZE = MAIN_GROW_SIZE;
const SIDEBAR_GROW_SIZE = 3;
// Axis height
// NOTE: This isn't a perfect solution - changes in font size etc within charts could change the ideal height here.
exports.SIDEBAR_GROW_SIZE = SIDEBAR_GROW_SIZE;
const FIXED_AXIS_HEIGHT = 24;

// Font size of the request titles/links in the left sidebar
exports.FIXED_AXIS_HEIGHT = FIXED_AXIS_HEIGHT;
const RESOURCE_TITLE_FONT_SIZE = 14;

// number of items to display in canvas, since canvas can only have limited size
exports.RESOURCE_TITLE_FONT_SIZE = RESOURCE_TITLE_FONT_SIZE;
const CANVAS_MAX_ITEMS = 150;
exports.CANVAS_MAX_ITEMS = CANVAS_MAX_ITEMS;
const CHART_LEGEND_PADDING = 33;
exports.CHART_LEGEND_PADDING = CHART_LEGEND_PADDING;
const CHART_HEADER_HEIGHT = 146;
exports.CHART_HEADER_HEIGHT = CHART_HEADER_HEIGHT;
