// @kbn/security-solution-plugin/common/api/detection_engine/prebuilt_rules/perform_rule_installation/perform_rule_installation_route.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SkipRuleInstallReason = exports.RuleVersionSpecifier = exports.PerformRuleInstallationRequestBody = exports.InstallSpecificRulesRequest = exports.InstallAllRulesRequest = void 0;
var t = _interopRequireWildcard(require("io-ts"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const RuleVersionSpecifier = t.exact(t.type({
  rule_id: t.string,
  version: t.number
}));
exports.RuleVersionSpecifier = RuleVersionSpecifier;
const InstallSpecificRulesRequest = t.exact(t.type({
  mode: t.literal(`SPECIFIC_RULES`),
  rules: t.array(RuleVersionSpecifier)
}));
exports.InstallSpecificRulesRequest = InstallSpecificRulesRequest;
const InstallAllRulesRequest = t.exact(t.type({
  mode: t.literal(`ALL_RULES`)
}));
exports.InstallAllRulesRequest = InstallAllRulesRequest;
const PerformRuleInstallationRequestBody = t.union([InstallAllRulesRequest, InstallSpecificRulesRequest]);
exports.PerformRuleInstallationRequestBody = PerformRuleInstallationRequestBody;
let SkipRuleInstallReason;
exports.SkipRuleInstallReason = SkipRuleInstallReason;
(function (SkipRuleInstallReason) {
  SkipRuleInstallReason["ALREADY_INSTALLED"] = "ALREADY_INSTALLED";
})(SkipRuleInstallReason || (exports.SkipRuleInstallReason = SkipRuleInstallReason = {}));
