// @kbn/security-solution-plugin/server/lib/entity_analytics/risk_engine/routes/privileges.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.riskEnginePrivilegesRoute = void 0;
var _utils = require("@kbn/lists-plugin/server/routes/utils");
var _securitysolutionEsUtils = require("@kbn/securitysolution-es-utils");
var _constants = require("../../../../../common/constants");
var _get_user_risk_engine_privileges = require("../get_user_risk_engine_privileges");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const riskEnginePrivilegesRoute = (router, getStartServices) => {
  router.versioned.get({
    access: 'internal',
    path: _constants.RISK_ENGINE_PRIVILEGES_URL,
    options: {
      tags: ['access:securitySolution', `access:${_constants.APP_ID}-entity-analytics`]
    }
  }).addVersion({
    version: '1',
    validate: false
  }, async (context, request, response) => {
    const siemResponse = (0, _utils.buildSiemResponse)(response);
    const [_, {
      security
    }] = await getStartServices();
    const body = await (0, _get_user_risk_engine_privileges.getUserRiskEnginePrivileges)(request, security);
    try {
      return response.ok({
        body
      });
    } catch (e) {
      const error = (0, _securitysolutionEsUtils.transformError)(e);
      return siemResponse.error({
        statusCode: error.statusCode,
        body: error.message
      });
    }
  });
};
exports.riskEnginePrivilegesRoute = riskEnginePrivilegesRoute;
