// @kbn/expressions-plugin/server/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ExpressionFunction", {
  enumerable: true,
  get: function () {
    return _common.ExpressionFunction;
  }
});
Object.defineProperty(exports, "ExpressionRenderer", {
  enumerable: true,
  get: function () {
    return _common.ExpressionRenderer;
  }
});
Object.defineProperty(exports, "ExpressionType", {
  enumerable: true,
  get: function () {
    return _common.ExpressionType;
  }
});
Object.defineProperty(exports, "FontStyle", {
  enumerable: true,
  get: function () {
    return _common.FontStyle;
  }
});
Object.defineProperty(exports, "FontWeight", {
  enumerable: true,
  get: function () {
    return _common.FontWeight;
  }
});
Object.defineProperty(exports, "Overflow", {
  enumerable: true,
  get: function () {
    return _common.Overflow;
  }
});
Object.defineProperty(exports, "TextAlignment", {
  enumerable: true,
  get: function () {
    return _common.TextAlignment;
  }
});
Object.defineProperty(exports, "TextDecoration", {
  enumerable: true,
  get: function () {
    return _common.TextDecoration;
  }
});
Object.defineProperty(exports, "buildExpression", {
  enumerable: true,
  get: function () {
    return _common.buildExpression;
  }
});
Object.defineProperty(exports, "buildExpressionFunction", {
  enumerable: true,
  get: function () {
    return _common.buildExpressionFunction;
  }
});
Object.defineProperty(exports, "format", {
  enumerable: true,
  get: function () {
    return _common.format;
  }
});
Object.defineProperty(exports, "formatExpression", {
  enumerable: true,
  get: function () {
    return _common.formatExpression;
  }
});
Object.defineProperty(exports, "isExpressionAstBuilder", {
  enumerable: true,
  get: function () {
    return _common.isExpressionAstBuilder;
  }
});
Object.defineProperty(exports, "parse", {
  enumerable: true,
  get: function () {
    return _common.parse;
  }
});
Object.defineProperty(exports, "parseExpression", {
  enumerable: true,
  get: function () {
    return _common.parseExpression;
  }
});
exports.plugin = plugin;
var _common = require("../common");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

// Kibana Platform.

async function plugin(initializerContext) {
  const {
    ExpressionsServerPlugin
  } = await Promise.resolve().then(() => _interopRequireWildcard(require('./plugin')));
  return new ExpressionsServerPlugin(initializerContext);
}

// Static exports.
