// @kbn/security-solution-plugin/public/timelines/components/timeline/body/renderers/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.USER_NAME_FIELD_NAME = exports.SIGNAL_STATUS_FIELD_NAME = exports.SIGNAL_RULE_NAME_FIELD_NAME = exports.SIGNAL_ASSIGNEE_IDS_FIELD_NAME = exports.RULE_REFERENCE_FIELD_NAME = exports.REFERENCE_URL_FIELD_NAME = exports.REASON_FIELD_NAME = exports.QUARANTINED_PATH_FIELD_NAME = exports.MESSAGE_FIELD_NAME = exports.IP_FIELD_TYPE = exports.HOST_NAME_FIELD_NAME = exports.GEO_FIELD_TYPE = exports.EVENT_URL_FIELD_NAME = exports.EVENT_SUMMARY_FIELD_NAME = exports.EVENT_MODULE_FIELD_NAME = exports.DATE_FIELD_TYPE = exports.AGENT_STATUS_FIELD_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DATE_FIELD_TYPE = 'date';
exports.DATE_FIELD_TYPE = DATE_FIELD_TYPE;
const HOST_NAME_FIELD_NAME = 'host.name';
exports.HOST_NAME_FIELD_NAME = HOST_NAME_FIELD_NAME;
const USER_NAME_FIELD_NAME = 'user.name';
exports.USER_NAME_FIELD_NAME = USER_NAME_FIELD_NAME;
const IP_FIELD_TYPE = 'ip';
exports.IP_FIELD_TYPE = IP_FIELD_TYPE;
const GEO_FIELD_TYPE = 'geo_point';
exports.GEO_FIELD_TYPE = GEO_FIELD_TYPE;
const MESSAGE_FIELD_NAME = 'message';
exports.MESSAGE_FIELD_NAME = MESSAGE_FIELD_NAME;
const EVENT_MODULE_FIELD_NAME = 'event.module';
exports.EVENT_MODULE_FIELD_NAME = EVENT_MODULE_FIELD_NAME;
const RULE_REFERENCE_FIELD_NAME = 'rule.reference';
exports.RULE_REFERENCE_FIELD_NAME = RULE_REFERENCE_FIELD_NAME;
const REFERENCE_URL_FIELD_NAME = 'reference.url';
exports.REFERENCE_URL_FIELD_NAME = REFERENCE_URL_FIELD_NAME;
const EVENT_URL_FIELD_NAME = 'event.url';
exports.EVENT_URL_FIELD_NAME = EVENT_URL_FIELD_NAME;
const SIGNAL_RULE_NAME_FIELD_NAME = 'kibana.alert.rule.name';
exports.SIGNAL_RULE_NAME_FIELD_NAME = SIGNAL_RULE_NAME_FIELD_NAME;
const SIGNAL_STATUS_FIELD_NAME = 'kibana.alert.workflow_status';
exports.SIGNAL_STATUS_FIELD_NAME = SIGNAL_STATUS_FIELD_NAME;
const SIGNAL_ASSIGNEE_IDS_FIELD_NAME = 'kibana.alert.workflow_assignee_ids';
exports.SIGNAL_ASSIGNEE_IDS_FIELD_NAME = SIGNAL_ASSIGNEE_IDS_FIELD_NAME;
const AGENT_STATUS_FIELD_NAME = 'agent.status';
exports.AGENT_STATUS_FIELD_NAME = AGENT_STATUS_FIELD_NAME;
const QUARANTINED_PATH_FIELD_NAME = 'quarantined.path';
exports.QUARANTINED_PATH_FIELD_NAME = QUARANTINED_PATH_FIELD_NAME;
const REASON_FIELD_NAME = 'kibana.alert.reason';
exports.REASON_FIELD_NAME = REASON_FIELD_NAME;
const EVENT_SUMMARY_FIELD_NAME = 'eventSummary';
exports.EVENT_SUMMARY_FIELD_NAME = EVENT_SUMMARY_FIELD_NAME;
