// @kbn/reporting-export-types-csv/csv_v2.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CsvV2ExportType = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _boom = _interopRequireDefault(require("@hapi/boom"));
var _generateCsv = require("@kbn/generate-csv");
var _reportingCommon = require("@kbn/reporting-common");
var _reportingExportTypesCsvCommon = require("@kbn/reporting-export-types-csv-common");
var _reportingServer = require("@kbn/reporting-server");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

class CsvV2ExportType extends _reportingServer.ExportType {
  constructor(...args) {
    super(...args);
    (0, _defineProperty2.default)(this, "id", _reportingExportTypesCsvCommon.CSV_REPORT_TYPE_V2);
    (0, _defineProperty2.default)(this, "name", _reportingExportTypesCsvCommon.CSV_REPORT_TYPE_V2);
    (0, _defineProperty2.default)(this, "jobType", _reportingExportTypesCsvCommon.CSV_REPORT_TYPE_V2);
    (0, _defineProperty2.default)(this, "jobContentEncoding", 'base64');
    (0, _defineProperty2.default)(this, "jobContentExtension", 'csv');
    (0, _defineProperty2.default)(this, "validLicenses", [_reportingCommon.LICENSE_TYPE_TRIAL, _reportingCommon.LICENSE_TYPE_BASIC, _reportingCommon.LICENSE_TYPE_CLOUD_STANDARD, _reportingCommon.LICENSE_TYPE_GOLD, _reportingCommon.LICENSE_TYPE_PLATINUM, _reportingCommon.LICENSE_TYPE_ENTERPRISE]);
    (0, _defineProperty2.default)(this, "createJob", async (jobParams, _context, req) => {
      // 1. Validation of locatorParams
      const {
        locatorParams
      } = jobParams;
      const {
        id,
        params
      } = locatorParams[0];
      if (!locatorParams || !Array.isArray(locatorParams) || locatorParams.length !== 1 || id !== 'DISCOVER_APP_LOCATOR' || !params) {
        throw _boom.default.badRequest('Invalid Job params: must contain a single Discover App locator');
      }
      if (!params || !params.savedSearchId || typeof params.savedSearchId !== 'string') {
        throw _boom.default.badRequest('Invalid Discover App locator: must contain a savedSearchId');
      }

      // use Discover contract to get the title of the report from job params
      const {
        discover: discoverPluginStart
      } = this.startDeps;
      const locatorClient = await discoverPluginStart.locator.asScopedClient(req);
      const title = await locatorClient.titleFromLocator(params);
      return {
        ...jobParams,
        title,
        objectType: 'search',
        isDeprecated: false
      };
    });
    (0, _defineProperty2.default)(this, "runTask", async (jobId, job, cancellationToken, stream) => {
      const config = this.config;
      const {
        encryptionKey,
        csv: csvConfig
      } = config;
      const logger = this.logger.get(`execute:${jobId}`);
      const headers = await (0, _reportingServer.decryptJobHeaders)(encryptionKey, job.headers, logger);
      const fakeRequest = this.getFakeRequest(headers, job.spaceId, logger);
      const uiSettings = await this.getUiSettingsClient(fakeRequest, logger);
      const fieldFormatsRegistry = await (0, _reportingServer.getFieldFormats)().fieldFormatServiceFactory(uiSettings);
      const {
        data: dataPluginStart,
        discover: discoverPluginStart
      } = this.startDeps;
      const data = dataPluginStart.search.asScoped(fakeRequest);
      const {
        locatorParams
      } = job;
      const {
        params
      } = locatorParams[0];

      // use Discover contract to convert the job params into inputs for CsvGenerator
      const locatorClient = await discoverPluginStart.locator.asScopedClient(fakeRequest);
      const columns = await locatorClient.columnsFromLocator(params);
      const searchSource = await locatorClient.searchSourceFromLocator(params);
      const es = this.startDeps.esClient.asScoped(fakeRequest);
      const searchSourceStart = await dataPluginStart.search.searchSource.asScoped(fakeRequest);
      const clients = {
        uiSettings,
        data,
        es
      };
      const dependencies = {
        searchSourceStart,
        fieldFormatsRegistry
      };
      const csv = new _generateCsv.CsvGenerator({
        columns,
        searchSource: searchSource.getSerializedFields(true),
        ...job
      }, csvConfig, clients, dependencies, cancellationToken, logger, stream);
      return await csv.generateData();
    });
    this.logger = this.logger.get('csv-export-v2');
  }
}
exports.CsvV2ExportType = CsvV2ExportType;
