// @kbn/apm-plugin/common/mobile/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MobileSpanType = exports.MobileSpanSubtype = exports.MapTypes = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
let MapTypes;
exports.MapTypes = MapTypes;
(function (MapTypes) {
  MapTypes["Http"] = "http_requests";
  MapTypes["Session"] = "unique_sessions";
})(MapTypes || (exports.MapTypes = MapTypes = {}));
let MobileSpanSubtype;
exports.MobileSpanSubtype = MobileSpanSubtype;
(function (MobileSpanSubtype) {
  MobileSpanSubtype["Http"] = "http";
})(MobileSpanSubtype || (exports.MobileSpanSubtype = MobileSpanSubtype = {}));
let MobileSpanType;
exports.MobileSpanType = MobileSpanType;
(function (MobileSpanType) {
  MobileSpanType["External"] = "external";
})(MobileSpanType || (exports.MobileSpanType = MobileSpanType = {}));
