// @kbn/alerting-plugin/common/maintenance_window.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.filterStateStore = exports.MaintenanceWindowStatus = exports.MANAGEMENT_APP_ID = exports.MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE = exports.MAINTENANCE_WINDOW_PATHS = exports.MAINTENANCE_WINDOW_FEATURE_ID = exports.MAINTENANCE_WINDOW_DEEP_LINK_IDS = exports.MAINTENANCE_WINDOW_DATE_FORMAT = exports.MAINTENANCE_WINDOW_API_PRIVILEGES = exports.MAINTENANCE_WINDOWS_APP_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
let MaintenanceWindowStatus;
exports.MaintenanceWindowStatus = MaintenanceWindowStatus;
(function (MaintenanceWindowStatus) {
  MaintenanceWindowStatus["Running"] = "running";
  MaintenanceWindowStatus["Upcoming"] = "upcoming";
  MaintenanceWindowStatus["Finished"] = "finished";
  MaintenanceWindowStatus["Archived"] = "archived";
})(MaintenanceWindowStatus || (exports.MaintenanceWindowStatus = MaintenanceWindowStatus = {}));
const filterStateStore = {
  APP_STATE: 'appState',
  GLOBAL_STATE: 'globalState'
};

/**
 * @deprecated Use the data/maintenance_window types instead
 */

/**
 * @deprecated Use the data/maintenance_window types instead
 */

/**
 * @deprecated Use the application/maintenance_window types instead
 */
exports.filterStateStore = filterStateStore;
const MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE = 'maintenance-window';
exports.MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE = MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE;
const MAINTENANCE_WINDOW_FEATURE_ID = 'maintenanceWindow';
exports.MAINTENANCE_WINDOW_FEATURE_ID = MAINTENANCE_WINDOW_FEATURE_ID;
const MAINTENANCE_WINDOW_API_PRIVILEGES = {
  READ_MAINTENANCE_WINDOW: 'read-maintenance-window',
  WRITE_MAINTENANCE_WINDOW: 'write-maintenance-window'
};
exports.MAINTENANCE_WINDOW_API_PRIVILEGES = MAINTENANCE_WINDOW_API_PRIVILEGES;
const MAINTENANCE_WINDOWS_APP_ID = 'maintenanceWindows';
exports.MAINTENANCE_WINDOWS_APP_ID = MAINTENANCE_WINDOWS_APP_ID;
const MANAGEMENT_APP_ID = 'management';
exports.MANAGEMENT_APP_ID = MANAGEMENT_APP_ID;
const MAINTENANCE_WINDOW_PATHS = {
  alerting: {
    maintenanceWindows: `/${MAINTENANCE_WINDOWS_APP_ID}`,
    maintenanceWindowsCreate: '/create',
    maintenanceWindowsEdit: '/edit/:maintenanceWindowId'
  }
};
exports.MAINTENANCE_WINDOW_PATHS = MAINTENANCE_WINDOW_PATHS;
const MAINTENANCE_WINDOW_DEEP_LINK_IDS = {
  maintenanceWindows: MAINTENANCE_WINDOWS_APP_ID,
  maintenanceWindowsCreate: 'create',
  maintenanceWindowsEdit: 'edit'
};
exports.MAINTENANCE_WINDOW_DEEP_LINK_IDS = MAINTENANCE_WINDOW_DEEP_LINK_IDS;
const MAINTENANCE_WINDOW_DATE_FORMAT = 'MM/DD/YY hh:mm A';
exports.MAINTENANCE_WINDOW_DATE_FORMAT = MAINTENANCE_WINDOW_DATE_FORMAT;
