// @kbn/alerting-plugin/common/rules_settings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RULES_SETTINGS_SAVED_OBJECT_TYPE = exports.RULES_SETTINGS_QUERY_DELAY_SAVED_OBJECT_ID = exports.RULES_SETTINGS_FLAPPING_SAVED_OBJECT_ID = exports.RULES_SETTINGS_FEATURE_ID = exports.READ_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = exports.READ_FLAPPING_SETTINGS_SUB_FEATURE_ID = exports.MIN_STATUS_CHANGE_THRESHOLD = exports.MIN_QUERY_DELAY = exports.MIN_LOOK_BACK_WINDOW = exports.MAX_STATUS_CHANGE_THRESHOLD = exports.MAX_QUERY_DELAY = exports.MAX_LOOK_BACK_WINDOW = exports.DISABLE_FLAPPING_SETTINGS = exports.DEFAULT_STATUS_CHANGE_THRESHOLD = exports.DEFAULT_SERVERLESS_QUERY_DELAY_SETTINGS = exports.DEFAULT_SERVERLESS_QUERY_DELAY = exports.DEFAULT_QUERY_DELAY_SETTINGS = exports.DEFAULT_QUERY_DELAY = exports.DEFAULT_LOOK_BACK_WINDOW = exports.DEFAULT_FLAPPING_SETTINGS = exports.API_PRIVILEGES = exports.ALL_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = exports.ALL_FLAPPING_SETTINGS_SUB_FEATURE_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const MIN_LOOK_BACK_WINDOW = 2;
exports.MIN_LOOK_BACK_WINDOW = MIN_LOOK_BACK_WINDOW;
const MAX_LOOK_BACK_WINDOW = 20;
exports.MAX_LOOK_BACK_WINDOW = MAX_LOOK_BACK_WINDOW;
const MIN_STATUS_CHANGE_THRESHOLD = 2;
exports.MIN_STATUS_CHANGE_THRESHOLD = MIN_STATUS_CHANGE_THRESHOLD;
const MAX_STATUS_CHANGE_THRESHOLD = 20;
exports.MAX_STATUS_CHANGE_THRESHOLD = MAX_STATUS_CHANGE_THRESHOLD;
const MIN_QUERY_DELAY = 0;
exports.MIN_QUERY_DELAY = MIN_QUERY_DELAY;
const MAX_QUERY_DELAY = 60;
exports.MAX_QUERY_DELAY = MAX_QUERY_DELAY;
const RULES_SETTINGS_FEATURE_ID = 'rulesSettings';
exports.RULES_SETTINGS_FEATURE_ID = RULES_SETTINGS_FEATURE_ID;
const ALL_FLAPPING_SETTINGS_SUB_FEATURE_ID = 'allFlappingSettings';
exports.ALL_FLAPPING_SETTINGS_SUB_FEATURE_ID = ALL_FLAPPING_SETTINGS_SUB_FEATURE_ID;
const READ_FLAPPING_SETTINGS_SUB_FEATURE_ID = 'readFlappingSettings';
exports.READ_FLAPPING_SETTINGS_SUB_FEATURE_ID = READ_FLAPPING_SETTINGS_SUB_FEATURE_ID;
const ALL_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = 'allQueryDelaySettings';
exports.ALL_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = ALL_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID;
const READ_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = 'readQueryDelaySettings';
exports.READ_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID = READ_QUERY_DELAY_SETTINGS_SUB_FEATURE_ID;
const API_PRIVILEGES = {
  READ_FLAPPING_SETTINGS: 'read-flapping-settings',
  WRITE_FLAPPING_SETTINGS: 'write-flapping-settings',
  READ_QUERY_DELAY_SETTINGS: 'read-query-delay-settings',
  WRITE_QUERY_DELAY_SETTINGS: 'write-query-delay-settings'
};
exports.API_PRIVILEGES = API_PRIVILEGES;
const RULES_SETTINGS_SAVED_OBJECT_TYPE = 'rules-settings';
exports.RULES_SETTINGS_SAVED_OBJECT_TYPE = RULES_SETTINGS_SAVED_OBJECT_TYPE;
const RULES_SETTINGS_FLAPPING_SAVED_OBJECT_ID = 'rules-settings';
exports.RULES_SETTINGS_FLAPPING_SAVED_OBJECT_ID = RULES_SETTINGS_FLAPPING_SAVED_OBJECT_ID;
const RULES_SETTINGS_QUERY_DELAY_SAVED_OBJECT_ID = 'query-delay-settings';
exports.RULES_SETTINGS_QUERY_DELAY_SAVED_OBJECT_ID = RULES_SETTINGS_QUERY_DELAY_SAVED_OBJECT_ID;
const DEFAULT_LOOK_BACK_WINDOW = 20;
exports.DEFAULT_LOOK_BACK_WINDOW = DEFAULT_LOOK_BACK_WINDOW;
const DEFAULT_STATUS_CHANGE_THRESHOLD = 4;
exports.DEFAULT_STATUS_CHANGE_THRESHOLD = DEFAULT_STATUS_CHANGE_THRESHOLD;
const DEFAULT_QUERY_DELAY = 0;
exports.DEFAULT_QUERY_DELAY = DEFAULT_QUERY_DELAY;
const DEFAULT_SERVERLESS_QUERY_DELAY = 15;
exports.DEFAULT_SERVERLESS_QUERY_DELAY = DEFAULT_SERVERLESS_QUERY_DELAY;
const DEFAULT_FLAPPING_SETTINGS = {
  enabled: true,
  lookBackWindow: DEFAULT_LOOK_BACK_WINDOW,
  statusChangeThreshold: DEFAULT_STATUS_CHANGE_THRESHOLD
};
exports.DEFAULT_FLAPPING_SETTINGS = DEFAULT_FLAPPING_SETTINGS;
const DISABLE_FLAPPING_SETTINGS = {
  ...DEFAULT_FLAPPING_SETTINGS,
  enabled: false
};
exports.DISABLE_FLAPPING_SETTINGS = DISABLE_FLAPPING_SETTINGS;
const DEFAULT_QUERY_DELAY_SETTINGS = {
  delay: DEFAULT_QUERY_DELAY
};
exports.DEFAULT_QUERY_DELAY_SETTINGS = DEFAULT_QUERY_DELAY_SETTINGS;
const DEFAULT_SERVERLESS_QUERY_DELAY_SETTINGS = {
  delay: DEFAULT_SERVERLESS_QUERY_DELAY
};
exports.DEFAULT_SERVERLESS_QUERY_DELAY_SETTINGS = DEFAULT_SERVERLESS_QUERY_DELAY_SETTINGS;
