// @kbn/index-management-plugin/public/application/components/component_templates/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UIM_COMPONENT_TEMPLATE_UPDATE = exports.UIM_COMPONENT_TEMPLATE_LIST_LOAD = exports.UIM_COMPONENT_TEMPLATE_DETAILS = exports.UIM_COMPONENT_TEMPLATE_DELETE_MANY = exports.UIM_COMPONENT_TEMPLATE_DELETE = exports.UIM_COMPONENT_TEMPLATE_CREATE = exports.MANAGED_BY_FLEET = exports.APP_CLUSTER_REQUIRED_PRIVILEGES = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// ui metric constants
const UIM_COMPONENT_TEMPLATE_LIST_LOAD = 'component_template_list_load';
exports.UIM_COMPONENT_TEMPLATE_LIST_LOAD = UIM_COMPONENT_TEMPLATE_LIST_LOAD;
const UIM_COMPONENT_TEMPLATE_DELETE = 'component_template_delete';
exports.UIM_COMPONENT_TEMPLATE_DELETE = UIM_COMPONENT_TEMPLATE_DELETE;
const UIM_COMPONENT_TEMPLATE_DELETE_MANY = 'component_template_delete_many';
exports.UIM_COMPONENT_TEMPLATE_DELETE_MANY = UIM_COMPONENT_TEMPLATE_DELETE_MANY;
const UIM_COMPONENT_TEMPLATE_DETAILS = 'component_template_details';
exports.UIM_COMPONENT_TEMPLATE_DETAILS = UIM_COMPONENT_TEMPLATE_DETAILS;
const UIM_COMPONENT_TEMPLATE_CREATE = 'component_template_create';
exports.UIM_COMPONENT_TEMPLATE_CREATE = UIM_COMPONENT_TEMPLATE_CREATE;
const UIM_COMPONENT_TEMPLATE_UPDATE = 'component_template_update';

// privileges
exports.UIM_COMPONENT_TEMPLATE_UPDATE = UIM_COMPONENT_TEMPLATE_UPDATE;
const APP_CLUSTER_REQUIRED_PRIVILEGES = ['manage_index_templates'];
exports.APP_CLUSTER_REQUIRED_PRIVILEGES = APP_CLUSTER_REQUIRED_PRIVILEGES;
const MANAGED_BY_FLEET = 'fleet';
exports.MANAGED_BY_FLEET = MANAGED_BY_FLEET;
