// @kbn/fleet-plugin/public/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CustomAssetsAccordion", {
  enumerable: true,
  get: function () {
    return _custom_assets_accordion.CustomAssetsAccordion;
  }
});
Object.defineProperty(exports, "PackageIcon", {
  enumerable: true,
  get: function () {
    return _package_icon.PackageIcon;
  }
});
Object.defineProperty(exports, "PackagePolicyEditorDatastreamMappings", {
  enumerable: true,
  get: function () {
    return _datastream_mappings.PackagePolicyEditorDatastreamMappings;
  }
});
Object.defineProperty(exports, "PackagePolicyEditorDatastreamPipelines", {
  enumerable: true,
  get: function () {
    return _datastream_pipelines.PackagePolicyEditorDatastreamPipelines;
  }
});
Object.defineProperty(exports, "pagePathGetters", {
  enumerable: true,
  get: function () {
    return _constants.pagePathGetters;
  }
});
Object.defineProperty(exports, "pkgKeyFromPackageInfo", {
  enumerable: true,
  get: function () {
    return _services.pkgKeyFromPackageInfo;
  }
});
exports.plugin = void 0;
var _plugin = require("./plugin");
var _constants = require("./constants");
var _services = require("./services");
var _custom_assets_accordion = require("./components/custom_assets_accordion");
var _package_icon = require("./components/package_icon");
var _datastream_pipelines = require("./applications/fleet/sections/agent_policy/create_package_policy_page/components/datastream_pipelines");
var _datastream_mappings = require("./applications/fleet/sections/agent_policy/create_package_policy_page/components/datastream_mappings");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const plugin = initializerContext => {
  return new _plugin.FleetPlugin(initializerContext);
};

// Export Package editor components for custom editors
exports.plugin = plugin;
