// @kbn/infra-plugin/public/components/asset_details/links/link_to_alerts.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LinkToAlertsRule = void 0;
var _react = _interopRequireDefault(require("react"));
var _i18nReact = require("@kbn/i18n-react");
var _eui = require("@elastic/eui");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const LinkToAlertsRule = ({
  onClick
}) => {
  return /*#__PURE__*/_react.default.createElement(_eui.EuiButtonEmpty, {
    "data-test-subj": "infraAssetDetailsCreateAlertsRuleButton",
    onClick: onClick,
    size: "xs",
    iconSide: "left",
    flush: "both",
    iconType: "bell"
  }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.infra.infra.assetDetails.alerts.createAlertLink",
    defaultMessage: "Create rule"
  }));
};
exports.LinkToAlertsRule = LinkToAlertsRule;
