// @kbn/synthetics-plugin/public/apps/synthetics/state/monitor_list/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.updateManagementPageStateAction = exports.quietFetchMonitorListAction = exports.fetchUpsertSuccessAction = exports.fetchUpsertMonitorAction = exports.fetchUpsertFailureAction = exports.fetchMonitorListAction = exports.fetchMonitorFiltersAction = exports.enableMonitorAlertAction = exports.clearMonitorUpsertStatus = exports.cleanMonitorListState = void 0;
var _toolkit = require("@reduxjs/toolkit");
var _actions = require("../utils/actions");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const fetchMonitorListAction = (0, _actions.createAsyncAction)('fetchMonitorListAction');
exports.fetchMonitorListAction = fetchMonitorListAction;
const quietFetchMonitorListAction = (0, _toolkit.createAction)('quietFetchMonitorListAction');
exports.quietFetchMonitorListAction = quietFetchMonitorListAction;
const fetchUpsertMonitorAction = (0, _toolkit.createAction)('fetchUpsertMonitor');
exports.fetchUpsertMonitorAction = fetchUpsertMonitorAction;
const fetchUpsertSuccessAction = (0, _toolkit.createAction)('fetchUpsertMonitorSuccess');
exports.fetchUpsertSuccessAction = fetchUpsertSuccessAction;
const fetchUpsertFailureAction = (0, _toolkit.createAction)('fetchUpsertMonitorFailure');
exports.fetchUpsertFailureAction = fetchUpsertFailureAction;
const enableMonitorAlertAction = (0, _actions.createAsyncAction)('enableMonitorAlertAction');
exports.enableMonitorAlertAction = enableMonitorAlertAction;
const clearMonitorUpsertStatus = (0, _toolkit.createAction)('clearMonitorUpsertStatus');
exports.clearMonitorUpsertStatus = clearMonitorUpsertStatus;
const updateManagementPageStateAction = (0, _toolkit.createAction)('updateManagementPageState');
exports.updateManagementPageStateAction = updateManagementPageStateAction;
const cleanMonitorListState = (0, _toolkit.createAction)('cleanMonitorListState');
exports.cleanMonitorListState = cleanMonitorListState;
const fetchMonitorFiltersAction = (0, _actions.createAsyncAction)('fetchMonitorFiltersAction');
exports.fetchMonitorFiltersAction = fetchMonitorFiltersAction;
