// @kbn/security-solution-plugin/common/api/detection_engine/model/rule_schema/specific_attributes/threshold_attributes.gen.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThresholdWithCardinality = exports.ThresholdValue = exports.ThresholdNormalized = exports.ThresholdFieldNormalized = exports.ThresholdField = exports.ThresholdCardinality = exports.ThresholdAlertSuppression = exports.Threshold = void 0;
var _zod = require("zod");
var _common_attributes = require("../common_attributes.gen");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/*
 * NOTICE: Do not edit this file manually.
 * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
 */

const ThresholdCardinality = _zod.z.array(_zod.z.object({
  field: _zod.z.string(),
  value: _zod.z.number().int().min(0)
}));

/**
 * Threshold value
 */
exports.ThresholdCardinality = ThresholdCardinality;
const ThresholdValue = _zod.z.number().int().min(1);

/**
 * Field to aggregate on
 */
exports.ThresholdValue = ThresholdValue;
const ThresholdField = _zod.z.union([_zod.z.string(), _zod.z.array(_zod.z.string())]);

/**
 * Field to aggregate on
 */
exports.ThresholdField = ThresholdField;
const ThresholdFieldNormalized = _zod.z.array(_zod.z.string());
exports.ThresholdFieldNormalized = ThresholdFieldNormalized;
const Threshold = _zod.z.object({
  field: ThresholdField,
  value: ThresholdValue,
  cardinality: ThresholdCardinality.optional()
});
exports.Threshold = Threshold;
const ThresholdNormalized = _zod.z.object({
  field: ThresholdFieldNormalized,
  value: ThresholdValue,
  cardinality: ThresholdCardinality.optional()
});
exports.ThresholdNormalized = ThresholdNormalized;
const ThresholdWithCardinality = _zod.z.object({
  field: ThresholdFieldNormalized,
  value: ThresholdValue,
  cardinality: ThresholdCardinality
});
exports.ThresholdWithCardinality = ThresholdWithCardinality;
const ThresholdAlertSuppression = _zod.z.object({
  duration: _common_attributes.AlertSuppressionDuration
});
exports.ThresholdAlertSuppression = ThresholdAlertSuppression;
