// @kbn/cases-plugin/public/components/configure_cases/closure_options_radio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ClosureOptionsRadio = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var i18n = _interopRequireWildcard(require("./translations"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const radios = [{
  id: 'close-by-user',
  label: i18n.CASE_CLOSURE_OPTIONS_MANUAL
}, {
  id: 'close-by-pushing',
  label: i18n.CASE_CLOSURE_OPTIONS_NEW_INCIDENT
}];
const ClosureOptionsRadioComponent = ({
  closureTypeSelected,
  disabled,
  onChangeClosureType
}) => {
  const onChangeLocal = (0, _react.useCallback)(id => {
    onChangeClosureType(id);
  }, [onChangeClosureType]);
  return /*#__PURE__*/_react.default.createElement(_eui.EuiRadioGroup, {
    disabled: disabled,
    options: radios,
    idSelected: closureTypeSelected,
    onChange: onChangeLocal,
    name: "closure_options",
    "data-test-subj": "closure-options-radio-group"
  });
};
ClosureOptionsRadioComponent.displayName = 'ClosureOptionsRadio';
const ClosureOptionsRadio = /*#__PURE__*/_react.default.memo(ClosureOptionsRadioComponent);
exports.ClosureOptionsRadio = ClosureOptionsRadio;
