// @kbn/enterprise-search-plugin/public/applications/workplace_search/views/role_mappings/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.USER_ROLE_TYPE_DESCRIPTION = exports.SPECIFIC_GROUPS_LABEL = exports.SPECIFIC_GROUPS_DESCRIPTION = exports.ROLE_MAPPING_UPDATED_MESSAGE = exports.ROLE_MAPPING_DELETED_MESSAGE = exports.ROLE_MAPPING_CREATED_MESSAGE = exports.ROLE_MAPPINGS_TABLE_HEADER = exports.GROUP_ASSIGNMENT_LABEL = exports.GROUP_ASSIGNMENT_INVALID_ERROR = exports.DEFAULT_GROUP_NAME = exports.ALL_GROUPS_LABEL = exports.ALL_GROUPS_DESCRIPTION = exports.ADMIN_ROLE_TYPE_DESCRIPTION = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const ROLE_MAPPING_DELETED_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMappingDeletedMessage', {
  defaultMessage: 'Successfully deleted role mapping'
});
exports.ROLE_MAPPING_DELETED_MESSAGE = ROLE_MAPPING_DELETED_MESSAGE;
const ROLE_MAPPING_CREATED_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMappingCreatedMessage', {
  defaultMessage: 'Role mapping successfully created.'
});
exports.ROLE_MAPPING_CREATED_MESSAGE = ROLE_MAPPING_CREATED_MESSAGE;
const ROLE_MAPPING_UPDATED_MESSAGE = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMappingUpdatedMessage', {
  defaultMessage: 'Role mapping successfully updated.'
});
exports.ROLE_MAPPING_UPDATED_MESSAGE = ROLE_MAPPING_UPDATED_MESSAGE;
const DEFAULT_GROUP_NAME = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.defaultGroupName', {
  defaultMessage: 'Default'
});
exports.DEFAULT_GROUP_NAME = DEFAULT_GROUP_NAME;
const ADMIN_ROLE_TYPE_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.adminRoleTypeDescription', {
  defaultMessage: 'Admins have complete access to all organization-wide settings, including content source, group and user management functionality.'
});
exports.ADMIN_ROLE_TYPE_DESCRIPTION = ADMIN_ROLE_TYPE_DESCRIPTION;
const USER_ROLE_TYPE_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.userRoleTypeDescription', {
  defaultMessage: "Users' feature access is limited to search interfaces and personal settings management."
});
exports.USER_ROLE_TYPE_DESCRIPTION = USER_ROLE_TYPE_DESCRIPTION;
const GROUP_ASSIGNMENT_INVALID_ERROR = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.groupAssignmentInvalidError', {
  defaultMessage: 'At least one assigned group is required.'
});
exports.GROUP_ASSIGNMENT_INVALID_ERROR = GROUP_ASSIGNMENT_INVALID_ERROR;
const GROUP_ASSIGNMENT_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.groupAssignmentLabel', {
  defaultMessage: 'Group assignment'
});
exports.GROUP_ASSIGNMENT_LABEL = GROUP_ASSIGNMENT_LABEL;
const ROLE_MAPPINGS_TABLE_HEADER = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.roleMappingsTableHeader', {
  defaultMessage: 'Group Access'
});
exports.ROLE_MAPPINGS_TABLE_HEADER = ROLE_MAPPINGS_TABLE_HEADER;
const ALL_GROUPS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.allGroupsLabel', {
  defaultMessage: 'Assign to all groups'
});
exports.ALL_GROUPS_LABEL = ALL_GROUPS_LABEL;
const ALL_GROUPS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.allGroupsDescription', {
  defaultMessage: 'Assigning to all groups includes all current and future groups as created and administered at a later date.'
});
exports.ALL_GROUPS_DESCRIPTION = ALL_GROUPS_DESCRIPTION;
const SPECIFIC_GROUPS_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.specificGroupsLabel', {
  defaultMessage: 'Assign to specific groups'
});
exports.SPECIFIC_GROUPS_LABEL = SPECIFIC_GROUPS_LABEL;
const SPECIFIC_GROUPS_DESCRIPTION = _i18n.i18n.translate('xpack.enterpriseSearch.workplaceSearch.roleMapping.specificGroupsDescription', {
  defaultMessage: 'Assign to a select set of groups statically.'
});
exports.SPECIFIC_GROUPS_DESCRIPTION = SPECIFIC_GROUPS_DESCRIPTION;
