// @kbn/cloud-defend-plugin/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.STATUS_ROUTE_PATH = exports.PROCESS_INDEX_PATTERN_DEFAULT_NS = exports.PROCESS_INDEX_PATTERN = exports.PROCESS_DATASET = exports.POLICIES_ROUTE_PATH = exports.POLICIES_PACKAGE_POLICY_PREFIX = exports.PLUGIN_NAME = exports.PLUGIN_ID = exports.LOGS_CLOUD_DEFEND_PATTERN = exports.INTEGRATION_PACKAGE_NAME = exports.INPUT_CONTROL = exports.FILE_INDEX_PATTERN_DEFAULT_NS = exports.FILE_INDEX_PATTERN = exports.FILE_DATASET = exports.DEFAULT_POLICIES_PER_PAGE = exports.CURRENT_API_VERSION = exports.CLOUD_DEFEND_FLEET_PACKAGE_KUERY = exports.ALERTS_INDEX_PATTERN_DEFAULT_NS = exports.ALERTS_INDEX_PATTERN = exports.ALERTS_DATASET = void 0;
var _common = require("@kbn/fleet-plugin/common");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PLUGIN_ID = 'cloudDefend';
exports.PLUGIN_ID = PLUGIN_ID;
const PLUGIN_NAME = 'Cloud Defend';
exports.PLUGIN_NAME = PLUGIN_NAME;
const INTEGRATION_PACKAGE_NAME = 'cloud_defend';
exports.INTEGRATION_PACKAGE_NAME = INTEGRATION_PACKAGE_NAME;
const INPUT_CONTROL = 'cloud_defend/control';
exports.INPUT_CONTROL = INPUT_CONTROL;
const LOGS_CLOUD_DEFEND_PATTERN = 'logs-cloud_defend.*';
exports.LOGS_CLOUD_DEFEND_PATTERN = LOGS_CLOUD_DEFEND_PATTERN;
const ALERTS_DATASET = 'cloud_defend.alerts';
exports.ALERTS_DATASET = ALERTS_DATASET;
const ALERTS_INDEX_PATTERN = 'logs-cloud_defend.alerts*';
exports.ALERTS_INDEX_PATTERN = ALERTS_INDEX_PATTERN;
const ALERTS_INDEX_PATTERN_DEFAULT_NS = 'logs-cloud_defend.alerts-default';
exports.ALERTS_INDEX_PATTERN_DEFAULT_NS = ALERTS_INDEX_PATTERN_DEFAULT_NS;
const FILE_DATASET = 'cloud_defend.file';
exports.FILE_DATASET = FILE_DATASET;
const FILE_INDEX_PATTERN = 'logs-cloud_defend.file*';
exports.FILE_INDEX_PATTERN = FILE_INDEX_PATTERN;
const FILE_INDEX_PATTERN_DEFAULT_NS = 'logs-cloud_defend.file-default';
exports.FILE_INDEX_PATTERN_DEFAULT_NS = FILE_INDEX_PATTERN_DEFAULT_NS;
const PROCESS_DATASET = 'cloud_defend.process';
exports.PROCESS_DATASET = PROCESS_DATASET;
const PROCESS_INDEX_PATTERN = 'logs-cloud_defend.process*';
exports.PROCESS_INDEX_PATTERN = PROCESS_INDEX_PATTERN;
const PROCESS_INDEX_PATTERN_DEFAULT_NS = 'logs-cloud_defend.process-default';
exports.PROCESS_INDEX_PATTERN_DEFAULT_NS = PROCESS_INDEX_PATTERN_DEFAULT_NS;
const CURRENT_API_VERSION = '1';
exports.CURRENT_API_VERSION = CURRENT_API_VERSION;
const POLICIES_ROUTE_PATH = '/internal/cloud_defend/policies';
exports.POLICIES_ROUTE_PATH = POLICIES_ROUTE_PATH;
const STATUS_ROUTE_PATH = '/internal/cloud_defend/status';
exports.STATUS_ROUTE_PATH = STATUS_ROUTE_PATH;
const CLOUD_DEFEND_FLEET_PACKAGE_KUERY = `${_common.PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${INTEGRATION_PACKAGE_NAME}`;
exports.CLOUD_DEFEND_FLEET_PACKAGE_KUERY = CLOUD_DEFEND_FLEET_PACKAGE_KUERY;
const DEFAULT_POLICIES_PER_PAGE = 20;
exports.DEFAULT_POLICIES_PER_PAGE = DEFAULT_POLICIES_PER_PAGE;
const POLICIES_PACKAGE_POLICY_PREFIX = 'package_policy.';
exports.POLICIES_PACKAGE_POLICY_PREFIX = POLICIES_PACKAGE_POLICY_PREFIX;
