// @kbn/stack-connectors-plugin/public/connector_types/lib/servicenow/credentials_api_url.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CredentialsApiUrl = void 0;
var _react = _interopRequireWildcard(require("react"));
var _i18nReact = require("@kbn/i18n-react");
var _eui = require("@elastic/eui");
var _hook_form_lib = require("@kbn/es-ui-shared-plugin/static/forms/hook_form_lib");
var _helpers = require("@kbn/es-ui-shared-plugin/static/forms/helpers");
var _components = require("@kbn/es-ui-shared-plugin/static/forms/components");
var _public = require("@kbn/triggers-actions-ui-plugin/public");
var i18n = _interopRequireWildcard(require("./translations"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const {
  urlField
} = _helpers.fieldValidators;
const CredentialsApiUrlComponent = ({
  isLoading,
  readOnly,
  pathPrefix = ''
}) => {
  const {
    docLinks
  } = (0, _public.useKibana)().services;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiFormRow, {
    fullWidth: true
  }, /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.stackConnectors.components.serviceNow.apiUrlHelpLabel",
    defaultMessage: "Provide the full URL to the desired ServiceNow instance. If you don't have one, {instance}.",
    values: {
      instance: /*#__PURE__*/_react.default.createElement(_eui.EuiLink, {
        href: docLinks.links.alerting.serviceNowAction,
        target: "_blank"
      }, i18n.SETUP_DEV_INSTANCE)
    }
  }))), /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
    size: "l"
  }), /*#__PURE__*/_react.default.createElement(_hook_form_lib.UseField, {
    path: `${pathPrefix}config.apiUrl`,
    component: _components.TextField,
    config: {
      label: i18n.API_URL_LABEL,
      validations: [{
        validator: urlField(i18n.API_URL_INVALID)
      }]
    },
    componentProps: {
      euiFieldProps: {
        'data-test-subj': 'credentialsApiUrlFromInput',
        isLoading,
        readOnly,
        disabled: readOnly || isLoading
      }
    }
  }));
};
const CredentialsApiUrl = /*#__PURE__*/(0, _react.memo)(CredentialsApiUrlComponent);
exports.CredentialsApiUrl = CredentialsApiUrl;
