// @kbn/security-solution-features/src/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SERVER_APP_ID = exports.LEGACY_NOTIFICATIONS_ID = exports.CLOUD_POSTURE_APP_ID = exports.CLOUD_DEFEND_APP_ID = exports.CASES_FEATURE_ID = exports.ASSISTANT_FEATURE_ID = exports.APP_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// Same as the plugin id defined by Security Solution
const APP_ID = 'securitySolution';
exports.APP_ID = APP_ID;
const SERVER_APP_ID = 'siem';
exports.SERVER_APP_ID = SERVER_APP_ID;
const CASES_FEATURE_ID = 'securitySolutionCases';
exports.CASES_FEATURE_ID = CASES_FEATURE_ID;
const ASSISTANT_FEATURE_ID = 'securitySolutionAssistant';

// Same as the plugin id defined by Cloud Security Posture
exports.ASSISTANT_FEATURE_ID = ASSISTANT_FEATURE_ID;
const CLOUD_POSTURE_APP_ID = 'csp';

// Same as the plugin id defined by Defend for containers (cloud_defend)
exports.CLOUD_POSTURE_APP_ID = CLOUD_POSTURE_APP_ID;
const CLOUD_DEFEND_APP_ID = 'cloudDefend';

/**
 * Id for the notifications alerting type
 * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
 */
exports.CLOUD_DEFEND_APP_ID = CLOUD_DEFEND_APP_ID;
const LEGACY_NOTIFICATIONS_ID = `siem.notifications`;
exports.LEGACY_NOTIFICATIONS_ID = LEGACY_NOTIFICATIONS_ID;
