// @kbn/lens-plugin/public/datasources/form_based/operations/definitions/terms/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.supportedTypes = exports.MULTI_KEY_VISUAL_SEPARATOR = exports.MAX_TERMS_OTHER_ENABLED = exports.MAXIMUM_MAX_DOC_COUNT = exports.DEFAULT_SIZE = exports.DEFAULT_MAX_DOC_COUNT = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DEFAULT_SIZE = 3;
// Elasticsearch limit
exports.DEFAULT_SIZE = DEFAULT_SIZE;
const MAXIMUM_MAX_DOC_COUNT = 100;
exports.MAXIMUM_MAX_DOC_COUNT = MAXIMUM_MAX_DOC_COUNT;
const DEFAULT_MAX_DOC_COUNT = 1;
exports.DEFAULT_MAX_DOC_COUNT = DEFAULT_MAX_DOC_COUNT;
const supportedTypes = new Set(['string', 'boolean', 'number', 'ip']);
exports.supportedTypes = supportedTypes;
const MULTI_KEY_VISUAL_SEPARATOR = '›';
exports.MULTI_KEY_VISUAL_SEPARATOR = MULTI_KEY_VISUAL_SEPARATOR;
const MAX_TERMS_OTHER_ENABLED = 1000;
exports.MAX_TERMS_OTHER_ENABLED = MAX_TERMS_OTHER_ENABLED;
