// @kbn/security-solution-plugin/public/common/store/sourcerer/actions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setSourcererScopeLoading = exports.setSourcererDataViews = exports.setSignalIndexName = exports.setSelectedDataView = exports.setDataViewLoading = exports.setDataView = void 0;
var _typescriptFsa = _interopRequireDefault(require("typescript-fsa"));
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const actionCreator = (0, _typescriptFsa.default)('x-pack/security_solution/local/sourcerer');
const setDataView = actionCreator('SET_DATA_VIEW');
exports.setDataView = setDataView;
const setDataViewLoading = actionCreator('SET_DATA_VIEW_LOADING');
exports.setDataViewLoading = setDataViewLoading;
const setSignalIndexName = actionCreator('SET_SIGNAL_INDEX_NAME');
exports.setSignalIndexName = setSignalIndexName;
const setSourcererDataViews = actionCreator('SET_SOURCERER_DATA_VIEWS');
exports.setSourcererDataViews = setSourcererDataViews;
const setSourcererScopeLoading = actionCreator('SET_SOURCERER_SCOPE_LOADING');
exports.setSourcererScopeLoading = setSourcererScopeLoading;
const setSelectedDataView = actionCreator('SET_SELECTED_DATA_VIEW');
exports.setSelectedDataView = setSelectedDataView;
