// @kbn/osquery-plugin/public/routes/packs/list/empty_state.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PacksTableEmptyState = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _add_pack_button = require("../../../packs/add_pack_button");
var _osquery_icon = require("../../../components/osquery_icon");
var _load_integration_assets = require("./load_integration_assets");
var _translations = require("./translations");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PacksTableEmptyStateComponent = () => {
  const actions = (0, _react.useMemo)(() => /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, null, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, null, /*#__PURE__*/_react.default.createElement(_load_integration_assets.LoadIntegrationAssetsButton, {
    fill: true
  })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, null, /*#__PURE__*/_react.default.createElement(_add_pack_button.AddPackButton, {
    fill: false
  }))), []);
  return /*#__PURE__*/_react.default.createElement(_eui.EuiEmptyPrompt, {
    icon: /*#__PURE__*/_react.default.createElement(_osquery_icon.OsqueryIcon, null),
    color: "transparent",
    title: /*#__PURE__*/_react.default.createElement("h2", null, _translations.PRE_BUILT_TITLE),
    body: /*#__PURE__*/_react.default.createElement("p", null, _translations.PRE_BUILT_MSG),
    actions: actions
  });
};
const PacksTableEmptyState = /*#__PURE__*/_react.default.memo(PacksTableEmptyStateComponent);
exports.PacksTableEmptyState = PacksTableEmptyState;
