// @kbn/ml-plugin/common/constants/app.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PLUGIN_ID = exports.PLUGIN_ICON_SOLUTION = exports.PLUGIN_ICON = exports.ML_INTERNAL_BASE_PATH = exports.ML_EXTERNAL_BASE_PATH = exports.ML_APP_ROUTE = exports.ML_APP_NAME = void 0;
exports.initEnabledFeatures = initEnabledFeatures;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PLUGIN_ID = 'ml';
exports.PLUGIN_ID = PLUGIN_ID;
const PLUGIN_ICON = 'machineLearningApp';
exports.PLUGIN_ICON = PLUGIN_ICON;
const PLUGIN_ICON_SOLUTION = 'logoKibana';
exports.PLUGIN_ICON_SOLUTION = PLUGIN_ICON_SOLUTION;
const ML_APP_NAME = _i18n.i18n.translate('xpack.ml.navMenu.mlAppNameText', {
  defaultMessage: 'Machine Learning'
});
exports.ML_APP_NAME = ML_APP_NAME;
const ML_APP_ROUTE = '/app/ml';
exports.ML_APP_ROUTE = ML_APP_ROUTE;
const ML_INTERNAL_BASE_PATH = '/internal/ml';
exports.ML_INTERNAL_BASE_PATH = ML_INTERNAL_BASE_PATH;
const ML_EXTERNAL_BASE_PATH = '/api/ml';
exports.ML_EXTERNAL_BASE_PATH = ML_EXTERNAL_BASE_PATH;
function initEnabledFeatures(enabledFeatures, config) {
  var _config$ad, _config$dfa, _config$nlp;
  if (((_config$ad = config.ad) === null || _config$ad === void 0 ? void 0 : _config$ad.enabled) !== undefined) {
    enabledFeatures.ad = config.ad.enabled;
  }
  if (((_config$dfa = config.dfa) === null || _config$dfa === void 0 ? void 0 : _config$dfa.enabled) !== undefined) {
    enabledFeatures.dfa = config.dfa.enabled;
  }
  if (((_config$nlp = config.nlp) === null || _config$nlp === void 0 ? void 0 : _config$nlp.enabled) !== undefined) {
    enabledFeatures.nlp = config.nlp.enabled;
  }
}
