// @kbn/security-solution-plugin/public/common/store/inputs/actions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toggleTimelineLinkTo = exports.toggleSocTrendsLinkTo = exports.stopAutoReload = exports.startAutoReload = exports.setTimelineRangeDatePicker = exports.setSearchBarFilter = exports.setSavedQuery = exports.setRelativeRangeDatePicker = exports.setQuery = exports.setInspectionParameter = exports.setFullScreen = exports.setFilterQuery = exports.setDuration = exports.setAbsoluteRangeDatePicker = exports.removeLinkTo = exports.deleteOneQuery = exports.deleteAllQuery = exports.addLinkTo = void 0;
var _typescriptFsa = _interopRequireDefault(require("typescript-fsa"));
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const actionCreator = (0, _typescriptFsa.default)('x-pack/security_solution/local/inputs');
const setAbsoluteRangeDatePicker = actionCreator('SET_ABSOLUTE_RANGE_DATE_PICKER');
exports.setAbsoluteRangeDatePicker = setAbsoluteRangeDatePicker;
const setTimelineRangeDatePicker = actionCreator('SET_TIMELINE_RANGE_DATE_PICKER');
exports.setTimelineRangeDatePicker = setTimelineRangeDatePicker;
const setRelativeRangeDatePicker = actionCreator('SET_RELATIVE_RANGE_DATE_PICKER');
exports.setRelativeRangeDatePicker = setRelativeRangeDatePicker;
const setDuration = actionCreator('SET_DURATION');
exports.setDuration = setDuration;
const startAutoReload = actionCreator('START_KQL_AUTO_RELOAD');
exports.startAutoReload = startAutoReload;
const stopAutoReload = actionCreator('STOP_KQL_AUTO_RELOAD');
exports.stopAutoReload = stopAutoReload;
const setFullScreen = actionCreator('SET_FULL_SCREEN');
exports.setFullScreen = setFullScreen;
const setQuery = actionCreator('SET_QUERY');
exports.setQuery = setQuery;
const deleteOneQuery = actionCreator('DELETE_QUERY');
exports.deleteOneQuery = deleteOneQuery;
const setInspectionParameter = actionCreator('SET_INSPECTION_PARAMETER');
exports.setInspectionParameter = setInspectionParameter;
const deleteAllQuery = actionCreator('DELETE_ALL_QUERY');
exports.deleteAllQuery = deleteAllQuery;
const toggleTimelineLinkTo = actionCreator('TOGGLE_TIMELINE_LINK_TO');
exports.toggleTimelineLinkTo = toggleTimelineLinkTo;
const toggleSocTrendsLinkTo = actionCreator('TOGGLE_SOC_TRENDS_LINK_TO');
exports.toggleSocTrendsLinkTo = toggleSocTrendsLinkTo;
const removeLinkTo = actionCreator('REMOVE_LINK_TO');
exports.removeLinkTo = removeLinkTo;
const addLinkTo = actionCreator('ADD_LINK_TO');
exports.addLinkTo = addLinkTo;
const setFilterQuery = actionCreator('SET_FILTER_QUERY');
exports.setFilterQuery = setFilterQuery;
const setSavedQuery = actionCreator('SET_SAVED_QUERY');
exports.setSavedQuery = setSavedQuery;
const setSearchBarFilter = actionCreator('SET_SEARCH_BAR_FILTER');
exports.setSearchBarFilter = setSearchBarFilter;
