// @kbn/vis-type-timelion-plugin/public/timelion_options.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = TimelionOptions;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _public = require("@kbn/kibana-react-plugin/public");
var _components = require("./components");
require("./timelion_options.scss");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

function TimelionOptions({
  services,
  stateParams,
  setValue,
  setValidity
}) {
  const setInterval = (0, _react.useCallback)(value => setValue('interval', value), [setValue]);
  const setExpressionInput = (0, _react.useCallback)(value => setValue('expression', value), [setValue]);
  return /*#__PURE__*/_react.default.createElement(_public.KibanaContextProvider, {
    services: services
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiPanel, {
    className: "visEditorSidebar__timelionOptions",
    paddingSize: "s"
  }, /*#__PURE__*/_react.default.createElement(_components.TimelionInterval, {
    value: stateParams.interval,
    setValue: setInterval,
    setValidity: setValidity
  }), /*#__PURE__*/_react.default.createElement(_components.TimelionExpressionInput, {
    value: stateParams.expression,
    setValue: setExpressionInput
  })));
}

// default export required for React.Lazy
// eslint-disable-next-line import/no-default-export
module.exports = exports.default;
