// @kbn/es-ui-shared-plugin/static/forms/components/fields/file_picker_field.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilePickerField = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _hook_form_lib = require("../../hook_form_lib");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const ONE_MEGABYTE = 1048576;
const FilePickerField = ({
  field,
  euiFieldProps = {},
  idAria,
  maxFileSize = ONE_MEGABYTE,
  ...rest
}) => {
  const {
    isInvalid,
    errorMessage
  } = (0, _hook_form_lib.getFieldValidityAndErrorMessage)(field);
  const onChange = (0, _react.useCallback)(async files => {
    if (files) {
      const filesArr = [];
      try {
        for (const file of files) {
          if (file.size > maxFileSize) throw new Error(`${file.name} is too large, maximum size is ${Math.floor(maxFileSize) / 1024}kb`);
          const fileBuffer = await file.arrayBuffer();
          // Base64 encode the file
          const fileData = window.btoa(String.fromCharCode(...new Uint8Array(fileBuffer)));
          filesArr.push(fileData);
        }
        if (!euiFieldProps.multiple) field.setValue(filesArr[0]);else field.setValue(filesArr);
      } catch (e) {
        field.setErrors([e]);
      }
    } else {
      field.setValue(null);
    }
  }, [field, maxFileSize, euiFieldProps]);
  return /*#__PURE__*/_react.default.createElement(_eui.EuiFormRow, (0, _extends2.default)({
    label: field.label,
    labelAppend: field.labelAppend,
    helpText: typeof field.helpText === 'function' ? field.helpText() : field.helpText,
    error: errorMessage,
    isInvalid: isInvalid,
    fullWidth: true,
    describedByIds: idAria ? [idAria] : undefined
  }, rest), /*#__PURE__*/_react.default.createElement(_eui.EuiFilePicker, (0, _extends2.default)({
    isInvalid: isInvalid,
    onChange: onChange,
    isLoading: field.isValidating,
    "data-test-subj": "input"
  }, euiFieldProps)));
};
exports.FilePickerField = FilePickerField;
