// @kbn/enterprise-search-plugin/public/applications/shared/constants/labels.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.USERNAME_LABEL = exports.TYPE_LABEL = exports.PASSWORD_LABEL = exports.OPTIONAL_LABEL = exports.NATIVE_LABEL = exports.NAME_LABEL = exports.LEARN_MORE_LINK = exports.EMAIL_LABEL = exports.DESCRIPTION_LABEL = exports.CONNECTOR_CLIENT_LABEL = exports.BETA_LABEL = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const USERNAME_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.usernameLabel', {
  defaultMessage: 'Username'
});
exports.USERNAME_LABEL = USERNAME_LABEL;
const PASSWORD_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.passwordLabel', {
  defaultMessage: 'Password'
});
exports.PASSWORD_LABEL = PASSWORD_LABEL;
const TYPE_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.typeLabel', {
  defaultMessage: 'Type'
});
exports.TYPE_LABEL = TYPE_LABEL;
const EMAIL_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.emailLabel', {
  defaultMessage: 'Email'
});
exports.EMAIL_LABEL = EMAIL_LABEL;
const BETA_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.betaLabel', {
  defaultMessage: 'Beta'
});
exports.BETA_LABEL = BETA_LABEL;
const NATIVE_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.nativeLabel', {
  defaultMessage: 'Native'
});
exports.NATIVE_LABEL = NATIVE_LABEL;
const NAME_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.nameLabel', {
  defaultMessage: 'Name'
});
exports.NAME_LABEL = NAME_LABEL;
const DESCRIPTION_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.descriptionLabel', {
  defaultMessage: 'Description'
});
exports.DESCRIPTION_LABEL = DESCRIPTION_LABEL;
const OPTIONAL_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.optionalLabel', {
  defaultMessage: 'Optional'
});
exports.OPTIONAL_LABEL = OPTIONAL_LABEL;
const LEARN_MORE_LINK = _i18n.i18n.translate('xpack.enterpriseSearch.learnMore.link', {
  defaultMessage: 'Learn more'
});
exports.LEARN_MORE_LINK = LEARN_MORE_LINK;
const CONNECTOR_CLIENT_LABEL = _i18n.i18n.translate('xpack.enterpriseSearch.connectorClientLabel', {
  defaultMessage: 'Connector Client'
});
exports.CONNECTOR_CLIENT_LABEL = CONNECTOR_CLIENT_LABEL;
