// @kbn/triggers-actions-ui-plugin/public/application/sections/alert_summary_widget/components/constants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WIDGET_TITLE = exports.TOTAL_ALERT_COUNT_DATA_TEST_SUBJ = exports.TOOLTIP_DATE_FORMAT = exports.ALL_ALERT_COLOR = exports.ALERT_COUNT_FORMAT = exports.ALERTS_LABEL = exports.ACTIVE_NOW_LABEL = exports.ACTIVE_ALERT_COUNT_DATA_TEST_SUBJ = void 0;
var _eui = require("@elastic/eui");
var _i18nReact = require("@kbn/i18n-react");
var _react = _interopRequireDefault(require("react"));
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const TOOLTIP_DATE_FORMAT = 'YYYY-MM-DD HH:mm';
exports.TOOLTIP_DATE_FORMAT = TOOLTIP_DATE_FORMAT;
const ALERT_COUNT_FORMAT = '0.[00]a';
exports.ALERT_COUNT_FORMAT = ALERT_COUNT_FORMAT;
const ACTIVE_ALERT_COUNT_DATA_TEST_SUBJ = 'activeAlertCount';
exports.ACTIVE_ALERT_COUNT_DATA_TEST_SUBJ = ACTIVE_ALERT_COUNT_DATA_TEST_SUBJ;
const TOTAL_ALERT_COUNT_DATA_TEST_SUBJ = 'totalAlertCount';
exports.TOTAL_ALERT_COUNT_DATA_TEST_SUBJ = TOTAL_ALERT_COUNT_DATA_TEST_SUBJ;
const visColors = (0, _eui.euiPaletteColorBlind)();
const ALL_ALERT_COLOR = visColors[1];
exports.ALL_ALERT_COLOR = ALL_ALERT_COLOR;
const WIDGET_TITLE = /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
  id: "xpack.triggersActionsUI.sections.alertsSummaryWidget.title",
  defaultMessage: "Alert activity"
});
exports.WIDGET_TITLE = WIDGET_TITLE;
const ALERTS_LABEL = /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
  id: "xpack.triggersActionsUI.sections.alertsSummaryWidget.alerts",
  defaultMessage: "Alerts"
});
exports.ALERTS_LABEL = ALERTS_LABEL;
const ACTIVE_NOW_LABEL = /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
  id: "xpack.triggersActionsUI.sections.alertsSummaryWidget.activeNow",
  defaultMessage: "Active now"
});
exports.ACTIVE_NOW_LABEL = ACTIVE_NOW_LABEL;
