// @kbn/cloud-security-posture-plugin/public/pages/compliance_dashboard/test_subjects.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VULNERABILITY_DASHBOARD_PAGE_HEADER = exports.VULNERABILITY_DASHBOARD_CONTAINER = exports.MISSING_FINDINGS_NO_DATA_CONFIG = exports.KUBERNETES_DASHBOARD_TAB = exports.KUBERNETES_DASHBOARD_CONTAINER = exports.DASHBOARD_TABLE_HEADER_SCORE_TEST_ID = exports.DASHBOARD_TABLE_COLUMN_SCORE_TEST_ID = exports.DASHBOARD_SUMMARY_CONTAINER = exports.DASHBOARD_COUNTER_CARDS = exports.DASHBOARD_CONTAINER = exports.DASHBOARD_COMPLIANCE_SCORE_CHART = exports.CLOUD_POSTURE_DASHBOARD_PAGE_HEADER = exports.CLOUD_DASHBOARD_TAB = exports.CLOUD_DASHBOARD_CONTAINER = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const MISSING_FINDINGS_NO_DATA_CONFIG = 'missing-findings-no-data-config';
exports.MISSING_FINDINGS_NO_DATA_CONFIG = MISSING_FINDINGS_NO_DATA_CONFIG;
const DASHBOARD_CONTAINER = 'dashboard-container';
exports.DASHBOARD_CONTAINER = DASHBOARD_CONTAINER;
const DASHBOARD_SUMMARY_CONTAINER = 'dashboard-summary-section';
exports.DASHBOARD_SUMMARY_CONTAINER = DASHBOARD_SUMMARY_CONTAINER;
const KUBERNETES_DASHBOARD_CONTAINER = 'kubernetes-dashboard-container';
exports.KUBERNETES_DASHBOARD_CONTAINER = KUBERNETES_DASHBOARD_CONTAINER;
const CLOUD_DASHBOARD_CONTAINER = 'cloud-dashboard-container';
exports.CLOUD_DASHBOARD_CONTAINER = CLOUD_DASHBOARD_CONTAINER;
const CLOUD_POSTURE_DASHBOARD_PAGE_HEADER = 'cloud-posture-dashboard-page-header';
exports.CLOUD_POSTURE_DASHBOARD_PAGE_HEADER = CLOUD_POSTURE_DASHBOARD_PAGE_HEADER;
const VULNERABILITY_DASHBOARD_PAGE_HEADER = 'vulnerability-dashboard-page-header';
exports.VULNERABILITY_DASHBOARD_PAGE_HEADER = VULNERABILITY_DASHBOARD_PAGE_HEADER;
const VULNERABILITY_DASHBOARD_CONTAINER = 'vulnerability-dashboard-container';
exports.VULNERABILITY_DASHBOARD_CONTAINER = VULNERABILITY_DASHBOARD_CONTAINER;
const DASHBOARD_COUNTER_CARDS = {
  CLUSTERS_EVALUATED: 'dashboard-counter-card-clusters-evaluated',
  RESOURCES_EVALUATED: 'dashboard-counter-card-resources-evaluated',
  FAILING_FINDINGS: 'dashboard-counter-card-failing-findings'
};
exports.DASHBOARD_COUNTER_CARDS = DASHBOARD_COUNTER_CARDS;
const DASHBOARD_TABLE_HEADER_SCORE_TEST_ID = 'csp:dashboard-sections-table-header-score';
exports.DASHBOARD_TABLE_HEADER_SCORE_TEST_ID = DASHBOARD_TABLE_HEADER_SCORE_TEST_ID;
const DASHBOARD_TABLE_COLUMN_SCORE_TEST_ID = 'csp:dashboard-sections-table-column-score';
exports.DASHBOARD_TABLE_COLUMN_SCORE_TEST_ID = DASHBOARD_TABLE_COLUMN_SCORE_TEST_ID;
const KUBERNETES_DASHBOARD_TAB = 'kubernetes-dashboard-tab';
exports.KUBERNETES_DASHBOARD_TAB = KUBERNETES_DASHBOARD_TAB;
const CLOUD_DASHBOARD_TAB = 'cloud-dashboard-tab';
exports.CLOUD_DASHBOARD_TAB = CLOUD_DASHBOARD_TAB;
const DASHBOARD_COMPLIANCE_SCORE_CHART = {
  COMPLIANCE_SCORE: 'dashboard-summary-section-compliance-score'
};
exports.DASHBOARD_COMPLIANCE_SCORE_CHART = DASHBOARD_COMPLIANCE_SCORE_CHART;
