// @kbn/stack-connectors-plugin/public/connector_types/cases_webhook/action_variables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.urlVarsExt = exports.urlVars = exports.commentVars = exports.casesVars = void 0;
var i18n = _interopRequireWildcard(require("./translations"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const casesVars = [{
  name: 'case.title',
  description: i18n.CASE_TITLE_DESC,
  useWithTripleBracesInTemplates: true
}, {
  name: 'case.description',
  description: i18n.CASE_DESCRIPTION_DESC,
  useWithTripleBracesInTemplates: true
}, {
  name: 'case.tags',
  description: i18n.CASE_TAGS_DESC,
  useWithTripleBracesInTemplates: true
}, {
  name: 'case.id',
  description: i18n.CASE_ID_DESC,
  useWithTripleBracesInTemplates: true
}, {
  name: 'case.status',
  description: i18n.CASE_SEVERITY_DESC,
  useWithTripleBracesInTemplates: true
}, {
  name: 'case.severity',
  description: i18n.CASE_STATUS_DESC,
  useWithTripleBracesInTemplates: true
}];
exports.casesVars = casesVars;
const commentVars = [{
  name: 'case.comment',
  description: i18n.CASE_COMMENT_DESC,
  useWithTripleBracesInTemplates: true
}];
exports.commentVars = commentVars;
const urlVars = [{
  name: 'external.system.id',
  description: i18n.EXTERNAL_ID_DESC,
  useWithTripleBracesInTemplates: true
}];
exports.urlVars = urlVars;
const urlVarsExt = [...urlVars, {
  name: 'external.system.title',
  description: i18n.EXTERNAL_TITLE_DESC,
  useWithTripleBracesInTemplates: true
}];
exports.urlVarsExt = urlVarsExt;
