// @kbn/security-solution-plugin/public/flyout/document_details/right/components/test_ids.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VISUALIZATIONS_TEST_ID = exports.VISUALIZATIONS_SECTION_HEADER_TEST_ID = exports.SUMMARY_ROW_VALUE_TEST_ID = exports.SUMMARY_ROW_LOADING_TEST_ID = exports.SUMMARY_ROW_ICON_TEST_ID = exports.STATUS_TITLE_TEST_ID = exports.STATUS_BUTTON_TEST_ID = exports.SHARE_BUTTON_TEST_ID = exports.SEVERITY_VALUE_TEST_ID = exports.SESSION_PREVIEW_TEST_ID = exports.RULE_SUMMARY_BUTTON_TEST_ID = exports.RISK_SCORE_VALUE_TEST_ID = exports.RISK_SCORE_TITLE_TEST_ID = exports.RESPONSE_SECTION_TEST_ID = exports.RESPONSE_SECTION_HEADER_TEST_ID = exports.RESPONSE_SECTION_CONTENT_TEST_ID = exports.RESPONSE_EMPTY_TEST_ID = exports.RESPONSE_BUTTON_TEST_ID = exports.REASON_TITLE_TEST_ID = exports.REASON_DETAILS_TEST_ID = exports.REASON_DETAILS_PREVIEW_BUTTON_TEST_ID = exports.PREVALENCE_TEST_ID = exports.MITRE_ATTACK_TITLE_TEST_ID = exports.MITRE_ATTACK_DETAILS_TEST_ID = exports.INVESTIGATION_SECTION_TEST_ID = exports.INVESTIGATION_SECTION_HEADER_TEST_ID = exports.INVESTIGATION_SECTION_CONTENT_TEST_ID = exports.INVESTIGATION_GUIDE_TEST_ID = exports.INVESTIGATION_GUIDE_LOADING_TEST_ID = exports.INVESTIGATION_GUIDE_BUTTON_TEST_ID = exports.INSIGHTS_THREAT_INTELLIGENCE_TEST_ID = exports.INSIGHTS_TEST_ID = exports.INSIGHTS_HEADER_TEST_ID = exports.INSIGHTS_ENTITIES_TEST_ID = exports.HIGHLIGHTED_FIELDS_TITLE_TEST_ID = exports.HIGHLIGHTED_FIELDS_LINKED_CELL_TEST_ID = exports.HIGHLIGHTED_FIELDS_DETAILS_TEST_ID = exports.HIGHLIGHTED_FIELDS_CELL_TEST_ID = exports.HIGHLIGHTED_FIELDS_BASIC_CELL_TEST_ID = exports.HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID = exports.FLYOUT_HEADER_TITLE_TEST_ID = exports.ENTITIES_USER_OVERVIEW_TEST_ID = exports.ENTITIES_USER_OVERVIEW_RISK_LEVEL_TEST_ID = exports.ENTITIES_USER_OVERVIEW_LOADING_TEST_ID = exports.ENTITIES_USER_OVERVIEW_LINK_TEST_ID = exports.ENTITIES_USER_OVERVIEW_LAST_SEEN_TEST_ID = exports.ENTITIES_USER_OVERVIEW_DOMAIN_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_RISK_LEVEL_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_OS_FAMILY_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_LOADING_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_LINK_TEST_ID = exports.ENTITIES_HOST_OVERVIEW_LAST_SEEN_TEST_ID = exports.DESCRIPTION_TITLE_TEST_ID = exports.DESCRIPTION_DETAILS_TEST_ID = exports.CORRELATIONS_TEST_ID = exports.CORRELATIONS_SUPPRESSED_ALERTS_TEST_ID = exports.CORRELATIONS_SUPPRESSED_ALERTS_TECHNICAL_PREVIEW_TEST_ID = exports.CORRELATIONS_RELATED_CASES_TEST_ID = exports.CORRELATIONS_RELATED_ALERTS_BY_SESSION_TEST_ID = exports.CORRELATIONS_RELATED_ALERTS_BY_SAME_SOURCE_EVENT_TEST_ID = exports.CORRELATIONS_RELATED_ALERTS_BY_ANCESTRY_TEST_ID = exports.CHAT_BUTTON_TEST_ID = exports.ASSIGNEES_TITLE_TEST_ID = exports.ASSIGNEES_HEADER_TEST_ID = exports.ASSIGNEES_ADD_BUTTON_TEST_ID = exports.ANALYZER_PREVIEW_TEST_ID = exports.ANALYZER_PREVIEW_LOADING_TEST_ID = exports.ABOUT_SECTION_TEST_ID = exports.ABOUT_SECTION_HEADER_TEST_ID = exports.ABOUT_SECTION_CONTENT_TEST_ID = void 0;
var _test_ids = require("../../../shared/test_ids");
var _expandable_section = require("./expandable_section");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

/* Header */

const FLYOUT_HEADER_TEST_ID = `${_test_ids.PREFIX}Header`;
const FLYOUT_HEADER_TITLE_TEST_ID = `${FLYOUT_HEADER_TEST_ID}Title`;
exports.FLYOUT_HEADER_TITLE_TEST_ID = FLYOUT_HEADER_TITLE_TEST_ID;
const STATUS_TITLE_TEST_ID = `${FLYOUT_HEADER_TEST_ID}StatusTitle`;
exports.STATUS_TITLE_TEST_ID = STATUS_TITLE_TEST_ID;
const STATUS_BUTTON_TEST_ID = 'rule-status-badge';
exports.STATUS_BUTTON_TEST_ID = STATUS_BUTTON_TEST_ID;
const SEVERITY_VALUE_TEST_ID = 'severity';
exports.SEVERITY_VALUE_TEST_ID = SEVERITY_VALUE_TEST_ID;
const RISK_SCORE_TITLE_TEST_ID = `${FLYOUT_HEADER_TEST_ID}RiskScoreTitle`;
exports.RISK_SCORE_TITLE_TEST_ID = RISK_SCORE_TITLE_TEST_ID;
const RISK_SCORE_VALUE_TEST_ID = `${FLYOUT_HEADER_TEST_ID}RiskScoreValue`;
exports.RISK_SCORE_VALUE_TEST_ID = RISK_SCORE_VALUE_TEST_ID;
const SHARE_BUTTON_TEST_ID = `${FLYOUT_HEADER_TEST_ID}ShareButton`;
exports.SHARE_BUTTON_TEST_ID = SHARE_BUTTON_TEST_ID;
const CHAT_BUTTON_TEST_ID = 'newChatById';
exports.CHAT_BUTTON_TEST_ID = CHAT_BUTTON_TEST_ID;
const ASSIGNEES_HEADER_TEST_ID = `${FLYOUT_HEADER_TEST_ID}AssigneesHeader`;
exports.ASSIGNEES_HEADER_TEST_ID = ASSIGNEES_HEADER_TEST_ID;
const ASSIGNEES_TITLE_TEST_ID = `${FLYOUT_HEADER_TEST_ID}AssigneesTitle`;
exports.ASSIGNEES_TITLE_TEST_ID = ASSIGNEES_TITLE_TEST_ID;
const ASSIGNEES_ADD_BUTTON_TEST_ID = `${FLYOUT_HEADER_TEST_ID}AssigneesAddButton`;

/* About section */
exports.ASSIGNEES_ADD_BUTTON_TEST_ID = ASSIGNEES_ADD_BUTTON_TEST_ID;
const ABOUT_SECTION_TEST_ID = `${_test_ids.PREFIX}AboutSection`;
exports.ABOUT_SECTION_TEST_ID = ABOUT_SECTION_TEST_ID;
const ABOUT_SECTION_HEADER_TEST_ID = ABOUT_SECTION_TEST_ID + _expandable_section.HEADER_TEST_ID;
exports.ABOUT_SECTION_HEADER_TEST_ID = ABOUT_SECTION_HEADER_TEST_ID;
const ABOUT_SECTION_CONTENT_TEST_ID = ABOUT_SECTION_TEST_ID + _expandable_section.CONTENT_TEST_ID;
exports.ABOUT_SECTION_CONTENT_TEST_ID = ABOUT_SECTION_CONTENT_TEST_ID;
const RULE_SUMMARY_BUTTON_TEST_ID = `${_test_ids.PREFIX}RuleSummaryButton`;
exports.RULE_SUMMARY_BUTTON_TEST_ID = RULE_SUMMARY_BUTTON_TEST_ID;
const DESCRIPTION_TEST_ID = `${_test_ids.PREFIX}Description`;
const DESCRIPTION_TITLE_TEST_ID = `${DESCRIPTION_TEST_ID}Title`;
exports.DESCRIPTION_TITLE_TEST_ID = DESCRIPTION_TITLE_TEST_ID;
const DESCRIPTION_DETAILS_TEST_ID = `${DESCRIPTION_TEST_ID}Details`;
exports.DESCRIPTION_DETAILS_TEST_ID = DESCRIPTION_DETAILS_TEST_ID;
const REASON_TEST_ID = `${_test_ids.PREFIX}Reason`;
const REASON_TITLE_TEST_ID = `${REASON_TEST_ID}Title`;
exports.REASON_TITLE_TEST_ID = REASON_TITLE_TEST_ID;
const REASON_DETAILS_TEST_ID = `${REASON_TEST_ID}Details`;
exports.REASON_DETAILS_TEST_ID = REASON_DETAILS_TEST_ID;
const REASON_DETAILS_PREVIEW_BUTTON_TEST_ID = `${REASON_TEST_ID}PreviewButton`;
exports.REASON_DETAILS_PREVIEW_BUTTON_TEST_ID = REASON_DETAILS_PREVIEW_BUTTON_TEST_ID;
const MITRE_ATTACK_TEST_ID = `${_test_ids.PREFIX}MitreAttack`;
const MITRE_ATTACK_TITLE_TEST_ID = `${MITRE_ATTACK_TEST_ID}Title`;
exports.MITRE_ATTACK_TITLE_TEST_ID = MITRE_ATTACK_TITLE_TEST_ID;
const MITRE_ATTACK_DETAILS_TEST_ID = `${MITRE_ATTACK_TEST_ID}Details`;

/* Investigation section */
exports.MITRE_ATTACK_DETAILS_TEST_ID = MITRE_ATTACK_DETAILS_TEST_ID;
const INVESTIGATION_SECTION_TEST_ID = `${_test_ids.PREFIX}InvestigationSection`;
exports.INVESTIGATION_SECTION_TEST_ID = INVESTIGATION_SECTION_TEST_ID;
const INVESTIGATION_SECTION_HEADER_TEST_ID = INVESTIGATION_SECTION_TEST_ID + _expandable_section.HEADER_TEST_ID;
exports.INVESTIGATION_SECTION_HEADER_TEST_ID = INVESTIGATION_SECTION_HEADER_TEST_ID;
const INVESTIGATION_SECTION_CONTENT_TEST_ID = INVESTIGATION_SECTION_TEST_ID + _expandable_section.CONTENT_TEST_ID;
exports.INVESTIGATION_SECTION_CONTENT_TEST_ID = INVESTIGATION_SECTION_CONTENT_TEST_ID;
const INVESTIGATION_GUIDE_TEST_ID = `${_test_ids.PREFIX}InvestigationGuide`;
exports.INVESTIGATION_GUIDE_TEST_ID = INVESTIGATION_GUIDE_TEST_ID;
const INVESTIGATION_GUIDE_BUTTON_TEST_ID = `${INVESTIGATION_GUIDE_TEST_ID}Button`;
exports.INVESTIGATION_GUIDE_BUTTON_TEST_ID = INVESTIGATION_GUIDE_BUTTON_TEST_ID;
const INVESTIGATION_GUIDE_LOADING_TEST_ID = `${INVESTIGATION_GUIDE_TEST_ID}Loading`;
exports.INVESTIGATION_GUIDE_LOADING_TEST_ID = INVESTIGATION_GUIDE_LOADING_TEST_ID;
const HIGHLIGHTED_FIELDS_TEST_ID = `${_test_ids.PREFIX}HighlightedFields`;
const HIGHLIGHTED_FIELDS_TITLE_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}Title`;
exports.HIGHLIGHTED_FIELDS_TITLE_TEST_ID = HIGHLIGHTED_FIELDS_TITLE_TEST_ID;
const HIGHLIGHTED_FIELDS_DETAILS_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}Details`;
exports.HIGHLIGHTED_FIELDS_DETAILS_TEST_ID = HIGHLIGHTED_FIELDS_DETAILS_TEST_ID;
const HIGHLIGHTED_FIELDS_CELL_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}Cell`;
exports.HIGHLIGHTED_FIELDS_CELL_TEST_ID = HIGHLIGHTED_FIELDS_CELL_TEST_ID;
const HIGHLIGHTED_FIELDS_BASIC_CELL_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}BasicCell`;
exports.HIGHLIGHTED_FIELDS_BASIC_CELL_TEST_ID = HIGHLIGHTED_FIELDS_BASIC_CELL_TEST_ID;
const HIGHLIGHTED_FIELDS_LINKED_CELL_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}LinkedCell`;
exports.HIGHLIGHTED_FIELDS_LINKED_CELL_TEST_ID = HIGHLIGHTED_FIELDS_LINKED_CELL_TEST_ID;
const HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID = `${HIGHLIGHTED_FIELDS_TEST_ID}AgentStatusCell`;

/* Insights section */
exports.HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID = HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID;
const INSIGHTS_TEST_ID = `${_test_ids.PREFIX}Insights`;
exports.INSIGHTS_TEST_ID = INSIGHTS_TEST_ID;
const INSIGHTS_HEADER_TEST_ID = `${INSIGHTS_TEST_ID}Header`;

/* Summary row */
exports.INSIGHTS_HEADER_TEST_ID = INSIGHTS_HEADER_TEST_ID;
const SUMMARY_ROW_LOADING_TEST_ID = dataTestSubj => `${dataTestSubj}Loading`;
exports.SUMMARY_ROW_LOADING_TEST_ID = SUMMARY_ROW_LOADING_TEST_ID;
const SUMMARY_ROW_ICON_TEST_ID = dataTestSubj => `${dataTestSubj}Icon`;
exports.SUMMARY_ROW_ICON_TEST_ID = SUMMARY_ROW_ICON_TEST_ID;
const SUMMARY_ROW_VALUE_TEST_ID = dataTestSubj => `${dataTestSubj}Value`;

/* Entities */
exports.SUMMARY_ROW_VALUE_TEST_ID = SUMMARY_ROW_VALUE_TEST_ID;
const INSIGHTS_ENTITIES_TEST_ID = `${_test_ids.PREFIX}InsightsEntities`;
exports.INSIGHTS_ENTITIES_TEST_ID = INSIGHTS_ENTITIES_TEST_ID;
const ENTITIES_USER_OVERVIEW_TEST_ID = `${INSIGHTS_ENTITIES_TEST_ID}UserOverview`;
exports.ENTITIES_USER_OVERVIEW_TEST_ID = ENTITIES_USER_OVERVIEW_TEST_ID;
const ENTITIES_USER_OVERVIEW_LOADING_TEST_ID = `${ENTITIES_USER_OVERVIEW_TEST_ID}Loading`;
exports.ENTITIES_USER_OVERVIEW_LOADING_TEST_ID = ENTITIES_USER_OVERVIEW_LOADING_TEST_ID;
const ENTITIES_USER_OVERVIEW_LINK_TEST_ID = `${ENTITIES_USER_OVERVIEW_TEST_ID}Link`;
exports.ENTITIES_USER_OVERVIEW_LINK_TEST_ID = ENTITIES_USER_OVERVIEW_LINK_TEST_ID;
const ENTITIES_USER_OVERVIEW_DOMAIN_TEST_ID = `${ENTITIES_USER_OVERVIEW_TEST_ID}Domain`;
exports.ENTITIES_USER_OVERVIEW_DOMAIN_TEST_ID = ENTITIES_USER_OVERVIEW_DOMAIN_TEST_ID;
const ENTITIES_USER_OVERVIEW_LAST_SEEN_TEST_ID = `${ENTITIES_USER_OVERVIEW_TEST_ID}LastSeen`;
exports.ENTITIES_USER_OVERVIEW_LAST_SEEN_TEST_ID = ENTITIES_USER_OVERVIEW_LAST_SEEN_TEST_ID;
const ENTITIES_USER_OVERVIEW_RISK_LEVEL_TEST_ID = `${ENTITIES_USER_OVERVIEW_TEST_ID}RiskLevel`;
exports.ENTITIES_USER_OVERVIEW_RISK_LEVEL_TEST_ID = ENTITIES_USER_OVERVIEW_RISK_LEVEL_TEST_ID;
const ENTITIES_HOST_OVERVIEW_TEST_ID = `${INSIGHTS_ENTITIES_TEST_ID}HostOverview`;
exports.ENTITIES_HOST_OVERVIEW_TEST_ID = ENTITIES_HOST_OVERVIEW_TEST_ID;
const ENTITIES_HOST_OVERVIEW_LOADING_TEST_ID = `${ENTITIES_HOST_OVERVIEW_TEST_ID}Loading`;
exports.ENTITIES_HOST_OVERVIEW_LOADING_TEST_ID = ENTITIES_HOST_OVERVIEW_LOADING_TEST_ID;
const ENTITIES_HOST_OVERVIEW_LINK_TEST_ID = `${ENTITIES_HOST_OVERVIEW_TEST_ID}Link`;
exports.ENTITIES_HOST_OVERVIEW_LINK_TEST_ID = ENTITIES_HOST_OVERVIEW_LINK_TEST_ID;
const ENTITIES_HOST_OVERVIEW_OS_FAMILY_TEST_ID = `${ENTITIES_HOST_OVERVIEW_TEST_ID}OsFamily`;
exports.ENTITIES_HOST_OVERVIEW_OS_FAMILY_TEST_ID = ENTITIES_HOST_OVERVIEW_OS_FAMILY_TEST_ID;
const ENTITIES_HOST_OVERVIEW_LAST_SEEN_TEST_ID = `${ENTITIES_HOST_OVERVIEW_TEST_ID}LastSeen`;
exports.ENTITIES_HOST_OVERVIEW_LAST_SEEN_TEST_ID = ENTITIES_HOST_OVERVIEW_LAST_SEEN_TEST_ID;
const ENTITIES_HOST_OVERVIEW_RISK_LEVEL_TEST_ID = `${ENTITIES_HOST_OVERVIEW_TEST_ID}RiskLevel`;

/* Threat intelligence */
exports.ENTITIES_HOST_OVERVIEW_RISK_LEVEL_TEST_ID = ENTITIES_HOST_OVERVIEW_RISK_LEVEL_TEST_ID;
const INSIGHTS_THREAT_INTELLIGENCE_TEST_ID = `${_test_ids.PREFIX}InsightsThreatIntelligence`;

/* Correlations */
exports.INSIGHTS_THREAT_INTELLIGENCE_TEST_ID = INSIGHTS_THREAT_INTELLIGENCE_TEST_ID;
const CORRELATIONS_TEST_ID = `${_test_ids.PREFIX}Correlations`;
exports.CORRELATIONS_TEST_ID = CORRELATIONS_TEST_ID;
const CORRELATIONS_SUPPRESSED_ALERTS_TEST_ID = `${CORRELATIONS_TEST_ID}SuppressedAlerts`;
exports.CORRELATIONS_SUPPRESSED_ALERTS_TEST_ID = CORRELATIONS_SUPPRESSED_ALERTS_TEST_ID;
const CORRELATIONS_SUPPRESSED_ALERTS_TECHNICAL_PREVIEW_TEST_ID = `${CORRELATIONS_SUPPRESSED_ALERTS_TEST_ID}TechnicalPreview`;
exports.CORRELATIONS_SUPPRESSED_ALERTS_TECHNICAL_PREVIEW_TEST_ID = CORRELATIONS_SUPPRESSED_ALERTS_TECHNICAL_PREVIEW_TEST_ID;
const CORRELATIONS_RELATED_CASES_TEST_ID = `${CORRELATIONS_TEST_ID}RelatedCases`;
exports.CORRELATIONS_RELATED_CASES_TEST_ID = CORRELATIONS_RELATED_CASES_TEST_ID;
const CORRELATIONS_RELATED_ALERTS_BY_SESSION_TEST_ID = `${CORRELATIONS_TEST_ID}RelatedAlertsBySession`;
exports.CORRELATIONS_RELATED_ALERTS_BY_SESSION_TEST_ID = CORRELATIONS_RELATED_ALERTS_BY_SESSION_TEST_ID;
const CORRELATIONS_RELATED_ALERTS_BY_SAME_SOURCE_EVENT_TEST_ID = `${CORRELATIONS_TEST_ID}RelatedAlertsBySameSourceEvent`;
exports.CORRELATIONS_RELATED_ALERTS_BY_SAME_SOURCE_EVENT_TEST_ID = CORRELATIONS_RELATED_ALERTS_BY_SAME_SOURCE_EVENT_TEST_ID;
const CORRELATIONS_RELATED_ALERTS_BY_ANCESTRY_TEST_ID = `${CORRELATIONS_TEST_ID}RelatedAlertsByAncestry`;

/* Insights Prevalence */
exports.CORRELATIONS_RELATED_ALERTS_BY_ANCESTRY_TEST_ID = CORRELATIONS_RELATED_ALERTS_BY_ANCESTRY_TEST_ID;
const PREVALENCE_TEST_ID = `${_test_ids.PREFIX}InsightsPrevalence`;

/* Visualizations section */
exports.PREVALENCE_TEST_ID = PREVALENCE_TEST_ID;
const VISUALIZATIONS_TEST_ID = `${_test_ids.PREFIX}Visualizations`;
exports.VISUALIZATIONS_TEST_ID = VISUALIZATIONS_TEST_ID;
const VISUALIZATIONS_SECTION_HEADER_TEST_ID = `${VISUALIZATIONS_TEST_ID}Header`;
exports.VISUALIZATIONS_SECTION_HEADER_TEST_ID = VISUALIZATIONS_SECTION_HEADER_TEST_ID;
const ANALYZER_PREVIEW_TEST_ID = `${_test_ids.PREFIX}AnalyzerPreview`;
exports.ANALYZER_PREVIEW_TEST_ID = ANALYZER_PREVIEW_TEST_ID;
const ANALYZER_PREVIEW_LOADING_TEST_ID = `${ANALYZER_PREVIEW_TEST_ID}Loading`;
exports.ANALYZER_PREVIEW_LOADING_TEST_ID = ANALYZER_PREVIEW_LOADING_TEST_ID;
const SESSION_PREVIEW_TEST_ID = `${_test_ids.PREFIX}SessionPreview`;

/* Response section */
exports.SESSION_PREVIEW_TEST_ID = SESSION_PREVIEW_TEST_ID;
const RESPONSE_TEST_ID = `${_test_ids.PREFIX}Response`;
const RESPONSE_SECTION_TEST_ID = `${RESPONSE_TEST_ID}Section`;
exports.RESPONSE_SECTION_TEST_ID = RESPONSE_SECTION_TEST_ID;
const RESPONSE_SECTION_HEADER_TEST_ID = RESPONSE_SECTION_TEST_ID + _expandable_section.HEADER_TEST_ID;
exports.RESPONSE_SECTION_HEADER_TEST_ID = RESPONSE_SECTION_HEADER_TEST_ID;
const RESPONSE_SECTION_CONTENT_TEST_ID = RESPONSE_SECTION_TEST_ID + _expandable_section.CONTENT_TEST_ID;
exports.RESPONSE_SECTION_CONTENT_TEST_ID = RESPONSE_SECTION_CONTENT_TEST_ID;
const RESPONSE_BUTTON_TEST_ID = `${RESPONSE_TEST_ID}Button`;
exports.RESPONSE_BUTTON_TEST_ID = RESPONSE_BUTTON_TEST_ID;
const RESPONSE_EMPTY_TEST_ID = `${RESPONSE_TEST_ID}Empty`;
exports.RESPONSE_EMPTY_TEST_ID = RESPONSE_EMPTY_TEST_ID;
