// @kbn/ml-plugin/public/register_helper/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "registerCasesAttachments", {
  enumerable: true,
  get: function () {
    return _cases.registerCasesAttachments;
  }
});
Object.defineProperty(exports, "registerEmbeddables", {
  enumerable: true,
  get: function () {
    return _embeddables.registerEmbeddables;
  }
});
Object.defineProperty(exports, "registerManagementSection", {
  enumerable: true,
  get: function () {
    return _management.registerManagementSection;
  }
});
Object.defineProperty(exports, "registerMapExtension", {
  enumerable: true,
  get: function () {
    return _register_map_extension.registerMapExtension;
  }
});
Object.defineProperty(exports, "registerMlAlerts", {
  enumerable: true,
  get: function () {
    return _alerting.registerMlAlerts;
  }
});
Object.defineProperty(exports, "registerMlUiActions", {
  enumerable: true,
  get: function () {
    return _ui_actions.registerMlUiActions;
  }
});
Object.defineProperty(exports, "registerSearchLinks", {
  enumerable: true,
  get: function () {
    return _register_search_links.registerSearchLinks;
  }
});
var _embeddables = require("../embeddables");
var _management = require("../application/management");
var _ui_actions = require("../ui_actions");
var _register_search_links = require("./register_search_links");
var _alerting = require("../alerting");
var _register_map_extension = require("../maps/register_map_extension");
var _cases = require("../cases");
