// @kbn/monaco/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  monaco: true,
  XJsonLang: true,
  SQLLang: true,
  ESQL_LANG_ID: true,
  ESQL_THEME_ID: true,
  ESQLLang: true,
  BarePluginApi: true,
  registerLanguage: true
};
exports.BarePluginApi = void 0;
Object.defineProperty(exports, "ESQLLang", {
  enumerable: true,
  get: function () {
    return _esql.ESQLLang;
  }
});
Object.defineProperty(exports, "ESQL_LANG_ID", {
  enumerable: true,
  get: function () {
    return _esql.ESQL_LANG_ID;
  }
});
Object.defineProperty(exports, "ESQL_THEME_ID", {
  enumerable: true,
  get: function () {
    return _esql.ESQL_THEME_ID;
  }
});
Object.defineProperty(exports, "SQLLang", {
  enumerable: true,
  get: function () {
    return _sql.SQLLang;
  }
});
Object.defineProperty(exports, "XJsonLang", {
  enumerable: true,
  get: function () {
    return _xjson.XJsonLang;
  }
});
Object.defineProperty(exports, "monaco", {
  enumerable: true,
  get: function () {
    return _monaco_imports.monaco;
  }
});
Object.defineProperty(exports, "registerLanguage", {
  enumerable: true,
  get: function () {
    return _helpers.registerLanguage;
  }
});
require("./src/register_globals");
var _monaco_imports = require("./src/monaco_imports");
var _xjson = require("./src/xjson");
var _sql = require("./src/sql");
var _esql = require("./src/esql");
var _painless = require("./src/painless");
Object.keys(_painless).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _painless[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _painless[key];
    }
  });
});
var BarePluginApi = _interopRequireWildcard(require("monaco-editor/esm/vs/editor/editor.api"));
exports.BarePluginApi = BarePluginApi;
var _helpers = require("./src/helpers");
var _types = require("./src/types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
