// @kbn/security-solution-serverless/public/get_started/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ViewDashboardSteps = exports.ViewAlertsSteps = exports.SectionId = exports.QuickStartSectionCardsId = exports.OverviewSteps = exports.GetStartedWithAlertsCardsId = exports.GetStartedPageActions = exports.EnablePrebuiltRulesSteps = exports.CreateProjectSteps = exports.BadgeId = exports.AddIntegrationsSteps = exports.AddAndValidateYourDataCardsId = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
let SectionId;
exports.SectionId = SectionId;
(function (SectionId) {
  SectionId["quickStart"] = "quick_start";
  SectionId["addAndValidateYourData"] = "add_and_validate_your_data";
  SectionId["getStartedWithAlerts"] = "get_started_with_alerts";
})(SectionId || (exports.SectionId = SectionId = {}));
let QuickStartSectionCardsId;
exports.QuickStartSectionCardsId = QuickStartSectionCardsId;
(function (QuickStartSectionCardsId) {
  QuickStartSectionCardsId["createFirstProject"] = "create_first_project";
  QuickStartSectionCardsId["watchTheOverviewVideo"] = "watch_the_overview_video";
})(QuickStartSectionCardsId || (exports.QuickStartSectionCardsId = QuickStartSectionCardsId = {}));
let AddAndValidateYourDataCardsId;
exports.AddAndValidateYourDataCardsId = AddAndValidateYourDataCardsId;
(function (AddAndValidateYourDataCardsId) {
  AddAndValidateYourDataCardsId["addIntegrations"] = "add_integrations";
  AddAndValidateYourDataCardsId["viewDashboards"] = "view_dashboards";
})(AddAndValidateYourDataCardsId || (exports.AddAndValidateYourDataCardsId = AddAndValidateYourDataCardsId = {}));
let GetStartedWithAlertsCardsId;
exports.GetStartedWithAlertsCardsId = GetStartedWithAlertsCardsId;
(function (GetStartedWithAlertsCardsId) {
  GetStartedWithAlertsCardsId["enablePrebuiltRules"] = "enable_prebuilt_rules";
  GetStartedWithAlertsCardsId["viewAlerts"] = "view_alerts";
})(GetStartedWithAlertsCardsId || (exports.GetStartedWithAlertsCardsId = GetStartedWithAlertsCardsId = {}));
let BadgeId;
exports.BadgeId = BadgeId;
(function (BadgeId) {
  BadgeId["analytics"] = "analytics";
  BadgeId["cloud"] = "cloud";
  BadgeId["edr"] = "edr";
})(BadgeId || (exports.BadgeId = BadgeId = {}));
let CreateProjectSteps;
exports.CreateProjectSteps = CreateProjectSteps;
(function (CreateProjectSteps) {
  CreateProjectSteps["createFirstProject"] = "create_your_first_project";
})(CreateProjectSteps || (exports.CreateProjectSteps = CreateProjectSteps = {}));
let OverviewSteps;
exports.OverviewSteps = OverviewSteps;
(function (OverviewSteps) {
  OverviewSteps["getToKnowElasticSecurity"] = "watch_the_overview_video";
})(OverviewSteps || (exports.OverviewSteps = OverviewSteps = {}));
let AddIntegrationsSteps;
exports.AddIntegrationsSteps = AddIntegrationsSteps;
(function (AddIntegrationsSteps) {
  AddIntegrationsSteps["connectToDataSources"] = "add_integrations";
})(AddIntegrationsSteps || (exports.AddIntegrationsSteps = AddIntegrationsSteps = {}));
let ViewDashboardSteps;
exports.ViewDashboardSteps = ViewDashboardSteps;
(function (ViewDashboardSteps) {
  ViewDashboardSteps["analyzeData"] = "view_and_analyze_your_data";
})(ViewDashboardSteps || (exports.ViewDashboardSteps = ViewDashboardSteps = {}));
let EnablePrebuiltRulesSteps;
exports.EnablePrebuiltRulesSteps = EnablePrebuiltRulesSteps;
(function (EnablePrebuiltRulesSteps) {
  EnablePrebuiltRulesSteps["enablePrebuiltRules"] = "enable_prebuilt_rules";
})(EnablePrebuiltRulesSteps || (exports.EnablePrebuiltRulesSteps = EnablePrebuiltRulesSteps = {}));
let ViewAlertsSteps;
exports.ViewAlertsSteps = ViewAlertsSteps;
(function (ViewAlertsSteps) {
  ViewAlertsSteps["viewAlerts"] = "view_alerts";
})(ViewAlertsSteps || (exports.ViewAlertsSteps = ViewAlertsSteps = {}));
let GetStartedPageActions;
exports.GetStartedPageActions = GetStartedPageActions;
(function (GetStartedPageActions) {
  GetStartedPageActions["AddFinishedStep"] = "addFinishedStep";
  GetStartedPageActions["RemoveFinishedStep"] = "removeFinishedStep";
  GetStartedPageActions["ToggleProduct"] = "toggleProduct";
  GetStartedPageActions["ToggleExpandedStep"] = "toggleExpandedStep";
})(GetStartedPageActions || (exports.GetStartedPageActions = GetStartedPageActions = {}));
