// @kbn/security-solution-plugin/public/detections/components/rules/anomaly_threshold_slider/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AnomalyThresholdSlider = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const AnomalyThresholdSlider = ({
  describedByIds = [],
  field
}) => {
  const threshold = field.value;
  const onThresholdChange = (0, _react.useCallback)(event => {
    const thresholdValue = Number(event.currentTarget.value);
    field.setValue(thresholdValue);
  }, [field]);
  return /*#__PURE__*/_react.default.createElement(_eui.EuiFormRow, {
    label: field.label,
    "data-test-subj": "anomalyThresholdSlider",
    describedByIds: describedByIds
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexGroup, null, /*#__PURE__*/_react.default.createElement(_eui.EuiFlexItem, null, /*#__PURE__*/_react.default.createElement(_eui.EuiRange, {
    value: threshold,
    onChange: onThresholdChange,
    fullWidth: true,
    showInput: true,
    showRange: true,
    showTicks: true,
    tickInterval: 25,
    min: 0,
    max: 100
  }))));
};
exports.AnomalyThresholdSlider = AnomalyThresholdSlider;
