// @kbn/synthetics-plugin/public/apps/synthetics/components/monitor_details/monitor_status/labels.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VIEW_HISTORY_LABEL = exports.STATUS_LABEL = exports.SKIPPED_LABEL = exports.LAST_24_HOURS_LABEL = exports.FAILED_LABEL = exports.ERROR_LABEL = exports.COMPLETE_LABEL = exports.BRUSH_AREA_MESSAGE = exports.AVAILABILITY_LABEL = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const AVAILABILITY_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.availability.label', {
  defaultMessage: 'Availability'
});
exports.AVAILABILITY_LABEL = AVAILABILITY_LABEL;
const COMPLETE_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.complete.label', {
  defaultMessage: 'Complete'
});
exports.COMPLETE_LABEL = COMPLETE_LABEL;
const FAILED_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.failed.label', {
  defaultMessage: 'Failed'
});
exports.FAILED_LABEL = FAILED_LABEL;
const SKIPPED_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.skipped.label', {
  defaultMessage: 'Skipped'
});
exports.SKIPPED_LABEL = SKIPPED_LABEL;
const ERROR_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.error.label', {
  defaultMessage: 'Error'
});
exports.ERROR_LABEL = ERROR_LABEL;
const STATUS_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.status', {
  defaultMessage: 'Status'
});
exports.STATUS_LABEL = STATUS_LABEL;
const LAST_24_HOURS_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.last24Hours', {
  defaultMessage: 'Last 24 hours'
});
exports.LAST_24_HOURS_LABEL = LAST_24_HOURS_LABEL;
const VIEW_HISTORY_LABEL = _i18n.i18n.translate('xpack.synthetics.monitorDetails.viewHistory', {
  defaultMessage: 'View History'
});
exports.VIEW_HISTORY_LABEL = VIEW_HISTORY_LABEL;
const BRUSH_AREA_MESSAGE = _i18n.i18n.translate('xpack.synthetics.monitorDetails.brushArea.message', {
  defaultMessage: 'Brush an area for higher fidelity'
});
exports.BRUSH_AREA_MESSAGE = BRUSH_AREA_MESSAGE;
