// @kbn/observability-plugin/public/pages/rule_details/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RULE_DETAILS_TAB_URL_STORAGE_KEY = exports.RULE_DETAILS_SEARCH_BAR_URL_STORAGE_KEY = exports.RULE_DETAILS_PAGE_ID = exports.RULE_DETAILS_EXECUTION_TAB = exports.RULE_DETAILS_ALERTS_TAB = exports.RULE_DETAILS_ALERTS_SEARCH_BAR_ID = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const RULE_DETAILS_EXECUTION_TAB = 'execution';
exports.RULE_DETAILS_EXECUTION_TAB = RULE_DETAILS_EXECUTION_TAB;
const RULE_DETAILS_ALERTS_TAB = 'alerts';
exports.RULE_DETAILS_ALERTS_TAB = RULE_DETAILS_ALERTS_TAB;
const RULE_DETAILS_TAB_URL_STORAGE_KEY = 'tabId';
exports.RULE_DETAILS_TAB_URL_STORAGE_KEY = RULE_DETAILS_TAB_URL_STORAGE_KEY;
const RULE_DETAILS_SEARCH_BAR_URL_STORAGE_KEY = 'searchBarParams';
exports.RULE_DETAILS_SEARCH_BAR_URL_STORAGE_KEY = RULE_DETAILS_SEARCH_BAR_URL_STORAGE_KEY;
const RULE_DETAILS_PAGE_ID = 'rule-details-alerts-o11y';
exports.RULE_DETAILS_PAGE_ID = RULE_DETAILS_PAGE_ID;
const RULE_DETAILS_ALERTS_SEARCH_BAR_ID = 'rule-details-alerts-search-bar-o11y';
exports.RULE_DETAILS_ALERTS_SEARCH_BAR_ID = RULE_DETAILS_ALERTS_SEARCH_BAR_ID;
