// @kbn/security-plugin/public/management/roles/edit_role/privileges/kibana/privilege_summary/privilege_summary.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrivilegeSummary = void 0;
var _eui = require("@elastic/eui");
var _react = _interopRequireWildcard(require("react"));
var _i18nReact = require("@kbn/i18n-react");
var _privilege_summary_table = require("./privilege_summary_table");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PrivilegeSummary = props => {
  const [isOpen, setIsOpen] = (0, _react.useState)(false);
  const numberOfPrivilegeDefinitions = props.role.kibana.length;
  const flyoutSize = numberOfPrivilegeDefinitions > 5 ? 'l' : 'm';
  return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement(_eui.EuiButtonEmpty, {
    onClick: () => setIsOpen(true),
    "data-test-subj": "viewPrivilegeSummaryButton"
  }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.security.management.editRole.privilegeSummary.viewSummaryButtonText",
    defaultMessage: "View privilege summary"
  })), isOpen && /*#__PURE__*/_react.default.createElement(_eui.EuiFlyout, {
    onClose: () => setIsOpen(false),
    size: flyoutSize,
    maskProps: {
      headerZindexLocation: 'below'
    }
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutHeader, null, /*#__PURE__*/_react.default.createElement(_eui.EuiTitle, {
    size: "m"
  }, /*#__PURE__*/_react.default.createElement("h2", null, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.security.management.editRole.privilegeSummary.modalHeaderTitle",
    defaultMessage: "Privilege summary"
  })))), /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutBody, null, /*#__PURE__*/_react.default.createElement(_privilege_summary_table.PrivilegeSummaryTable, {
    role: props.role,
    spaces: props.spaces,
    kibanaPrivileges: props.kibanaPrivileges,
    canCustomizeSubFeaturePrivileges: props.canCustomizeSubFeaturePrivileges,
    spacesApiUi: props.spacesApiUi
  })), /*#__PURE__*/_react.default.createElement(_eui.EuiFlyoutFooter, null, /*#__PURE__*/_react.default.createElement(_eui.EuiButton, {
    onClick: () => setIsOpen(false)
  }, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.security.management.editRole.privilegeSummary.closeSummaryButtonText",
    defaultMessage: "Close"
  })))));
};
exports.PrivilegeSummary = PrivilegeSummary;
