// @kbn/fleet-plugin/public/constants/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  INDEX_NAME: true,
  CUSTOM_LOGS_INTEGRATION_NAME: true,
  DURATION_APM_SETTINGS_VARS: true,
  PLUGIN_ID: true,
  INTEGRATIONS_PLUGIN_ID: true,
  EPM_API_ROUTES: true,
  AGENT_API_ROUTES: true,
  SO_SEARCH_LIMIT: true,
  AGENT_POLICY_SAVED_OBJECT_TYPE: true,
  AGENTS_PREFIX: true,
  PACKAGE_POLICY_SAVED_OBJECT_TYPE: true,
  FLEET_SERVER_PACKAGE: true,
  AGENTS_INDEX: true,
  ENROLLMENT_API_KEYS_INDEX: true,
  AUTO_UPDATE_PACKAGES: true,
  KEEP_POLICIES_UP_TO_DATE_PACKAGES: true,
  AUTO_UPGRADE_POLICIES_PACKAGES: true,
  LOCATORS_IDS: true,
  FLEET_ENROLLMENT_API_PREFIX: true
};
Object.defineProperty(exports, "AGENTS_INDEX", {
  enumerable: true,
  get: function () {
    return _constants.AGENTS_INDEX;
  }
});
Object.defineProperty(exports, "AGENTS_PREFIX", {
  enumerable: true,
  get: function () {
    return _constants.AGENTS_PREFIX;
  }
});
Object.defineProperty(exports, "AGENT_API_ROUTES", {
  enumerable: true,
  get: function () {
    return _constants.AGENT_API_ROUTES;
  }
});
Object.defineProperty(exports, "AGENT_POLICY_SAVED_OBJECT_TYPE", {
  enumerable: true,
  get: function () {
    return _constants.AGENT_POLICY_SAVED_OBJECT_TYPE;
  }
});
Object.defineProperty(exports, "AUTO_UPDATE_PACKAGES", {
  enumerable: true,
  get: function () {
    return _constants.AUTO_UPDATE_PACKAGES;
  }
});
Object.defineProperty(exports, "AUTO_UPGRADE_POLICIES_PACKAGES", {
  enumerable: true,
  get: function () {
    return _constants.AUTO_UPGRADE_POLICIES_PACKAGES;
  }
});
exports.DURATION_APM_SETTINGS_VARS = exports.CUSTOM_LOGS_INTEGRATION_NAME = void 0;
Object.defineProperty(exports, "ENROLLMENT_API_KEYS_INDEX", {
  enumerable: true,
  get: function () {
    return _constants.ENROLLMENT_API_KEYS_INDEX;
  }
});
Object.defineProperty(exports, "EPM_API_ROUTES", {
  enumerable: true,
  get: function () {
    return _constants.EPM_API_ROUTES;
  }
});
Object.defineProperty(exports, "FLEET_ENROLLMENT_API_PREFIX", {
  enumerable: true,
  get: function () {
    return _constants.FLEET_ENROLLMENT_API_PREFIX;
  }
});
Object.defineProperty(exports, "FLEET_SERVER_PACKAGE", {
  enumerable: true,
  get: function () {
    return _constants.FLEET_SERVER_PACKAGE;
  }
});
exports.INDEX_NAME = void 0;
Object.defineProperty(exports, "INTEGRATIONS_PLUGIN_ID", {
  enumerable: true,
  get: function () {
    return _constants.INTEGRATIONS_PLUGIN_ID;
  }
});
Object.defineProperty(exports, "KEEP_POLICIES_UP_TO_DATE_PACKAGES", {
  enumerable: true,
  get: function () {
    return _constants.KEEP_POLICIES_UP_TO_DATE_PACKAGES;
  }
});
Object.defineProperty(exports, "LOCATORS_IDS", {
  enumerable: true,
  get: function () {
    return _constants.LOCATORS_IDS;
  }
});
Object.defineProperty(exports, "PACKAGE_POLICY_SAVED_OBJECT_TYPE", {
  enumerable: true,
  get: function () {
    return _constants.PACKAGE_POLICY_SAVED_OBJECT_TYPE;
  }
});
Object.defineProperty(exports, "PLUGIN_ID", {
  enumerable: true,
  get: function () {
    return _constants.PLUGIN_ID;
  }
});
Object.defineProperty(exports, "SO_SEARCH_LIMIT", {
  enumerable: true,
  get: function () {
    return _constants.SO_SEARCH_LIMIT;
  }
});
var _constants = require("../../common/constants");
var _page_paths = require("./page_paths");
Object.keys(_page_paths).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _page_paths[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _page_paths[key];
    }
  });
});
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const INDEX_NAME = _constants.INGEST_SAVED_OBJECT_INDEX;
exports.INDEX_NAME = INDEX_NAME;
const CUSTOM_LOGS_INTEGRATION_NAME = 'log';
exports.CUSTOM_LOGS_INTEGRATION_NAME = CUSTOM_LOGS_INTEGRATION_NAME;
const DURATION_APM_SETTINGS_VARS = {
  IDLE_TIMEOUT: 'idle_timeout',
  READ_TIMEOUT: 'read_timeout',
  SHUTDOWN_TIMEOUT: 'shutdown_timeout',
  TAIL_SAMPLING_INTERVAL: 'tail_sampling_interval',
  WRITE_TIMEOUT: 'write_timeout'
};
exports.DURATION_APM_SETTINGS_VARS = DURATION_APM_SETTINGS_VARS;
