// @kbn/cross-cluster-replication-plugin/public/app/constants/ui_metric.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UIM_FOLLOWER_INDEX_USE_ADVANCED_OPTIONS = exports.UIM_FOLLOWER_INDEX_UPDATE = exports.UIM_FOLLOWER_INDEX_UNFOLLOW_MANY = exports.UIM_FOLLOWER_INDEX_UNFOLLOW = exports.UIM_FOLLOWER_INDEX_SHOW_DETAILS_CLICK = exports.UIM_FOLLOWER_INDEX_RESUME_MANY = exports.UIM_FOLLOWER_INDEX_RESUME = exports.UIM_FOLLOWER_INDEX_PAUSE_MANY = exports.UIM_FOLLOWER_INDEX_PAUSE = exports.UIM_FOLLOWER_INDEX_LIST_LOAD = exports.UIM_FOLLOWER_INDEX_CREATE = exports.UIM_AUTO_FOLLOW_PATTERN_UPDATE = exports.UIM_AUTO_FOLLOW_PATTERN_SHOW_DETAILS_CLICK = exports.UIM_AUTO_FOLLOW_PATTERN_RESUME_MANY = exports.UIM_AUTO_FOLLOW_PATTERN_RESUME = exports.UIM_AUTO_FOLLOW_PATTERN_PAUSE_MANY = exports.UIM_AUTO_FOLLOW_PATTERN_PAUSE = exports.UIM_AUTO_FOLLOW_PATTERN_LIST_LOAD = exports.UIM_AUTO_FOLLOW_PATTERN_DELETE_MANY = exports.UIM_AUTO_FOLLOW_PATTERN_DELETE = exports.UIM_AUTO_FOLLOW_PATTERN_CREATE = exports.UIM_APP_NAME = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const UIM_APP_NAME = 'cross_cluster_replication';
exports.UIM_APP_NAME = UIM_APP_NAME;
const UIM_FOLLOWER_INDEX_LIST_LOAD = 'follower_index_list_load';
exports.UIM_FOLLOWER_INDEX_LIST_LOAD = UIM_FOLLOWER_INDEX_LIST_LOAD;
const UIM_FOLLOWER_INDEX_CREATE = 'follower_index_create';
exports.UIM_FOLLOWER_INDEX_CREATE = UIM_FOLLOWER_INDEX_CREATE;
const UIM_FOLLOWER_INDEX_UPDATE = 'follower_index_update';
exports.UIM_FOLLOWER_INDEX_UPDATE = UIM_FOLLOWER_INDEX_UPDATE;
const UIM_FOLLOWER_INDEX_PAUSE = 'follower_index_pause';
exports.UIM_FOLLOWER_INDEX_PAUSE = UIM_FOLLOWER_INDEX_PAUSE;
const UIM_FOLLOWER_INDEX_PAUSE_MANY = 'follower_index_pause_many';
exports.UIM_FOLLOWER_INDEX_PAUSE_MANY = UIM_FOLLOWER_INDEX_PAUSE_MANY;
const UIM_FOLLOWER_INDEX_RESUME = 'follower_index_resume';
exports.UIM_FOLLOWER_INDEX_RESUME = UIM_FOLLOWER_INDEX_RESUME;
const UIM_FOLLOWER_INDEX_RESUME_MANY = 'follower_index_resume_many';
exports.UIM_FOLLOWER_INDEX_RESUME_MANY = UIM_FOLLOWER_INDEX_RESUME_MANY;
const UIM_FOLLOWER_INDEX_UNFOLLOW = 'follower_index_unfollow';
exports.UIM_FOLLOWER_INDEX_UNFOLLOW = UIM_FOLLOWER_INDEX_UNFOLLOW;
const UIM_FOLLOWER_INDEX_UNFOLLOW_MANY = 'follower_index_unfollow_many';
exports.UIM_FOLLOWER_INDEX_UNFOLLOW_MANY = UIM_FOLLOWER_INDEX_UNFOLLOW_MANY;
const UIM_FOLLOWER_INDEX_USE_ADVANCED_OPTIONS = 'follower_index_use_advanced_options';
exports.UIM_FOLLOWER_INDEX_USE_ADVANCED_OPTIONS = UIM_FOLLOWER_INDEX_USE_ADVANCED_OPTIONS;
const UIM_FOLLOWER_INDEX_SHOW_DETAILS_CLICK = 'follower_index_show_details_click';
exports.UIM_FOLLOWER_INDEX_SHOW_DETAILS_CLICK = UIM_FOLLOWER_INDEX_SHOW_DETAILS_CLICK;
const UIM_AUTO_FOLLOW_PATTERN_LIST_LOAD = 'auto_follow_patter_list_load';
exports.UIM_AUTO_FOLLOW_PATTERN_LIST_LOAD = UIM_AUTO_FOLLOW_PATTERN_LIST_LOAD;
const UIM_AUTO_FOLLOW_PATTERN_CREATE = 'auto_follow_pattern_create';
exports.UIM_AUTO_FOLLOW_PATTERN_CREATE = UIM_AUTO_FOLLOW_PATTERN_CREATE;
const UIM_AUTO_FOLLOW_PATTERN_UPDATE = 'auto_follow_pattern_update';
exports.UIM_AUTO_FOLLOW_PATTERN_UPDATE = UIM_AUTO_FOLLOW_PATTERN_UPDATE;
const UIM_AUTO_FOLLOW_PATTERN_DELETE = 'auto_follow_pattern_delete';
exports.UIM_AUTO_FOLLOW_PATTERN_DELETE = UIM_AUTO_FOLLOW_PATTERN_DELETE;
const UIM_AUTO_FOLLOW_PATTERN_DELETE_MANY = 'auto_follow_pattern_delete_many';
exports.UIM_AUTO_FOLLOW_PATTERN_DELETE_MANY = UIM_AUTO_FOLLOW_PATTERN_DELETE_MANY;
const UIM_AUTO_FOLLOW_PATTERN_PAUSE = 'auto_follow_pattern_pause';
exports.UIM_AUTO_FOLLOW_PATTERN_PAUSE = UIM_AUTO_FOLLOW_PATTERN_PAUSE;
const UIM_AUTO_FOLLOW_PATTERN_PAUSE_MANY = 'auto_follow_pattern_pause_many';
exports.UIM_AUTO_FOLLOW_PATTERN_PAUSE_MANY = UIM_AUTO_FOLLOW_PATTERN_PAUSE_MANY;
const UIM_AUTO_FOLLOW_PATTERN_RESUME = 'auto_follow_pattern_resume';
exports.UIM_AUTO_FOLLOW_PATTERN_RESUME = UIM_AUTO_FOLLOW_PATTERN_RESUME;
const UIM_AUTO_FOLLOW_PATTERN_RESUME_MANY = 'auto_follow_pattern_resume_many';
exports.UIM_AUTO_FOLLOW_PATTERN_RESUME_MANY = UIM_AUTO_FOLLOW_PATTERN_RESUME_MANY;
const UIM_AUTO_FOLLOW_PATTERN_SHOW_DETAILS_CLICK = 'auto_follow_pattern_show_details_click';
exports.UIM_AUTO_FOLLOW_PATTERN_SHOW_DETAILS_CLICK = UIM_AUTO_FOLLOW_PATTERN_SHOW_DETAILS_CLICK;
