// @kbn/spaces-plugin/public/legacy_urls/components/embeddable_legacy_url_conflict_internal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmbeddableLegacyUrlConflictInternal = void 0;
var _eui = require("@elastic/eui");
var _react = _interopRequireWildcard(require("react"));
var _useAsync = _interopRequireDefault(require("react-use/lib/useAsync"));
var _i18n = require("@kbn/i18n");
var _i18nReact = require("@kbn/i18n-react");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const EmbeddableLegacyUrlConflictInternal = props => {
  const {
    spacesManager,
    getStartServices,
    targetType,
    sourceId
  } = props;
  const [expandError, setExpandError] = (0, _react.useState)(false);
  const {
    value: asyncParams
  } = (0, _useAsync.default)(async () => {
    const [{
      docLinks
    }] = await getStartServices();
    const {
      id: targetSpace
    } = await spacesManager.getActiveSpace();
    const docLink = docLinks.links.spaces.kibanaDisableLegacyUrlAliasesApi;
    const aliasJsonString = JSON.stringify({
      targetSpace,
      targetType,
      sourceId
    }, null, 2);
    return {
      docLink,
      aliasJsonString
    };
  }, [getStartServices, spacesManager]);
  const {
    docLink,
    aliasJsonString
  } = asyncParams !== null && asyncParams !== void 0 ? asyncParams : {};
  if (!aliasJsonString || !docLink) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
    id: "xpack.spaces.embeddableLegacyUrlConflict.messageText",
    defaultMessage: "We found 2 saved objects for this panel. Disable the legacy URL alias to fix this error."
  }), /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, null), expandError ? /*#__PURE__*/_react.default.createElement(_eui.EuiTextAlign, {
    textAlign: "left"
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiCallOut, {
    title: /*#__PURE__*/_react.default.createElement(_i18nReact.FormattedMessage, {
      id: "xpack.spaces.embeddableLegacyUrlConflict.calloutTitle",
      defaultMessage: "Copy this JSON and use it with the {documentationLink}",
      values: {
        documentationLink: /*#__PURE__*/_react.default.createElement(_eui.EuiLink, {
          external: true,
          href: docLink,
          target: "_blank"
        }, '_disable_legacy_url_aliases API')
      }
    }),
    color: "danger",
    iconType: "warning"
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiCodeBlock, {
    fontSize: "s",
    language: "json",
    isCopyable: true,
    paddingSize: "none"
  }, aliasJsonString))) : /*#__PURE__*/_react.default.createElement(_eui.EuiButtonEmpty, {
    onClick: () => setExpandError(true)
  }, _i18n.i18n.translate('xpack.spaces.embeddableLegacyUrlConflict.detailsButton', {
    defaultMessage: `View details`
  })));
};
exports.EmbeddableLegacyUrlConflictInternal = EmbeddableLegacyUrlConflictInternal;
