// @kbn/lens-plugin/public/datasources/form_based/operations/definitions/ranges/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TYPING_DEBOUNCE_TIME = exports.TO_PLACEHOLDER = exports.SLICES = exports.MODES = exports.MIN_HISTOGRAM_BARS = exports.FROM_PLACEHOLDER = exports.DEFAULT_INTERVAL = exports.AUTO_BARS = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const TYPING_DEBOUNCE_TIME = 256;
// Taken from the Visualize editor
exports.TYPING_DEBOUNCE_TIME = TYPING_DEBOUNCE_TIME;
const FROM_PLACEHOLDER = '\u2212\u221E';
exports.FROM_PLACEHOLDER = FROM_PLACEHOLDER;
const TO_PLACEHOLDER = '+\u221E';
exports.TO_PLACEHOLDER = TO_PLACEHOLDER;
const DEFAULT_INTERVAL = 1000;
exports.DEFAULT_INTERVAL = DEFAULT_INTERVAL;
const AUTO_BARS = 'auto';
exports.AUTO_BARS = AUTO_BARS;
const MIN_HISTOGRAM_BARS = 1;
exports.MIN_HISTOGRAM_BARS = MIN_HISTOGRAM_BARS;
const SLICES = 6;
exports.SLICES = SLICES;
const MODES = {
  Range: 'range',
  Histogram: 'histogram'
};
exports.MODES = MODES;
