// @kbn/monitoring-plugin/public/alerts/panel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AlertPanel = void 0;
var _react = _interopRequireWildcard(require("react"));
var _eui = require("@elastic/eui");
var _replace_tokens = require("./lib/replace_tokens");
var _setup_mode = require("../lib/setup_mode");
var _setup_mode_context = require("../components/setup_mode/setup_mode_context");
var _configuration = require("./configuration");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const AlertPanel = props => {
  const {
    alert,
    alertState
  } = props;
  const inSetupMode = (0, _setup_mode.isInSetupMode)(_react.default.useContext(_setup_mode_context.SetupModeContext));
  if (!alert) {
    return null;
  }
  if (inSetupMode || !alertState) {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        padding: '1rem'
      }
    }, /*#__PURE__*/_react.default.createElement(_configuration.AlertConfiguration, {
      alert: alert
    }));
  }
  const nextStepsUi = alertState.state.ui.message.nextSteps && alertState.state.ui.message.nextSteps.length ? /*#__PURE__*/_react.default.createElement(_eui.EuiListGroup, null, alertState.state.ui.message.nextSteps.map((step, index) => /*#__PURE__*/_react.default.createElement(_eui.EuiListGroupItem, {
    size: "s",
    key: index,
    label: (0, _replace_tokens.replaceTokens)(step)
  }))) : null;
  const {
    code
  } = alertState.state.ui.message;
  return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: '1rem'
    }
  }, /*#__PURE__*/_react.default.createElement(_eui.EuiTitle, {
    size: "xs"
  }, /*#__PURE__*/_react.default.createElement("h5", null, (0, _replace_tokens.replaceTokens)(alertState.state.ui.message))), code !== null && code !== void 0 && code.length ? /*#__PURE__*/_react.default.createElement(_eui.EuiCodeBlock, {
    fontSize: "s",
    paddingSize: "s",
    language: "json",
    isCopyable: true,
    overflowHeight: 150
  }, code) : null, nextStepsUi ? /*#__PURE__*/_react.default.createElement(_eui.EuiSpacer, {
    size: "s"
  }) : null, nextStepsUi), /*#__PURE__*/_react.default.createElement(_eui.EuiHorizontalRule, {
    margin: "s"
  }), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: '0 1rem 1rem 1rem'
    }
  }, /*#__PURE__*/_react.default.createElement(_configuration.AlertConfiguration, {
    alert: alert
  })));
};
exports.AlertPanel = AlertPanel;
