// @kbn/elastic-assistant-plugin/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.POST_ACTIONS_CONNECTOR_EXECUTE = exports.PLUGIN_NAME = exports.PLUGIN_ID = exports.KNOWLEDGE_BASE = exports.EVALUATE = exports.BASE_PATH = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const PLUGIN_ID = 'elasticAssistant';
exports.PLUGIN_ID = PLUGIN_ID;
const PLUGIN_NAME = 'elasticAssistant';
exports.PLUGIN_NAME = PLUGIN_NAME;
const BASE_PATH = '/internal/elastic_assistant';
exports.BASE_PATH = BASE_PATH;
const POST_ACTIONS_CONNECTOR_EXECUTE = `${BASE_PATH}/actions/connector/{connectorId}/_execute`;

// Knowledge Base
exports.POST_ACTIONS_CONNECTOR_EXECUTE = POST_ACTIONS_CONNECTOR_EXECUTE;
const KNOWLEDGE_BASE = `${BASE_PATH}/knowledge_base/{resource?}`;

// Model Evaluation
exports.KNOWLEDGE_BASE = KNOWLEDGE_BASE;
const EVALUATE = `${BASE_PATH}/evaluate`;
exports.EVALUATE = EVALUATE;
