// @kbn/stack-connectors-plugin/public/common/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OPENAI_CONNECTOR_ID", {
  enumerable: true,
  get: function () {
    return _constants.OPENAI_CONNECTOR_ID;
  }
});
Object.defineProperty(exports, "OpenAILogo", {
  enumerable: true,
  get: function () {
    return _logo.default;
  }
});
Object.defineProperty(exports, "OpenAiProviderType", {
  enumerable: true,
  get: function () {
    return _constants.OpenAiProviderType;
  }
});
Object.defineProperty(exports, "SENTINELONE_CONNECTOR_ID", {
  enumerable: true,
  get: function () {
    return _constants2.SENTINELONE_CONNECTOR_ID;
  }
});
Object.defineProperty(exports, "SUB_ACTION", {
  enumerable: true,
  get: function () {
    return _constants2.SUB_ACTION;
  }
});
Object.defineProperty(exports, "SentinelOneLogo", {
  enumerable: true,
  get: function () {
    return _logo2.default;
  }
});
var _logo = _interopRequireDefault(require("../connector_types/openai/logo"));
var _constants = require("../../common/openai/constants");
var _logo2 = _interopRequireDefault(require("../connector_types/sentinelone/logo"));
var _constants2 = require("../../common/sentinelone/constants");
