// @kbn/logs-shared-plugin/public/components/loading/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogsSharedLoadingStaticPanel = exports.LogsSharedLoadingStaticContentPanel = exports.LogsSharedLoadingPanel = void 0;
var _eui = require("@elastic/eui");
var React = _interopRequireWildcard(require("react"));
var _common = require("@kbn/kibana-react-plugin/common");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

class LogsSharedLoadingPanel extends React.PureComponent {
  render() {
    const {
      height,
      text,
      width
    } = this.props;
    return /*#__PURE__*/React.createElement(LogsSharedLoadingStaticPanel, {
      style: {
        height,
        width
      }
    }, /*#__PURE__*/React.createElement(LogsSharedLoadingStaticContentPanel, null, /*#__PURE__*/React.createElement(_eui.EuiPanel, null, /*#__PURE__*/React.createElement(_eui.EuiLoadingChart, {
      size: "m"
    }), /*#__PURE__*/React.createElement(_eui.EuiText, null, /*#__PURE__*/React.createElement("p", null, text)))));
  }
}
exports.LogsSharedLoadingPanel = LogsSharedLoadingPanel;
const LogsSharedLoadingStaticPanel = _common.euiStyled.div`
  position: relative;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
exports.LogsSharedLoadingStaticPanel = LogsSharedLoadingStaticPanel;
const LogsSharedLoadingStaticContentPanel = _common.euiStyled.div`
  flex: 0 0 auto;
  align-self: center;
  text-align: center;
`;
exports.LogsSharedLoadingStaticContentPanel = LogsSharedLoadingStaticContentPanel;
