// @kbn/enterprise-search-plugin/public/applications/app_search/components/result_settings/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SIZE_FIELD_MINIMUM = exports.SIZE_FIELD_MAXIMUM = exports.RESULT_SETTINGS_TITLE = exports.DISABLED_FIELD_SETTINGS = exports.DEFAULT_SNIPPET_SIZE = exports.DEFAULT_FIELD_SETTINGS = void 0;
var _i18n = require("@kbn/i18n");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

const DEFAULT_SNIPPET_SIZE = 100;
exports.DEFAULT_SNIPPET_SIZE = DEFAULT_SNIPPET_SIZE;
const SIZE_FIELD_MINIMUM = 20;
exports.SIZE_FIELD_MINIMUM = SIZE_FIELD_MINIMUM;
const SIZE_FIELD_MAXIMUM = 1000;
exports.SIZE_FIELD_MAXIMUM = SIZE_FIELD_MAXIMUM;
const RESULT_SETTINGS_TITLE = _i18n.i18n.translate('xpack.enterpriseSearch.appSearch.engine.resultSettings.title', {
  defaultMessage: 'Result Settings'
});
exports.RESULT_SETTINGS_TITLE = RESULT_SETTINGS_TITLE;
const DEFAULT_FIELD_SETTINGS = {
  raw: true,
  snippet: false,
  snippetFallback: false
};
exports.DEFAULT_FIELD_SETTINGS = DEFAULT_FIELD_SETTINGS;
const DISABLED_FIELD_SETTINGS = {
  raw: false,
  snippet: false,
  snippetFallback: false
};
exports.DISABLED_FIELD_SETTINGS = DISABLED_FIELD_SETTINGS;
